<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><meta name="keywords" content="Hexo Theme Redefine"><meta name="author" content="Yuchen"><link rel="preconnect" href="https://fonts.googleapis.com"><link rel="preconnect" href="https://fonts.gstatic.com" crossorigin><link rel="preconnect" href="https://cdn.jsdelivr.net" crossorigin><link rel="canonical" href="http://example.com/2025/03/16/gwas review/"><meta name="robots" content="index,follow"><meta name="googlebot" content="index,follow"><meta name="revisit-after" content="1 days"><meta name="description" content="Hexo Theme Redefine, Redefine Your Hexo Journey."><meta property="og:type" content="article"><meta property="og:title" content="Review of GWAS"><meta property="og:url" content="http://example.com/2025/03/16/GWAS%20review/index.html"><meta property="og:site_name" content="Hexo"><meta property="og:description" content="Hexo Theme Redefine, Redefine Your Hexo Journey."><meta property="og:locale" content="en_US"><meta property="og:image" content="http://example.com/images/redefine-og.webp"><meta property="article:published_time" content="2025-03-16T06:37:00.000Z"><meta property="article:modified_time" content="2025-04-08T12:24:43.884Z"><meta property="article:author" content="John Doe"><meta property="article:tag" content="Population Genetic Structure"><meta property="article:tag" content="Article"><meta property="article:tag" content="GWAS"><meta property="article:tag" content="linkage disequilibrium（LD）"><meta name="twitter:card" content="summary"><meta name="twitter:image" content="http://example.com/images/redefine-og.webp"><link rel="icon" type="image/png" href="/images/profile.jpg" sizes="192x192"><link rel="apple-touch-icon" sizes="180x180" href="/images/profile.jpg"><meta name="theme-color" content="#A31F34"><link rel="shortcut icon" href="/images/profile.jpg"><title>Review of GWAS | Steps Among the Stars</title><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/Chillax/chillax.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/css/build/tailwind.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/GeistMono/geist-mono.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/Geist/geist.css"><script id="hexo-configurations">window.config={hostname:"example.com",root:"/",language:"en"},window.theme={articles:{style:{font_size:"16px",line_height:1.5,image_border_radius:"14px",image_alignment:"center",image_caption:!1,link_icon:!0,delete_mask:!1,title_alignment:"left",headings_top_spacing:{h1:"3.2rem",h2:"2.4rem",h3:"1.9rem",h4:"1.6rem",h5:"1.4rem",h6:"1.3rem"}},word_count:{enable:!0,count:!0,min2read:!0},author_label:{enable:!1,auto:!1,list:[]},code_block:{copy:!0,style:"mac",highlight_theme:{light:"github",dark:"vs2015"},font:{enable:!1,family:null,url:null}},toc:{enable:!0,max_depth:6,number:!1,expand:!0,init_open:!0},copyright:{enable:!0,default:"CC BY-NC-ND"},lazyload:!0,pangu_js:!1,recommendation:{enable:!1,title:"推荐阅读",limit:3,mobile_limit:2,placeholder:"/images/wallhaven-wqery6-light.webp",skip_dirs:[]}},colors:{primary:"#A31F34",secondary:null,default_mode:"light"},global:{fonts:{chinese:{enable:!1,family:null,url:null},english:{enable:!1,family:null,url:null},title:{enable:!1,family:null,url:null}},content_max_width:"1000px",sidebar_width:"210px",hover:{shadow:!0,scale:!1},scroll_progress:{bar:!1,percentage:!1},website_counter:{url:"https://cn.vercount.one/js",enable:!0,site_pv:!0,site_uv:!0,post_pv:!0},single_page:!0,preloader:{enable:"ture",custom_message:"Welcome to my little corner of chaos and brilliance!"},open_graph:{enable:!0,image:"/images/redefine-og.webp",description:"Hexo Theme Redefine, Redefine Your Hexo Journey."},google_analytics:{enable:!1,id:null}},home_banner:{enable:!0,style:"fixed",image:{light:"/images/light.jpeg",dark:"/images/dark.png"},title:"Resonant Journals",subtitle:{text:["Welcome to my little corner of chaos and brilliance!"],hitokoto:{enable:!1,show_author:!1,api:"https://v1.hitokoto.cn"},typing_speed:20,backing_speed:80,starting_delay:500,backing_delay:1500,loop:!1,smart_backspace:!0},text_color:{light:"#fff",dark:"#d1d1b6"},text_style:{title_size:"2.8rem",subtitle_size:"1.5rem",line_height:1.2},custom_font:{enable:!1,family:null,url:null},social_links:{enable:"ture",style:"default",links:{github:null,instagram:null,zhihu:null,twitter:null,email:"yuchen.starfall@outlook.com"},qrs:{weixin:null}}},plugins:{feed:{enable:!1},aplayer:{enable:!1,type:"fixed",audios:[{name:null,artist:null,url:null,cover:null,lrc:null}]},mermaid:{enable:!1,version:"11.4.1"}},version:"2.8.2",navbar:{auto_hide:"ture",color:{left:"#f78736",right:"#367df7",transparency:35},width:{home:"1200px",pages:"1000px"},links:{Home:{path:"/",icon:"fa-regular fa-house"},Archives:{path:"/archives",icon:"fa-regular fa-archive"},Categories:{icon:"fa-solid fa-folder",path:"/categories/"},Tags:{icon:"fa-solid fa-tags",path:"/tags/"}},search:{enable:!1,preload:!0}},page_templates:{friends_column:2,tags_style:"blur"},home:{sidebar:{enable:!0,position:"left",first_item:"menu",announcement:"Ephemeral Trails",show_on_mobile:!0,links:{Archives:{path:"/archives",icon:"fa-regular fa-archive"},Tags:{path:"/tags",icon:"fa-regular fa-tags"},Categories:{path:"/categories",icon:"fa-regular fa-folder"}}},article_date_format:"auto",excerpt_length:200,categories:{enable:!0,limit:3},tags:{enable:!0,limit:3}},footerStart:"2024/12/5 00:00:00"},window.lang_ago={second:"%s seconds ago",minute:"%s minutes ago",hour:"%s hours ago",day:"%s days ago",week:"%s weeks ago",month:"%s months ago",year:"%s years ago"},window.data={masonry:!0}</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/fontawesome.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/brands.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/solid.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/regular.min.css"><meta name="generator" content="Hexo 7.3.0"></head><body><div class="progress-bar-container"><span class="pjax-progress-bar"></span></div><main class="page-container" id="swup"><div class="main-content-container flex flex-col justify-between min-h-dvh"><div class="main-content-header"><header class="navbar-container px-6 md:px-12"><div class="navbar-content transition-navbar"><div class="left"><a class="logo-title" href="/">Steps Among the Stars</a></div><div class="right"><div class="desktop"><ul class="navbar-list"><li class="navbar-item"><a href="/"><i class="fa-regular fa-house fa-fw"></i> HOME</a></li><li class="navbar-item"><a href="/archives"><i class="fa-regular fa-archive fa-fw"></i> ARCHIVES</a></li><li class="navbar-item"><a href="/categories/"><i class="fa-solid fa-folder fa-fw"></i> CATEGORIES</a></li><li class="navbar-item"><a href="/tags/"><i class="fa-solid fa-tags fa-fw"></i> TAGS</a></li></ul></div><div class="mobile"><div class="icon-item navbar-bar"><div class="navbar-bar-middle"></div></div></div></div></div><div class="navbar-drawer h-dvh w-full absolute top-0 left-0 bg-background-color flex flex-col justify-between"><ul class="drawer-navbar-list flex flex-col px-4 justify-center items-start"><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/"><span>HOME </span><i class="fa-regular fa-house fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/archives"><span>ARCHIVES </span><i class="fa-regular fa-archive fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/categories/"><span>CATEGORIES </span><i class="fa-solid fa-folder fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/tags/"><span>TAGS </span><i class="fa-solid fa-tags fa-sm fa-fw"></i></a></li></ul><div class="statistics flex justify-around my-2.5"><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/tags"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">48</div><div class="label text-third-text-color text-sm">Tags</div></a><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/categories"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">6</div><div class="label text-third-text-color text-sm">Categories</div></a><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/archives"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">51</div><div class="label text-third-text-color text-sm">Posts</div></a></div></div><div class="window-mask"></div></header></div><div class="main-content-body transition-fade-up"><div class="main-content"><div class="post-page-container flex relative justify-between box-border w-full h-full"><div class="article-content-container"><div class="article-title relative w-full"><img src="/img/img-GWAS/1.png" alt="Review of GWAS" class="w-full h-60 sm:h-72 md:h-80 object-cover sm:rounded-t-large dark:brightness-75"><div class="w-full flex items-center absolute bottom-0 justify-start"><h1 class="article-title-cover text-center mx-6 my-6 text-second-text-color bg-background-color-transparent px-4 py-3 text-3xl sm:text-4xl md:text-5xl font-semibold backdrop-blur-lg rounded-xl border border-border-color">Review of GWAS</h1></div></div><div class="article-header flex flex-row gap-2 items-center px-2 sm:px-6 md:px-8"><div class="avatar w-[46px] h-[46px] flex-shrink-0 rounded-medium border border-border-color p-[1px]"><img src="/images/profile.jpg"></div><div class="info flex flex-col justify-between"><div class="author flex items-center"><span class="name text-default-text-color text-lg font-semibold">Yuchen</span></div><div class="meta-info"><div class="article-meta-info"><span class="article-date article-meta-item"><i class="fa-regular fa-pen-fancy"></i>&nbsp; <span class="desktop">2025-03-16 14:37</span> <span class="mobile">2025-03-16 14:37</span> <span class="hover-info">Created</span> </span><span class="article-date article-meta-item"><i class="fa-regular fa-wrench"></i>&nbsp; <span class="desktop">2025-04-08 20:24:43</span> <span class="mobile">2025-04-08 20:24:43</span> <span class="hover-info">Updated</span> </span><span class="article-categories article-meta-item"><i class="fa-regular fa-folders"></i>&nbsp;<ul><li><a href="/categories/Literature/">Literature</a>&nbsp;</li></ul></span><span class="article-tags article-meta-item"><i class="fa-regular fa-tags"></i>&nbsp;<ul><li><a href="/tags/Population-Genetic-Structure/">Population Genetic Structure</a>&nbsp;</li><li>| <a href="/tags/Article/">Article</a>&nbsp;</li><li>| <a href="/tags/GWAS/">GWAS</a>&nbsp;</li><li>| <a href="/tags/linkage-disequilibrium%EF%BC%88LD%EF%BC%89/">linkage disequilibrium（LD）</a>&nbsp;</li></ul></span><span class="article-pv article-meta-item"><i class="fa-regular fa-eye"></i>&nbsp;<span id="busuanzi_value_page_pv"></span></span></div></div></div></div><div class="article-content markdown-body px-2 sm:px-6 md:px-8 pb-8"><blockquote><p>Title: Genome-wide association studies for common diseases and complex traits</p><p>Excerpt:介绍最早的关于GWAS介绍的Review之一，于GWAS刚刚萌芽的2005年发表，那时人类基因组测序刚刚完成，dbSNP（<strong>The Single Nucleotide Polymorphism Database</strong>）开始建立，Hapmap（Haplotype Map（一组在遗传上紧密连锁的 SNP，它们倾向于一起遗传））项目也开始启动，这些项目奠定了GWAS研究发展的基础。这篇综述该介绍了GWAS相比于传统遗传学方法的优缺点，当时可用的测序高通量测序方法，以及GWAS研究中需要注意的核心问题等。将传统遗传学与现代基因组学衔接的一篇开山之作之一。</p></blockquote><h3 id="Why-genome-wide-association-studies"><a href="#Why-genome-wide-association-studies" class="headerlink" title="Why genome-wide association studies?"></a>Why genome-wide association studies?</h3><p>Mapping the genes that underlie common disease and quantitative traits：</p><ul><li>Candidate-gene studies. (已知验证)<ul><li><strong>关联分析</strong>：比较患病群体和健康群体中特定基因变异的频率，寻找与疾病相关的变异。</li><li><strong>重测序</strong>：对候选基因进行深度测序，以发现新的或罕见的变异。</li></ul></li><li>Genome-wide studies.(探索未知)<ul><li><strong>Linkage mapping</strong>:通过家族研究，利用遗传标记追踪疾病或性状在家族中的传递模式，定位相关基因区域。</li><li><strong>Genome-wide association studies</strong>:通过大规模样本，扫描全基因组范围内的常见变异，寻找与疾病或性状显著相关的位点。</li></ul></li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-GWAS/2.png"></p><h4 id="Linkage-analysis"><a href="#Linkage-analysis" class="headerlink" title="Linkage analysis"></a>Linkage analysis</h4><ul><li><p>定位单基因“孟德尔”疾病的基因方面取得了巨大成功。为了使连锁分析成功，必须确保与疾病基因两侧的<strong>标记</strong>在家族中与疾病<strong>共分离</strong>（<strong>co-segregation</strong>）。</p><ul><li>在家族中，如果某个遗传标记与疾病表型总是（或几乎总是）一起出现，那么它们就是“共分离”的。</li><li>这意味着该标记与疾病基因在物理位置上非常接近，位于同一染色体区域。</li></ul></li><li><p>So each segregating disease allele will be found in the same 10–20 cM chromosomal background within each family.</p><ul><li>在家族中，染色体片段通常是以较大的区块（如10-20 cM）为单位遗传的。这是因为减数分裂过程中，染色体之间的重组事件相对较少。</li><li>因此，导致疾病的变异和其附近的遗传标记（如SNP）会作为一个整体在家族中传递。</li></ul></li><li><p>发现的基因通常只能解释疾病总体遗传力的一小部分。 – <strong>丢失的遗传力</strong>的出现</p><ul><li><strong>遗传力（Heritability）</strong>：指疾病风险中可归因于遗传因素的比例。若遗传力高（如80%），说明遗传因素占主导；若低（如20%），则环境因素更重要。在数学上使用<strong>总遗传方差</strong>（包括加性、显性和上位效应）<strong>占表型总方差的比例</strong>来表示</li><li>广义遗传力：$H^2&#x3D;\frac{V_G}{V_P}$<ul><li>$V_G$：遗传方差（Genetic Variance）</li><li>$V_P$：表型总方差（Phenotypic Variance）$&#x3D; V_G+V_E+V_{GE}$<ul><li>$V_E$：环境方差（Environmental Variance）</li><li>$V_{GE}$：基因-环境交互作用方差</li></ul></li></ul></li><li>狭义遗传力：$h^2&#x3D;\frac{V_A}{V_P}$<ul><li>$V_A$：加性遗传方差（亲代传递给子代的遗传效应）</li></ul></li><li>遗传力缺失指通过统计方法（如GWAS）发现的遗传变异仅能解释部分表型方差（定义为 <strong>h2 family</strong> 与 <strong>h2 GWAS</strong> 之间的差值。 <strong>h2 GWAS</strong> 之所以低于 <strong>h2 family</strong> ，），而家族或双生子研究估计的总遗传力更高，普遍认为的影响因素 1）<strong>未检测到的遗传变异</strong> 2）<strong>非加性遗传效应</strong> 3）**基因-环境交互作用（G×E）**4）<strong>统计方法与样本限制</strong><ul><li><strong>h2 family</strong> ： 双胞胎研究，通过比较同卵与异卵双胞胎的相似性，计算得到h2，通常为这三种中最高。</li><li><strong>h2 SNP</strong> ：GWAS研究所用chip上所有variants共同解释的方差 与 性状方差的比值，比 h2 family 低，但会显著高于h2 GWAS。可以使用GCTA的GREML模型来估计。</li><li><strong>h2 GWAS</strong> ：仅由GWAS所发现的某疾病相关variants解释的方差 与 性状方差的比值 ，三者中最低。</li></ul></li></ul></li><li><p>Importantly, linkage analysis is also much less powerful for identifying common genetic variants that have modest effects on disease</p><ul><li><strong>常见疾病和数量性状的遗传基础可能同时涉及两种类型的遗传变异</strong>：<ol><li><strong>常见等位基因</strong>（频率较高，如人群中&gt;1%的变异）</li><li><strong>罕见等位基因</strong>（频率较低，如人群中&lt;1%的变异）</li></ol></li><li>遗传异质性（genetic heterogeneity）的重要性，即不同变异类型共同塑造了常见疾病和数量性状的遗传基础</li></ul></li></ul><h4 id="Candidate-gene-resequencing-studies"><a href="#Candidate-gene-resequencing-studies" class="headerlink" title="Candidate-gene resequencing studies"></a>Candidate-gene resequencing studies</h4><p><strong>重测序</strong> 是指对特定基因或基因组区域进行深度测序，以全面检测该区域内所有可能的遗传变异（如单核苷酸多态性SNP、插入缺失Indel、结构变异等）。与全基因组测序（覆盖整个基因组）不同，重测序通常聚焦于已知的候选基因或疾病相关区域，旨在发现与表型（如疾病）直接相关的变异。 – 精准发现<strong>罕见</strong>&#x2F;编码区变异</p><ol><li><strong>发现罕见或低频变异</strong></li></ol><ul><li><strong>常见变异</strong>（如GWAS发现的SNP）通常频率高（&gt;1%），但效应小；</li><li><strong>罕见变异</strong>（频率&lt;1%）可能对疾病有较大影响，但需要深度测序才能发现。</li><li><strong>例子</strong>：MC4R基因中的<strong>罕见</strong>变异可能导致严重早发性肥胖。<ul><li>PPARG基因的Pro12Ala常见变异轻微增加糖尿病风险，而罕见无义突变导致严重代谢障碍。</li></ul></li></ul><ol start="2"><li><strong>验证候选基因的功能</strong></li></ol><ul><li>在连锁分析或GWAS定位的基因组区域内，重测序可精准识别致病基因的变异。</li><li><strong>例子</strong>：若某区域与糖尿病相关，重测序可筛查该区域内所有基因的变异，锁定具体致病位点。</li></ul><ol start="3"><li><strong>解析编码与非编码变异的影响</strong></li></ol><ul><li><strong>编码区变异</strong>（如错义突变）直接影响蛋白质功能；</li><li><strong>非编码区变异</strong>（如启动子区）可能调控基因表达，但功能解析较复杂。</li></ul><ol start="4"><li><strong>补充其他方法的不足</strong></li></ol><ul><li>GWAS难以检测罕见变异，连锁分析分辨率低，重测序可填补这些空白</li></ul><h4 id="Association-studies"><a href="#Association-studies" class="headerlink" title="Association studies"></a>Association studies</h4><p>关联研究是一种通过比较患病群体与健康群体中遗传变异的频率差异，寻找与疾病或性状相关的遗传标记的研究方法。</p><ul><li><strong>群体分层影响（Population Stratification）</strong> 是遗传关联研究中的一种潜在偏倚，指研究样本中存在不同祖先背景的亚群体，这些亚群体因遗传背景和疾病风险的差异，导致虚假的关联结果。<ul><li><strong>例子</strong>：假设一项研究同时纳入欧洲裔和非洲裔人群，若欧洲裔人群中某疾病更常见，而某个SNP在欧洲裔中频率也更高，即使该SNP与疾病无关，统计上也可能显示虚假关联。</li><li>可以通过**主成分分析（PCA）**将样本按祖先背景投影到二维空间，识别潜在的亚群体结构（如欧洲人、非洲人聚类成不同群组）</li><li>可以使用**基于家庭的关联研究（Family-Based Association Study）**避免群体分层（Population Stratification）的干扰</li></ul></li><li>关键技术：<strong>连锁不平衡（Linkage Disequilibrium, LD）</strong><ul><li><strong>定义</strong>：基因组中相邻变异非随机关联的现象，即某些等位基因倾向于共同遗传。<ul><li>通过LD，可用少量标签SNP（Tag SNP）代理附近多个相关SNP，大幅减少检测成本，同时覆盖大部分遗传变异。</li></ul></li><li><strong>强LD片段</strong>：基因组大部分区域由长片段（约500 kb）的强LD组成，片段内SNP高度相关，共享少数常见单倍型（Haplotype）。<ul><li><strong>单倍型的作用</strong>：一条染色体上相邻SNP常以固定组合（单倍型）共同传递，使得单个标签SNP即可代表整个单倍型区域。</li></ul></li><li>局限性<ul><li><strong>常见变异偏向（Common Disease-Common Variant Bias）</strong>：<ul><li>GWAS主要检测常见变异，可能忽略罕见高效应变异。</li></ul></li><li><strong>遗传力缺失（Missing Heritability）</strong>：<ul><li>已发现变异通常解释不足总遗传力的50%（如精神分裂症约25%）。</li></ul></li><li><strong>功能机制不明确</strong>：<ul><li>多数关联SNP位于非编码区，功能解析困难（需结合eQTL、染色质互作数据）。</li></ul></li><li><strong>群体分层干扰</strong>：<ul><li>病例与对照组祖先背景差异可能导致假阳性（需PCA校正）。</li></ul></li></ul></li></ul></li></ul><h3 id="Strategies-to-increase-efficiency"><a href="#Strategies-to-increase-efficiency" class="headerlink" title="Strategies to increase efficiency"></a>Strategies to increase efficiency</h3><h4 id="Selecting-an-appropriate-sample-size"><a href="#Selecting-an-appropriate-sample-size" class="headerlink" title="Selecting an appropriate sample size"></a>Selecting an appropriate sample size</h4><ul><li><p><strong>样本量困境</strong></p><ul><li><p>GWAS成本高昂，限制样本量会降低统计效力，但复杂性状相关变异的弱效应需大样本。</p></li><li><p>多重假设检验校正（如Bonferroni法）要求更严格的p值阈值（如α&#x3D;0.05，N&#x3D;100万次检验 → 阈值&#x3D; 0.05&#x2F;1,000,000&#x3D;5×10−80.05&#x2F;1,000,000&#x3D;5×10⁻⁸）。</p></li></ul></li><li><p><strong>统计效力（Power）的影响</strong></p><ul><li><p><strong>定义</strong>：<strong>统计效力</strong>（Statistical Power）指在假设检验中，当<strong>备择假设（H₁）为真</strong>时，正确拒绝**原假设（H₀）**的概率。简单来说，它是研究能够检测到真实存在效应的能力。（通常要求≥80%）。</p></li><li><p><strong>小效应（OR≈1.25）</strong>：</p><ul><li><p>效应越小，区分真实信号与随机噪声越难。</p></li><li><p>需要更多样本以提高检测灵敏度。</p><ul><li>若要求p&#x3D;5×10⁻⁸且效力80%，OR&#x3D;1.25需约6,000病例+6,000对照。</li></ul></li></ul></li></ul></li><li><p><strong>假阳性风险</strong></p><ul><li>宽松阈值下，50万SNP研究会产生2.5万假阳性，掩盖真实信号。</li><li>需后续验证研究以排除假阳性。</li></ul></li></ul><p>​</p><h4 id="A-multi-stage-approach"><a href="#A-multi-stage-approach" class="headerlink" title="A multi-stage approach"></a>A multi-stage approach</h4><ol><li><p><strong>初筛阶段</strong>：在全基因组范围内，使用较宽松的p值阈值筛选潜在关联位点。此阈值需平衡检测弱效应（如解释表型方差比例小的位点）的能力，同时允许一定数量的假阳性通过。</p></li><li><p><strong>验证阶段</strong>：对初筛通过的位点，在独立样本中重新检验（样本量≥初筛阶段）。由于仅需对少量位点分型，大幅降低成本。若使用不同分型技术，还可减少技术误差导致的假阳性。</p></li><li><p><strong>联合阈值设定</strong>：通过两阶段结果的联合分析，确保最终假阳性率低于5%。必要时可增加第三阶段验证以提高严谨性。</p><ul><li><p><strong>Bonferroni校正的局限性</strong>：假设所有检验独立，但实际SNP间存在连锁不平衡（LD），且单SNP与单倍型检验存在冗余，导致过度保守（需极低p值和超大样本）。</p></li><li><p><strong>替代方法</strong>：</p><ul><li><strong>置换检验（Permutation）</strong>：通过随机打乱表型与基因型数据，经验估计假阳性率。</li><li><strong>频率学派方法</strong>：计算全基因组显著性阈值（如考虑LD结构）。</li><li><strong>贝叶斯方法</strong>：评估关联真实性的后验概率。</li></ul></li></ul></li></ol><p><img lazyload src="/images/loading.svg" data-src="/img/img-GWAS/3.png"></p><h4 id="Founder-populations-and-pooled-samples"><a href="#Founder-populations-and-pooled-samples" class="headerlink" title="Founder populations and pooled samples"></a>Founder populations and pooled samples</h4><ol><li><strong>创始人群体（Founder Populations）</strong><ul><li><strong>原理</strong>：利用遗传同质性高的群体（如芬兰人、阿米什人），其基因组连锁不平衡（LD）范围更大，所需分型SNP数量减少。<ul><li><strong>遗传瓶颈效应</strong>：创始人群体祖先数量少，基因组LD区域更长，少量标签SNP（Tag SNP）即可覆盖全基因组变异。</li></ul></li><li><strong>优势</strong>：显著降低分型成本，尤其适用于单基因孟德尔病（如囊性纤维化）。</li><li><strong>局限</strong>：对复杂疾病（如糖尿病）的基因定位效果有限，因多基因微效变异仍需大样本。</li></ul></li><li><strong>混合样本（Pooled Samples）</strong><ul><li><strong>原理</strong>：将多个个体DNA混合后统一分型，通过检测等位基因频率差异推断关联。</li><li><strong>优势</strong>：成本降低可达30倍（如混合1,000样本仅需1次分型）。</li><li><strong>挑战</strong>：<ul><li>需高精度检测微小频率差异（如OR&#x3D;1.25对应等位频率变化&lt;1%）。</li><li>隐性效应（Recessive Model）难以识别（基因型频率差异无法通过混合频率反映）。</li><li>多表型研究需多次混合，成本优势减弱。</li><li>无法分析基因-基因或基因-环境交互作用。</li></ul></li></ul></li></ol><h4 id="Avoiding-false-positive-associations"><a href="#Avoiding-false-positive-associations" class="headerlink" title="Avoiding false-positive associations"></a>Avoiding false-positive associations</h4><p>Sources of false-positive associations can be divided into three main categories: 1）statistical fluctuations. 2) systematic biases. 3) technical artifacts</p><ul><li><p><strong>群体分层偏倚及其应对策略</strong><br>群体分层（Population Stratification）指研究人群中存在疾病患病率（或表型均值）不同的亚群体，若这些亚群体间遗传标记频率也存在差异，可能导致假阳性关联。例如：</p><ul><li><strong>假阳性场景</strong>：某SNP在欧洲裔中频率高，非裔中频率低，而病例组因抽样偏差欧洲裔比例过高，则该SNP可能被错误关联至疾病。</li><li><strong>检测与校正方法</strong><ol><li><strong>未连锁标记法</strong>：<ul><li>使用数十个无关联的遗传标记评估群体结构（如祖先信息标记，AIMs）。</li><li>若检测到分层，可通过统计模型（如主成分分析，PCA）校正p值或重新匹配病例-对照。</li></ul></li><li><strong>GWAS自身优势</strong>：<ul><li>全基因组数十万SNP可精确量化分层程度，通过基因组控制（Genomic Control）调整检验统计量。</li></ul></li><li><strong>家庭样本设计</strong>：<ul><li>如传递不平衡检验（TDT），利用父母-子代三角设计，避免分层偏倚。<ul><li>仅分析患病子代与未传递父母等位基因的传递频率。在无关联时，父母等位传递概率为50%；若显著偏离50%，则提示关联。</li><li><strong>核心逻辑</strong>：家庭内部等位传递独立于群体结构，因父母基因型已代表子代遗传背景。</li></ul></li><li>优点：天然抗分层，可研究亲源效应（如印记基因）。</li><li>缺点：统计效力低于病例对照研究，需父母样本，对晚发疾病存在年龄偏倚。</li></ul></li></ol></li></ul></li><li><p><strong>技术性偏差对GWAS的影响及应对策略</strong></p><ul><li><p><strong>问题根源</strong>：</p><ul><li><strong>批次效应</strong>：若病例和对照分型实验条件不一致（如不同实验批次、试剂、操作人员），可能导致系统性误差。<ul><li><strong>示例</strong>：某SNP在病例组中使用实验板A分型，对照组使用实验板B，若板A分型错误率高，可能产生虚假关联。</li></ul></li><li><strong>均等影响假设</strong>：分型错误或数据缺失若均等影响病例和对照，则不会导致偏倚。但若影响不均衡（如病例分型质量差于对照），则可能引入假阳性。</li><li><strong>分型错误与缺失数据</strong>：<ul><li><strong>杂合子分型失败</strong>：某些技术（如特定芯片）对杂合子（如AG）的分型准确性较低，导致数据缺失。</li><li><strong>示例</strong>：若某SNP的杂合子分型失败率为10%，而纯合子（AA&#x2F;GG）失败率为1%，则杂合子数据缺失更多。在TDT中，若父母传递杂合等位基因时数据缺失，可能导致传递频率计算偏差（如实际传递60%却记录为100%）。</li></ul></li><li><strong>哈迪-温伯格平衡（HWE）的局限性</strong>：<ul><li>若等位基因罕见（如频率&lt;5%），即使杂合子分型失败，HWE检验可能不显著偏离，掩盖问题。</li></ul></li></ul></li></ul></li></ul><h3 id="Analysing-gene–gene-interactions"><a href="#Analysing-gene–gene-interactions" class="headerlink" title="Analysing gene–gene interactions"></a>Analysing gene–gene interactions</h3><p>基因-基因相互作用（<strong>上位效应</strong>）被认为可能解释部分遗传力，<strong>但是</strong>，统计学上发现的互作关系很可能几乎<strong>不可能</strong>被确立为生物学上具有重要意义的现象。 – Cordell, H. J. Epistasis: what it means, what it doesn’t mean, and statistical methods to detect it in humans. Hum. Mol. Genet. 11, 2463–2468 (2002).</p><p>本篇作者都观点：实验数据方面，上位效应的证据来自多次观察到的<strong>遗传背景对转基因、小鼠基因敲除和自发突变效应的修饰作用</strong>。然而，在某些情况下，许多主要是<strong>加性和独立的因素</strong>可能共同决定了常见表型的遗传力。这意味着，在<strong>全基因组关联研究中，初步的主效应筛选在许多情况下可能已经足够成功，而无需在初始分析时考虑上位效应</strong>。</p><h3 id="Summary"><a href="#Summary" class="headerlink" title="Summary"></a>Summary</h3><ul><li>最全面的研究方法将是对<strong>大量病例-对照人群进行完整基因组重测序（whole-genome resequencing）</strong>。这一方法不受候选基因选择的限制，能够覆盖<strong>编码区和非编码区的所有变异</strong>，并且不同于 GWAS，它能够同时检测**稀有变异（rare variants）和常见变异（common variants）**在疾病中的作用。</li><li>在不久的将来，我们将能够<strong>高效且全面地检测基因组中的常见遗传变异</strong>在常见疾病和复杂性状中的作用。鉴于<strong>候选基因关联研究</strong>已取得初步成功，尽管它仅覆盖了基因组的极小部分，<strong>更全面的全基因组关联研究</strong>将极大推动我们对常见疾病和复杂性状遗传基础的理解。</li></ul></div><div class="post-copyright-info w-full my-8 px-2 sm:px-6 md:px-8"><div class="article-copyright-info-container"><ul><li><strong>Title:</strong> Review of GWAS</li><li><strong>Author:</strong> Yuchen</li><li><strong>Created at :</strong> 2025-03-16 14:37:00</li><li><strong>Updated at :</strong> 2025-04-08 20:24:43</li><li><strong>Link:</strong> https://yuchen-starfall.github.io/2025/03/16/GWAS review/</li><li><strong>License: </strong>This work is licensed under <a class="license" target="_blank" rel="noopener" href="https://creativecommons.org/licenses/by-nc-sa/4.0">CC BY-NC-SA 4.0</a>.</li></ul></div></div><ul class="post-tags-box text-lg mt-1.5 flex-wrap justify-center flex md:hidden"><li class="tag-item mx-0.5"><a href="/tags/Population-Genetic-Structure/">#Population Genetic Structure</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/Article/">#Article</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/GWAS/">#GWAS</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/linkage-disequilibrium%EF%BC%88LD%EF%BC%89/">#linkage disequilibrium（LD）</a>&nbsp;</li></ul><div class="article-nav my-8 flex justify-between items-center px-2 sm:px-6 md:px-8"><div class="article-prev border-border-color shadow-redefine-flat shadow-shadow-color-2 rounded-medium px-4 py-2 hover:shadow-redefine-flat-hover hover:shadow-shadow-color-2"><a class="prev" rel="prev" href="/2025/03/18/Deep-learning%20--%20Model%20initialization/"><span class="left arrow-icon flex justify-center items-center"><i class="fa-solid fa-chevron-left"></i> </span><span class="title flex justify-center items-center"><span class="post-nav-title-item">Deep-learn -- Model initialization</span> <span class="post-nav-item">Prev posts</span></span></a></div><div class="article-next border-border-color shadow-redefine-flat shadow-shadow-color-2 rounded-medium px-4 py-2 hover:shadow-redefine-flat-hover hover:shadow-shadow-color-2"><a class="next" rel="next" href="/2025/03/16/Foundation%20of%20R/"><span class="title flex justify-center items-center"><span class="post-nav-title-item">Foundation of R</span> <span class="post-nav-item">Next posts</span> </span><span class="right arrow-icon flex justify-center items-center"><i class="fa-solid fa-chevron-right"></i></span></a></div></div><div class="comment-container px-2 sm:px-6 md:px-8 pb-8"><div class="comments-container mt-10 w-full"><div id="comment-anchor" class="w-full h-2.5"></div><div class="comment-area-title w-full my-1.5 md:my-2.5 text-xl md:text-3xl font-bold">Comments</div><div id="giscus-container"></div><script data-swup-reload-script defer>async function loadGiscus(){const t={src:"https://giscus.app/client.js","data-repo":"Yuchen-starfall/Yuchen-starfall.github.io","data-repo-id":"R_kgDONY6r-Q","data-category":"Announcements# Github discussion category","data-category-id":"DIC_kwDONY6r-c4Ck8Zm","data-mapping":"pathname","data-strict":"0","data-reactions-enabled":"1","data-emit-metadata":"1","data-theme":"preferred_color_scheme","data-lang":"en","data-input-position":"bottom","data-loading":"lazy",crossorigin:"anonymous",async:!0},a=document.createElement("script");for(const e in t)a.setAttribute(e,t[e]);document.getElementById("giscus-container").appendChild(a)}{let t=setTimeout(()=>{loadGiscus(),clearTimeout(t)},1e3)}</script></div></div></div><div class="toc-content-container"><div class="post-toc-wrap"><div class="post-toc"><div class="toc-title">On this page</div><div class="page-title">Review of GWAS</div><ol class="nav"><li class="nav-item nav-level-3"><a class="nav-link" href="#Why-genome-wide-association-studies"><span class="nav-text">Why genome-wide association studies?</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Linkage-analysis"><span class="nav-text">Linkage analysis</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Candidate-gene-resequencing-studies"><span class="nav-text">Candidate-gene resequencing studies</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Association-studies"><span class="nav-text">Association studies</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Strategies-to-increase-efficiency"><span class="nav-text">Strategies to increase efficiency</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Selecting-an-appropriate-sample-size"><span class="nav-text">Selecting an appropriate sample size</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#A-multi-stage-approach"><span class="nav-text">A multi-stage approach</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Founder-populations-and-pooled-samples"><span class="nav-text">Founder populations and pooled samples</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Avoiding-false-positive-associations"><span class="nav-text">Avoiding false-positive associations</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Analysing-gene%E2%80%93gene-interactions"><span class="nav-text">Analysing gene–gene interactions</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Summary"><span class="nav-text">Summary</span></a></li></ol></div></div></div></div></div></div><div class="main-content-footer"><footer class="footer mt-5 py-5 h-auto text-base text-third-text-color relative border-t-2 border-t-border-color"><div class="info-container py-3 text-center"><div class="text-center">&copy; <span>2024</span> - 2025&nbsp;&nbsp;<i class="fa-solid fa-heart fa-beat" style="--fa-animation-duration:0.5s;color:#f54545"></i>&nbsp;&nbsp;<a href="/">Yuchen</a><p class="post-count space-x-0.5"><span>51 posts in total</span></p></div><script data-swup-reload-script src="https://cn.vercount.one/js"></script><div class="relative text-center lg:absolute lg:right-[20px] lg:top-1/2 lg:-translate-y-1/2 lg:text-right"><span id="busuanzi_container_site_uv" class="lg:!block"><span class="text-sm">VISITOR COUNT</span> <span id="busuanzi_value_site_uv"></span> </span><span id="busuanzi_container_site_pv" class="lg:!block"><span class="text-sm">TOTAL PAGE VIEWS</span> <span id="busuanzi_value_site_pv"></span></span></div><div class="relative text-center lg:absolute lg:left-[20px] lg:top-1/2 lg:-translate-y-1/2 lg:text-left"><span class="lg:block text-sm">POWERED BY <?xml version="1.0" encoding="utf-8"?><!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"><svg class="relative top-[2px] inline-block align-baseline" version="1.1" id="圖層_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="1rem" height="1rem" viewBox="0 0 512 512" enable-background="new 0 0 512 512" xml:space="preserve"><path fill="#0E83CD" d="M256.4,25.8l-200,115.5L56,371.5l199.6,114.7l200-115.5l0.4-230.2L256.4,25.8z M349,354.6l-18.4,10.7l-18.6-11V275H200v79.6l-18.4,10.7l-18.6-11v-197l18.5-10.6l18.5,10.8V237h112v-79.6l18.5-10.6l18.5,10.8V354.6z"/></svg><a target="_blank" class="text-base" href="https://hexo.io">Hexo</a></span> <span class="text-sm lg:block">THEME&nbsp;<a class="text-base" target="_blank" href="https://github.com/EvanNotFound/hexo-theme-redefine">Redefine v2.8.2</a></span></div><div>Blog up for <span class="odometer" id="runtime_days"></span> days <span class="odometer" id="runtime_hours"></span> hrs <span class="odometer" id="runtime_minutes"></span> Min <span class="odometer" id="runtime_seconds"></span> Sec</div><script data-swup-reload-script>try{function odometer_init(){document.querySelectorAll(".odometer").forEach(e=>{new Odometer({el:e,format:"( ddd).dd",duration:200})})}odometer_init()}catch(e){}</script></div></footer></div></div><div class="post-tools"><div class="post-tools-container"><ul class="article-tools-list"><li class="right-bottom-tools page-aside-toggle"><i class="fa-regular fa-outdent"></i></li><li class="go-comment"><i class="fa-regular fa-comments"></i></li></ul></div></div><div class="right-side-tools-container"><div class="side-tools-container"><ul class="hidden-tools-list"><li class="right-bottom-tools tool-font-adjust-plus flex justify-center items-center"><i class="fa-regular fa-magnifying-glass-plus"></i></li><li class="right-bottom-tools tool-font-adjust-minus flex justify-center items-center"><i class="fa-regular fa-magnifying-glass-minus"></i></li><li class="right-bottom-tools tool-dark-light-toggle flex justify-center items-center"><i class="fa-regular fa-moon"></i></li><li class="right-bottom-tools tool-scroll-to-top flex justify-center items-center"><i class="fa-regular fa-arrow-up"></i></li><li class="right-bottom-tools tool-scroll-to-bottom flex justify-center items-center"><i class="fa-regular fa-arrow-down"></i></li></ul><ul class="visible-tools-list"><li class="right-bottom-tools toggle-tools-list flex justify-center items-center"><i class="fa-regular fa-cog fa-spin"></i></li></ul></div></div><div class="image-viewer-container"><img src=""></div></main><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/Swup.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupSlideTheme.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupScriptsPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupProgressPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupScrollPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupPreloadPlugin.min.js"></script><script>const swup=new Swup({plugins:[new SwupScriptsPlugin({optin:!0}),new SwupProgressPlugin,new SwupScrollPlugin({offset:80}),new SwupSlideTheme({mainElement:".main-content-body"}),new SwupPreloadPlugin],containers:["#swup"]})</script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/imageViewer.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/utils.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/main.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/navbarShrink.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/scrollTopBottom.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/lightDarkSwitch.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/categoryList.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/codeBlock.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/lazyload.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/runtime.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/odometer.min.js"></script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/assets/odometer-theme-minimal.css"><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/Typed.min.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/typed.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/minimasonry.min.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/masonry.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/tocToggle.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/toc.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/tabs.js" data-swup-reload-script></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/moment-with-locales.min.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/essays.js" data-swup-reload-script></script></body></html>