<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><meta name="keywords" content="Hexo Theme Redefine"><meta name="author" content="Yuchen"><link rel="preconnect" href="https://fonts.googleapis.com"><link rel="preconnect" href="https://fonts.gstatic.com" crossorigin><link rel="preconnect" href="https://cdn.jsdelivr.net" crossorigin><link rel="canonical" href="http://example.com/2025/03/25/haplotype estimation and linkage disequilibrium analysis/"><meta name="robots" content="index,follow"><meta name="googlebot" content="index,follow"><meta name="revisit-after" content="1 days"><meta name="description" content="Hexo Theme Redefine, Redefine Your Hexo Journey."><meta property="og:type" content="article"><meta property="og:title" content="Haplotype estimation and linkage disequilibrium(LD) analysis"><meta property="og:url" content="http://example.com/2025/03/25/Haplotype%20estimation%20and%20linkage%20disequilibrium%20analysis/index.html"><meta property="og:site_name" content="Hexo"><meta property="og:description" content="Hexo Theme Redefine, Redefine Your Hexo Journey."><meta property="og:locale" content="en_US"><meta property="og:image" content="http://example.com/images/redefine-og.webp"><meta property="article:published_time" content="2025-03-25T12:36:00.000Z"><meta property="article:modified_time" content="2025-04-08T15:34:24.862Z"><meta property="article:author" content="John Doe"><meta property="article:tag" content="course"><meta property="article:tag" content="Biomedical Statistics"><meta property="article:tag" content="GWAS"><meta property="article:tag" content="linkage disequilibrium（LD）"><meta property="article:tag" content="Haplotype"><meta name="twitter:card" content="summary"><meta name="twitter:image" content="http://example.com/images/redefine-og.webp"><link rel="icon" type="image/png" href="/images/profile.jpg" sizes="192x192"><link rel="apple-touch-icon" sizes="180x180" href="/images/profile.jpg"><meta name="theme-color" content="#A31F34"><link rel="shortcut icon" href="/images/profile.jpg"><title>Haplotype estimation and linkage disequilibrium(LD) analysis | Steps Among the Stars</title><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/Chillax/chillax.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/css/build/tailwind.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/GeistMono/geist-mono.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/Geist/geist.css"><script id="hexo-configurations">window.config={hostname:"example.com",root:"/",language:"en"},window.theme={articles:{style:{font_size:"16px",line_height:1.5,image_border_radius:"14px",image_alignment:"center",image_caption:!1,link_icon:!0,delete_mask:!1,title_alignment:"left",headings_top_spacing:{h1:"3.2rem",h2:"2.4rem",h3:"1.9rem",h4:"1.6rem",h5:"1.4rem",h6:"1.3rem"}},word_count:{enable:!0,count:!0,min2read:!0},author_label:{enable:!1,auto:!1,list:[]},code_block:{copy:!0,style:"mac",highlight_theme:{light:"github",dark:"vs2015"},font:{enable:!1,family:null,url:null}},toc:{enable:!0,max_depth:6,number:!1,expand:!0,init_open:!0},copyright:{enable:!0,default:"CC BY-NC-ND"},lazyload:!0,pangu_js:!1,recommendation:{enable:!1,title:"推荐阅读",limit:3,mobile_limit:2,placeholder:"/images/wallhaven-wqery6-light.webp",skip_dirs:[]}},colors:{primary:"#A31F34",secondary:null,default_mode:"light"},global:{fonts:{chinese:{enable:!1,family:null,url:null},english:{enable:!1,family:null,url:null},title:{enable:!1,family:null,url:null}},content_max_width:"1000px",sidebar_width:"210px",hover:{shadow:!0,scale:!1},scroll_progress:{bar:!1,percentage:!1},website_counter:{url:"https://cn.vercount.one/js",enable:!0,site_pv:!0,site_uv:!0,post_pv:!0},single_page:!0,preloader:{enable:"ture",custom_message:"Welcome to my little corner of chaos and brilliance!"},open_graph:{enable:!0,image:"/images/redefine-og.webp",description:"Hexo Theme Redefine, Redefine Your Hexo Journey."},google_analytics:{enable:!1,id:null}},home_banner:{enable:!0,style:"fixed",image:{light:"/images/light.jpeg",dark:"/images/dark.png"},title:"Resonant Journals",subtitle:{text:["Welcome to my little corner of chaos and brilliance!"],hitokoto:{enable:!1,show_author:!1,api:"https://v1.hitokoto.cn"},typing_speed:20,backing_speed:80,starting_delay:500,backing_delay:1500,loop:!1,smart_backspace:!0},text_color:{light:"#fff",dark:"#d1d1b6"},text_style:{title_size:"2.8rem",subtitle_size:"1.5rem",line_height:1.2},custom_font:{enable:!1,family:null,url:null},social_links:{enable:"ture",style:"default",links:{github:null,instagram:null,zhihu:null,twitter:null,email:"yuchen.starfall@outlook.com"},qrs:{weixin:null}}},plugins:{feed:{enable:!1},aplayer:{enable:!1,type:"fixed",audios:[{name:null,artist:null,url:null,cover:null,lrc:null}]},mermaid:{enable:!1,version:"11.4.1"}},version:"2.8.2",navbar:{auto_hide:"ture",color:{left:"#f78736",right:"#367df7",transparency:35},width:{home:"1200px",pages:"1000px"},links:{Home:{path:"/",icon:"fa-regular fa-house"},Archives:{path:"/archives",icon:"fa-regular fa-archive"},Categories:{icon:"fa-solid fa-folder",path:"/categories/"},Tags:{icon:"fa-solid fa-tags",path:"/tags/"}},search:{enable:!1,preload:!0}},page_templates:{friends_column:2,tags_style:"blur"},home:{sidebar:{enable:!0,position:"left",first_item:"menu",announcement:"Ephemeral Trails",show_on_mobile:!0,links:{Archives:{path:"/archives",icon:"fa-regular fa-archive"},Tags:{path:"/tags",icon:"fa-regular fa-tags"},Categories:{path:"/categories",icon:"fa-regular fa-folder"}}},article_date_format:"auto",excerpt_length:200,categories:{enable:!0,limit:3},tags:{enable:!0,limit:3}},footerStart:"2024/12/5 00:00:00"},window.lang_ago={second:"%s seconds ago",minute:"%s minutes ago",hour:"%s hours ago",day:"%s days ago",week:"%s weeks ago",month:"%s months ago",year:"%s years ago"},window.data={masonry:!0}</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/fontawesome.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/brands.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/solid.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/regular.min.css"><meta name="generator" content="Hexo 7.3.0"></head><body><div class="progress-bar-container"><span class="pjax-progress-bar"></span></div><main class="page-container" id="swup"><div class="main-content-container flex flex-col justify-between min-h-dvh"><div class="main-content-header"><header class="navbar-container px-6 md:px-12"><div class="navbar-content transition-navbar"><div class="left"><a class="logo-title" href="/">Steps Among the Stars</a></div><div class="right"><div class="desktop"><ul class="navbar-list"><li class="navbar-item"><a href="/"><i class="fa-regular fa-house fa-fw"></i> HOME</a></li><li class="navbar-item"><a href="/archives"><i class="fa-regular fa-archive fa-fw"></i> ARCHIVES</a></li><li class="navbar-item"><a href="/categories/"><i class="fa-solid fa-folder fa-fw"></i> CATEGORIES</a></li><li class="navbar-item"><a href="/tags/"><i class="fa-solid fa-tags fa-fw"></i> TAGS</a></li></ul></div><div class="mobile"><div class="icon-item navbar-bar"><div class="navbar-bar-middle"></div></div></div></div></div><div class="navbar-drawer h-dvh w-full absolute top-0 left-0 bg-background-color flex flex-col justify-between"><ul class="drawer-navbar-list flex flex-col px-4 justify-center items-start"><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/"><span>HOME </span><i class="fa-regular fa-house fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/archives"><span>ARCHIVES </span><i class="fa-regular fa-archive fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/categories/"><span>CATEGORIES </span><i class="fa-solid fa-folder fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/tags/"><span>TAGS </span><i class="fa-solid fa-tags fa-sm fa-fw"></i></a></li></ul><div class="statistics flex justify-around my-2.5"><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/tags"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">48</div><div class="label text-third-text-color text-sm">Tags</div></a><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/categories"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">7</div><div class="label text-third-text-color text-sm">Categories</div></a><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/archives"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">52</div><div class="label text-third-text-color text-sm">Posts</div></a></div></div><div class="window-mask"></div></header></div><div class="main-content-body transition-fade-up"><div class="main-content"><div class="post-page-container flex relative justify-between box-border w-full h-full"><div class="article-content-container"><div class="article-title relative w-full"><div class="w-full flex items-center pt-6 justify-start"><h1 class="article-title-regular text-second-text-color tracking-tight text-4xl md:text-6xl font-semibold px-2 sm:px-6 md:px-8 py-3">Haplotype estimation and linkage disequilibrium(LD) analysis</h1></div></div><div class="article-header flex flex-row gap-2 items-center px-2 sm:px-6 md:px-8"><div class="avatar w-[46px] h-[46px] flex-shrink-0 rounded-medium border border-border-color p-[1px]"><img src="/images/profile.jpg"></div><div class="info flex flex-col justify-between"><div class="author flex items-center"><span class="name text-default-text-color text-lg font-semibold">Yuchen</span></div><div class="meta-info"><div class="article-meta-info"><span class="article-date article-meta-item"><i class="fa-regular fa-pen-fancy"></i>&nbsp; <span class="desktop">2025-03-25 20:36</span> <span class="mobile">2025-03-25 20:36</span> <span class="hover-info">Created</span> </span><span class="article-date article-meta-item"><i class="fa-regular fa-wrench"></i>&nbsp; <span class="desktop">2025-04-08 23:34:24</span> <span class="mobile">2025-04-08 23:34:24</span> <span class="hover-info">Updated</span> </span><span class="article-categories article-meta-item"><i class="fa-regular fa-folders"></i>&nbsp;<ul><li><a href="/categories/Heredity-and-evolution/">Heredity and evolution</a>&nbsp;</li></ul></span><span class="article-tags article-meta-item"><i class="fa-regular fa-tags"></i>&nbsp;<ul><li><a href="/tags/course/">course</a>&nbsp;</li><li>| <a href="/tags/Biomedical-Statistics/">Biomedical Statistics</a>&nbsp;</li><li>| <a href="/tags/GWAS/">GWAS</a>&nbsp;</li><li>| <a href="/tags/linkage-disequilibrium%EF%BC%88LD%EF%BC%89/">linkage disequilibrium（LD）</a>&nbsp;</li><li>| <a href="/tags/Haplotype/">Haplotype</a>&nbsp;</li></ul></span><span class="article-pv article-meta-item"><i class="fa-regular fa-eye"></i>&nbsp;<span id="busuanzi_value_page_pv"></span></span></div></div></div></div><div class="article-content markdown-body px-2 sm:px-6 md:px-8 pb-8"><p><strong>Haplotype</strong>：a unique combination of genetic markers present in a chromosome.（相邻位点的等位基因在同一条染色体上的排列方式）</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-LD/1.png"></p><h3 id="Question：如何从杂合子二倍体测序数据中分离出哪些reads是来自于一条染色体"><a href="#Question：如何从杂合子二倍体测序数据中分离出哪些reads是来自于一条染色体" class="headerlink" title="Question：如何从杂合子二倍体测序数据中分离出哪些reads是来自于一条染色体"></a>Question：如何从杂合子二倍体测序数据中分离出哪些reads是来自于一条染色体</h3><h4 id="单体型分型（Haplotype-Phasing）"><a href="#单体型分型（Haplotype-Phasing）" class="headerlink" title="单体型分型（Haplotype Phasing）"></a>单体型分型（Haplotype Phasing）</h4><ul><li>基于三代长读长（识别杂合位点形成的“气泡结构”（同一位置存在两条路径））</li><li>基于群体或家系的分型</li><li>Hi-C或链特异性测序</li><li>体细胞杂交（somatic cell hybrids）从二倍体（diploid）过渡到单倍体（haploid），是一种经典的实验手段，可用于分离同源染色体并识别单倍型（haplotype）<ul><li>杂种细胞逐渐丢失人类染色体，最终可能仅保留<strong>单条人类染色体</strong>（父源或母源的某一同源染色体）。</li></ul></li><li>Statistical inference<ul><li>Clark Algorithm<ul><li>基本逻辑：1. 若某个体的基因型无法用现有单体型库解释，则推断新的单体型。2. 优先选择最少的单体型组合解释所有样本的基因型数据。</li><li>识别纯合子</li><li>处理模棱两可</li></ul></li><li>EM (Arlequin)<ul><li>通过迭代优化隐变量（单体型）的概率分布，最大化数据的似然函数。</li></ul></li><li>PHASE<ul><li><strong>核心思想</strong>：结合<strong>贝叶斯统计</strong>与<strong>EM算法</strong>，通过马尔可夫链蒙特卡洛（MCMC）采样提高分型精度。</li><li>引入<strong>重组模型</strong>：允许单体型间存在历史重组事件，更贴近真实遗传过程。</li><li><strong>近似贪心算法</strong>：加速计算，适用于大规模数据。</li></ul></li></ul></li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-LD/2.png"></p><p>因此，BUSCO可以用于检测基因组的组装准确性：检查单条单体型是否包含<strong>完整</strong>保守基因</p><h3 id="Theory"><a href="#Theory" class="headerlink" title="Theory"></a>Theory</h3><p>Infinite Sites Assumption and Perfect Phylogeny</p><ul><li><strong>ISA是完美系统发育的基础</strong>：<br>若ISA成立（每个突变唯一且无回复），则突变模式天然满足完美系统发育的条件，树中无冲突分支。</li><li><strong>违反ISA → 破坏完美系统发育</strong>：<br>若发生回复突变或平行突变（同一位点多次突变），系统发育树中将出现冲突，无法构建完美树。</li></ul><h4 id="无限位点假设（Infinite-Sites-Assumption-ISA）"><a href="#无限位点假设（Infinite-Sites-Assumption-ISA）" class="headerlink" title="无限位点假设（Infinite Sites Assumption, ISA）"></a>无限位点假设（Infinite Sites Assumption, ISA）</h4><p>假设基因组中每个新的突变（如SNP）都发生在<strong>从未发生过突变</strong>的位点，即：</p><ul><li>每个突变是唯一的，不会重复发生在同一位置；</li><li>不存在<strong>回复突变</strong>（如A→T→A）；</li><li>每个突变在群体中仅出现一次（即单一起源）。</li></ul><p><strong>局限性</strong></p><ul><li><strong>现实不符</strong>：实际基因组位点有限，且存在热点突变（如CpG岛）、回复突变或平行突变。</li><li><strong>重组影响</strong>：重组事件会导致突变位点的组合复杂性，破坏ISA的假设</li></ul><h4 id="完美系统发育（Perfect-Phylogeny）"><a href="#完美系统发育（Perfect-Phylogeny）" class="headerlink" title="完美系统发育（Perfect Phylogeny）"></a>完美系统发育（Perfect Phylogeny）</h4><p>一种理想化的系统发育树结构，满足：</p><ul><li><strong>无冲突突变</strong>：每个突变（特征）在树中仅出现一次，且一旦出现即被所有后代继承（不会丢失或逆转）；</li><li><strong>二元性状兼容性</strong>：所有突变位点的状态变化（如0→1）在树中无矛盾（即符合“四配子法则（<strong>Four-Gamete Test</strong>）”）。<ul><li>如果在群体中观察到所有四种配子类型（AB、Ab、aB、ab），则说明两个位点之间<strong>至少发生过一次重组</strong>或存在<strong>同源突变</strong>（如平行突变或回复突变）。</li><li>如果发现3种配子，说明有突变情况产生</li></ul></li></ul><p><strong>核心思想</strong></p><p>通过突变事件的唯一性和继承性，构建一棵能完美解释所有遗传变异的树，无需处理冲突的突变模式。</p><p><strong>局限性</strong></p><ul><li><strong>现实复杂性</strong>：真实数据常存在回复突变、趋同进化或重组，导致树结构冲突。</li><li><strong>依赖无限位点假设</strong>：完美系统发育成立的前提是ISA成立（突变唯一且无回复）</li></ul><h3 id="Linkage-Disequilibrium"><a href="#Linkage-Disequilibrium" class="headerlink" title="Linkage Disequilibrium"></a>Linkage Disequilibrium</h3><p>是相邻位点之间的非随机关联，当一个位点上的某一等位基因与另一位点上的等位基因共同出现的概率大于随机组合的假设（<strong>底层逻辑是同源重组</strong>），则这两个位点之间存在连锁不平衡。</p><div class="note-large default"><div class="notel-title rounded-t-lg p-3 font-bold text-lg flex flex-row gap-2 items-center"><i class="notel-icon fa-solid fa-info"></i><p>LD</p></div><div class="notel-content"><p>HWE: random combination of allelic genes (<strong>same loci</strong>)</p><p>LE: random combination of alleles from <strong>different loci</strong></p></div></div><p>假设两位点 <strong>A</strong>（等位基因A&#x2F;a）和 <strong>B</strong>（等位基因B&#x2F;b），其连锁不平衡程度可通过 <strong>D值（Lewontin’s D）</strong> 衡量：</p><p>$$D&#x3D;P(AB)−P(A)⋅P(B)$$</p><ul><li><p>P(AB)：等位基因A和B同时出现的频率；</p></li><li><p>P(A)<em>P</em>(<em>A</em>) 和 P(B)<em>P</em>(<em>B</em>)：等位基因A和B各自独立出现的频率。</p></li><li><p>D&#x3D;0：完全连锁平衡（等位基因独立）；</p></li><li><p>D≠0：存在连锁不平衡（正或负关联）。</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-LD/3.png"></p></li></ul><h4 id="标准化D值并消除等位基因频率的影响"><a href="#标准化D值并消除等位基因频率的影响" class="headerlink" title="标准化D值并消除等位基因频率的影响"></a>标准化D值并消除等位基因频率的影响</h4><p>为了解决不同位点对的D值无法直接比较，同时D<strong>值的正负仅表示等位基因关联方向</strong>（如A-B或A-b），但<strong>绝对值大小无法直观反映关联强度</strong><br>$$<br>D’ &#x3D; \frac{D}{D_{\text{max}}}, \quad D_{\text{max}} &#x3D;<br>\begin{cases}<br>\min\left[ P(A)P(b), P(a)P(B) \right] &amp; D &gt; 0 \<br>\min\left[ P(A)P(B), P(a)P(b) \right] &amp; D &lt; 0<br>\end{cases}<br>$$</p><p><strong>范围</strong>：D′∈[−1,1]，绝对值越接近1，连锁不平衡越强 。当其绝对值为1时，AB、Ab、aB，缺失ab或反之。</p><h4 id="相关系数平方（r²）"><a href="#相关系数平方（r²）" class="headerlink" title="相关系数平方（r²）"></a>相关系数平方（r²）</h4><p><img lazyload src="/images/loading.svg" data-src="/img/img-LD/5.png"></p><p><strong>范围</strong>：</p><ul><li>r2∈[0,1]，反映两位点间等位基因的关联强度。</li><li>r2&#x3D;1：完全连锁不平衡（仅两种单体型存在）</li><li>r2&#x3D;0：完全连锁平衡。</li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-LD/4.png"></p><p>在关联研究中，优先用r²筛选tag SNP；在进化分析中，结合D’推断重组事件。</p><h4 id="影响因素"><a href="#影响因素" class="headerlink" title="影响因素"></a>影响因素</h4><ul><li>分子水平<ul><li>重组率的不均匀分布</li><li>突变率的不均匀分布<ul><li>在CpG二核苷酸位置附近，单核苷酸多态位点可能具有较高的突变率，从而与附近的位点表现出很弱的连锁不平衡甚至没有连锁不平衡，尽管实际上没有重组发生。</li></ul></li><li><strong>基因转换</strong><ul><li>人类基因组中基因转换的发生非常频繁，对紧密连锁位点之间的连锁不平衡产生重要影响。</li><li>倒位（无法同源重组）</li><li>植物里转座子的出现？？？！</li></ul></li></ul></li><li>群体水平<ul><li>遗传漂变</li><li>群体扩张</li><li>人群混合和人群迁移</li><li>群体结构</li><li>自然选择<ul><li>搭乘效应（hitchhiking effect)：有利基因被自然选择时，附近物理位置紧邻的其他基因（即使无关或轻微有害）也会被“顺带传播”，因为它们和有利基因在遗传时像被胶水粘在一起，难以分开（重组率低）。</li><li>上位选择（epistatic selection）：某个基因（如基因A）的作用可能完全依赖另一个基因（如基因B）的存在。自然选择会偏好<strong>特定基因组合的整体效果</strong>，而非单个基因的优劣。</li></ul></li></ul></li></ul><h4 id="Visualizing-LD"><a href="#Visualizing-LD" class="headerlink" title="Visualizing LD"></a>Visualizing LD</h4><p>如果某个区域的重组率较低，而 LD 矩阵显示<strong>大片红色区域</strong>（blocks），说明这个区域可能是一个<strong>连锁</strong>单元，基因变异较少打乱这种连锁关系。</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-LD/6.png"></p><h3 id="GWAS"><a href="#GWAS" class="headerlink" title="GWAS"></a>GWAS</h3><p>底层逻辑是寻找marker SNP与潜在位点的相关的统计学显著位点。</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-LD/7.png"></p><div class="note-large default"><div class="notel-title rounded-t-lg p-3 font-bold text-lg flex flex-row gap-2 items-center"><i class="notel-icon fa-solid fa-info"></i><p>test</p></div><div class="notel-content"><p>每次检验对应一个假设：“该SNP与表型无关”，即使所有SNP均无真实效应，仅凭随机性也会导致大量假阳性(1-n)^n。因此对于假设接受的统计学p-value需要进行adjust。</p><ol><li>Bonferroni校正(简单严格，广泛用于GWAS,10^-b的p-adjust来源)</li><li>FDR（错误发现率）控制</li></ol></div></div><h4 id="less-cost-as-few-SNPs-as-possible-for-association-studies"><a href="#less-cost-as-few-SNPs-as-possible-for-association-studies" class="headerlink" title="less cost: as few SNPs as possible for association studies."></a>less cost: as few SNPs as possible for association studies.</h4><p>基因区域发现了重组点的不均匀分布 – 使用代表性的SNP代表一个连锁block</p><p>因此可以不必测所有SNP，仅测具有代表性的SNP›</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-LD/8.png"></p><div class="note-large default"><div class="notel-title rounded-t-lg p-3 font-bold text-lg flex flex-row gap-2 items-center"><i class="notel-icon fa-solid fa-info"></i><p>Note</p></div><div class="notel-content"><p>图中展示的LD区域blocks是常见的，暂不考虑rare mutant</p></div></div><p><img lazyload src="/images/loading.svg" data-src="/img/img-LD/9.png"></p><h5 id="Tag-SNP"><a href="#Tag-SNP" class="headerlink" title="Tag SNP"></a>Tag SNP</h5><p><img lazyload src="/images/loading.svg" data-src="/img/img-LD/10.png"></p><h5 id="How-to-select-TagSNP"><a href="#How-to-select-TagSNP" class="headerlink" title="How to select TagSNP"></a>How to select TagSNP</h5><p>Greedy approach（贪婪算法）: keep selecting untagged SNP that is in LD with the most remaining untagged SNPs. – Greedy may not be optimal.</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-LD/11.png"></p><p>Optimal selection</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-LD/12.png"></p><h4 id="HapMap"><a href="#HapMap" class="headerlink" title="HapMap"></a>HapMap</h4><p>In my opinion，在完成人类全基因组测序之后，对于基因组的了解达到一个阶段。在想要进行association analysis过程过，发现测序昂贵，人们想尽一切办法去减少cost，发现了连锁遗传特点后，想要找到连锁blocks、常见的SNP变异，用blocks是单SNP作为tagSNP，进行分析。因此HapMap计划横空出世，解析了人类基因组的LD pattern，推断人类基因组的Recombination rates。为之后Microarray测序技术提供了位点选择。随时代的发展，到现如今的全基因组测序。</p></div><div class="post-copyright-info w-full my-8 px-2 sm:px-6 md:px-8"><div class="article-copyright-info-container"><ul><li><strong>Title:</strong> Haplotype estimation and linkage disequilibrium(LD) analysis</li><li><strong>Author:</strong> Yuchen</li><li><strong>Created at :</strong> 2025-03-25 20:36:00</li><li><strong>Updated at :</strong> 2025-04-08 23:34:24</li><li><strong>Link:</strong> https://yuchen-starfall.github.io/2025/03/25/Haplotype estimation and linkage disequilibrium analysis/</li><li><strong>License: </strong>This work is licensed under <a class="license" target="_blank" rel="noopener" href="https://creativecommons.org/licenses/by-nc-sa/4.0">CC BY-NC-SA 4.0</a>.</li></ul></div></div><ul class="post-tags-box text-lg mt-1.5 flex-wrap justify-center flex md:hidden"><li class="tag-item mx-0.5"><a href="/tags/course/">#course</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/Biomedical-Statistics/">#Biomedical Statistics</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/GWAS/">#GWAS</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/linkage-disequilibrium%EF%BC%88LD%EF%BC%89/">#linkage disequilibrium（LD）</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/Haplotype/">#Haplotype</a>&nbsp;</li></ul><div class="article-nav my-8 flex justify-between items-center px-2 sm:px-6 md:px-8"><div class="article-prev border-border-color shadow-redefine-flat shadow-shadow-color-2 rounded-medium px-4 py-2 hover:shadow-redefine-flat-hover hover:shadow-shadow-color-2"><a class="prev" rel="prev" href="/2025/04/01/GWAS%20statistics%20methods/"><span class="left arrow-icon flex justify-center items-center"><i class="fa-solid fa-chevron-left"></i> </span><span class="title flex justify-center items-center"><span class="post-nav-title-item">Statistical methods for population association studies</span> <span class="post-nav-item">Prev posts</span></span></a></div><div class="article-next border-border-color shadow-redefine-flat shadow-shadow-color-2 rounded-medium px-4 py-2 hover:shadow-redefine-flat-hover hover:shadow-shadow-color-2"><a class="next" rel="next" href="/2025/03/18/Deep-learning%20--%20Model%20initialization/"><span class="title flex justify-center items-center"><span class="post-nav-title-item">Deep-learn -- Model initialization</span> <span class="post-nav-item">Next posts</span> </span><span class="right arrow-icon flex justify-center items-center"><i class="fa-solid fa-chevron-right"></i></span></a></div></div><div class="comment-container px-2 sm:px-6 md:px-8 pb-8"><div class="comments-container mt-10 w-full"><div id="comment-anchor" class="w-full h-2.5"></div><div class="comment-area-title w-full my-1.5 md:my-2.5 text-xl md:text-3xl font-bold">Comments</div><div id="giscus-container"></div><script data-swup-reload-script defer>async function loadGiscus(){const t={src:"https://giscus.app/client.js","data-repo":"Yuchen-starfall/Yuchen-starfall.github.io","data-repo-id":"R_kgDONY6r-Q","data-category":"Announcements# Github discussion category","data-category-id":"DIC_kwDONY6r-c4Ck8Zm","data-mapping":"pathname","data-strict":"0","data-reactions-enabled":"1","data-emit-metadata":"1","data-theme":"preferred_color_scheme","data-lang":"en","data-input-position":"bottom","data-loading":"lazy",crossorigin:"anonymous",async:!0},a=document.createElement("script");for(const e in t)a.setAttribute(e,t[e]);document.getElementById("giscus-container").appendChild(a)}{let t=setTimeout(()=>{loadGiscus(),clearTimeout(t)},1e3)}</script></div></div></div><div class="toc-content-container"><div class="post-toc-wrap"><div class="post-toc"><div class="toc-title">On this page</div><div class="page-title">Haplotype estimation and linkage disequilibrium(LD) analysis</div><ol class="nav"><li class="nav-item nav-level-3"><a class="nav-link" href="#Question%EF%BC%9A%E5%A6%82%E4%BD%95%E4%BB%8E%E6%9D%82%E5%90%88%E5%AD%90%E4%BA%8C%E5%80%8D%E4%BD%93%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE%E4%B8%AD%E5%88%86%E7%A6%BB%E5%87%BA%E5%93%AA%E4%BA%9Breads%E6%98%AF%E6%9D%A5%E8%87%AA%E4%BA%8E%E4%B8%80%E6%9D%A1%E6%9F%93%E8%89%B2%E4%BD%93"><span class="nav-text">Question：如何从杂合子二倍体测序数据中分离出哪些reads是来自于一条染色体</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#%E5%8D%95%E4%BD%93%E5%9E%8B%E5%88%86%E5%9E%8B%EF%BC%88Haplotype-Phasing%EF%BC%89"><span class="nav-text">单体型分型（Haplotype Phasing）</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Theory"><span class="nav-text">Theory</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#%E6%97%A0%E9%99%90%E4%BD%8D%E7%82%B9%E5%81%87%E8%AE%BE%EF%BC%88Infinite-Sites-Assumption-ISA%EF%BC%89"><span class="nav-text">无限位点假设（Infinite Sites Assumption, ISA）</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#%E5%AE%8C%E7%BE%8E%E7%B3%BB%E7%BB%9F%E5%8F%91%E8%82%B2%EF%BC%88Perfect-Phylogeny%EF%BC%89"><span class="nav-text">完美系统发育（Perfect Phylogeny）</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Linkage-Disequilibrium"><span class="nav-text">Linkage Disequilibrium</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#%E6%A0%87%E5%87%86%E5%8C%96D%E5%80%BC%E5%B9%B6%E6%B6%88%E9%99%A4%E7%AD%89%E4%BD%8D%E5%9F%BA%E5%9B%A0%E9%A2%91%E7%8E%87%E7%9A%84%E5%BD%B1%E5%93%8D"><span class="nav-text">标准化D值并消除等位基因频率的影响</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#%E7%9B%B8%E5%85%B3%E7%B3%BB%E6%95%B0%E5%B9%B3%E6%96%B9%EF%BC%88r%C2%B2%EF%BC%89"><span class="nav-text">相关系数平方（r²）</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#%E5%BD%B1%E5%93%8D%E5%9B%A0%E7%B4%A0"><span class="nav-text">影响因素</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Visualizing-LD"><span class="nav-text">Visualizing LD</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#GWAS"><span class="nav-text">GWAS</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#less-cost-as-few-SNPs-as-possible-for-association-studies"><span class="nav-text">less cost: as few SNPs as possible for association studies.</span></a><ol class="nav-child"><li class="nav-item nav-level-5"><a class="nav-link" href="#Tag-SNP"><span class="nav-text">Tag SNP</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#How-to-select-TagSNP"><span class="nav-text">How to select TagSNP</span></a></li></ol></li><li class="nav-item nav-level-4"><a class="nav-link" href="#HapMap"><span class="nav-text">HapMap</span></a></li></ol></li></ol></div></div></div></div></div></div><div class="main-content-footer"><footer class="footer mt-5 py-5 h-auto text-base text-third-text-color relative border-t-2 border-t-border-color"><div class="info-container py-3 text-center"><div class="text-center">&copy; <span>2024</span> - 2025&nbsp;&nbsp;<i class="fa-solid fa-heart fa-beat" style="--fa-animation-duration:0.5s;color:#f54545"></i>&nbsp;&nbsp;<a href="/">Yuchen</a><p class="post-count space-x-0.5"><span>52 posts in total</span></p></div><script data-swup-reload-script src="https://cn.vercount.one/js"></script><div class="relative text-center lg:absolute lg:right-[20px] lg:top-1/2 lg:-translate-y-1/2 lg:text-right"><span id="busuanzi_container_site_uv" class="lg:!block"><span class="text-sm">VISITOR COUNT</span> <span id="busuanzi_value_site_uv"></span> </span><span id="busuanzi_container_site_pv" class="lg:!block"><span class="text-sm">TOTAL PAGE VIEWS</span> <span id="busuanzi_value_site_pv"></span></span></div><div class="relative text-center lg:absolute lg:left-[20px] lg:top-1/2 lg:-translate-y-1/2 lg:text-left"><span class="lg:block text-sm">POWERED BY <?xml version="1.0" encoding="utf-8"?><!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"><svg class="relative top-[2px] inline-block align-baseline" version="1.1" id="圖層_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="1rem" height="1rem" viewBox="0 0 512 512" enable-background="new 0 0 512 512" xml:space="preserve"><path fill="#0E83CD" d="M256.4,25.8l-200,115.5L56,371.5l199.6,114.7l200-115.5l0.4-230.2L256.4,25.8z M349,354.6l-18.4,10.7l-18.6-11V275H200v79.6l-18.4,10.7l-18.6-11v-197l18.5-10.6l18.5,10.8V237h112v-79.6l18.5-10.6l18.5,10.8V354.6z"/></svg><a target="_blank" class="text-base" href="https://hexo.io">Hexo</a></span> <span class="text-sm lg:block">THEME&nbsp;<a class="text-base" target="_blank" href="https://github.com/EvanNotFound/hexo-theme-redefine">Redefine v2.8.2</a></span></div><div>Blog up for <span class="odometer" id="runtime_days"></span> days <span class="odometer" id="runtime_hours"></span> hrs <span class="odometer" id="runtime_minutes"></span> Min <span class="odometer" id="runtime_seconds"></span> Sec</div><script data-swup-reload-script>try{function odometer_init(){document.querySelectorAll(".odometer").forEach(e=>{new Odometer({el:e,format:"( ddd).dd",duration:200})})}odometer_init()}catch(e){}</script></div></footer></div></div><div class="post-tools"><div class="post-tools-container"><ul class="article-tools-list"><li class="right-bottom-tools page-aside-toggle"><i class="fa-regular fa-outdent"></i></li><li class="go-comment"><i class="fa-regular fa-comments"></i></li></ul></div></div><div class="right-side-tools-container"><div class="side-tools-container"><ul class="hidden-tools-list"><li class="right-bottom-tools tool-font-adjust-plus flex justify-center items-center"><i class="fa-regular fa-magnifying-glass-plus"></i></li><li class="right-bottom-tools tool-font-adjust-minus flex justify-center items-center"><i class="fa-regular fa-magnifying-glass-minus"></i></li><li class="right-bottom-tools tool-dark-light-toggle flex justify-center items-center"><i class="fa-regular fa-moon"></i></li><li class="right-bottom-tools tool-scroll-to-top flex justify-center items-center"><i class="fa-regular fa-arrow-up"></i></li><li class="right-bottom-tools tool-scroll-to-bottom flex justify-center items-center"><i class="fa-regular fa-arrow-down"></i></li></ul><ul class="visible-tools-list"><li class="right-bottom-tools toggle-tools-list flex justify-center items-center"><i class="fa-regular fa-cog fa-spin"></i></li></ul></div></div><div class="image-viewer-container"><img src=""></div></main><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/Swup.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupSlideTheme.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupScriptsPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupProgressPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupScrollPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupPreloadPlugin.min.js"></script><script>const swup=new Swup({plugins:[new SwupScriptsPlugin({optin:!0}),new SwupProgressPlugin,new SwupScrollPlugin({offset:80}),new SwupSlideTheme({mainElement:".main-content-body"}),new SwupPreloadPlugin],containers:["#swup"]})</script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/imageViewer.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/utils.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/main.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/navbarShrink.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/scrollTopBottom.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/lightDarkSwitch.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/categoryList.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/codeBlock.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/lazyload.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/runtime.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/odometer.min.js"></script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/assets/odometer-theme-minimal.css"><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/Typed.min.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/typed.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/minimasonry.min.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/masonry.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/tocToggle.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/toc.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/tabs.js" data-swup-reload-script></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/moment-with-locales.min.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/essays.js" data-swup-reload-script></script></body></html>