<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><meta name="keywords" content="Hexo Theme Redefine"><meta name="author" content="Yuchen"><link rel="preconnect" href="https://fonts.googleapis.com"><link rel="preconnect" href="https://fonts.gstatic.com" crossorigin><link rel="preconnect" href="https://cdn.jsdelivr.net" crossorigin><link rel="canonical" href="http://example.com/2025/06/22/pan-genome of wild and cultivated soybeans/"><meta name="robots" content="index,follow"><meta name="googlebot" content="index,follow"><meta name="revisit-after" content="1 days"><meta name="description" content="Hexo Theme Redefine, Redefine Your Hexo Journey."><meta property="og:type" content="article"><meta property="og:title" content="Pan-Genome of Wild and Cultivated Soybeans"><meta property="og:url" content="http://example.com/2025/06/22/Pan-Genome%20of%20Wild%20and%20Cultivated%20Soybeans/index.html"><meta property="og:site_name" content="Hexo"><meta property="og:description" content="Hexo Theme Redefine, Redefine Your Hexo Journey."><meta property="og:locale" content="en_US"><meta property="og:image" content="http://example.com/images/redefine-og.webp"><meta property="article:published_time" content="2025-06-22T02:56:00.000Z"><meta property="article:modified_time" content="2025-06-24T06:59:06.328Z"><meta property="article:author" content="John Doe"><meta property="article:tag" content="Article"><meta property="article:tag" content="Pangenome"><meta name="twitter:card" content="summary"><meta name="twitter:image" content="http://example.com/images/redefine-og.webp"><link rel="icon" type="image/png" href="/images/profile.jpg" sizes="192x192"><link rel="apple-touch-icon" sizes="180x180" href="/images/profile.jpg"><meta name="theme-color" content="#A31F34"><link rel="shortcut icon" href="/images/profile.jpg"><title>Pan-Genome of Wild and Cultivated Soybeans | Steps Among the Stars</title><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/Chillax/chillax.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/css/build/tailwind.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/GeistMono/geist-mono.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/Geist/geist.css"><script id="hexo-configurations">window.config={hostname:"example.com",root:"/",language:"en"},window.theme={articles:{style:{font_size:"16px",line_height:1.5,image_border_radius:"14px",image_alignment:"center",image_caption:!1,link_icon:!0,delete_mask:!1,title_alignment:"left",headings_top_spacing:{h1:"3.2rem",h2:"2.4rem",h3:"1.9rem",h4:"1.6rem",h5:"1.4rem",h6:"1.3rem"}},word_count:{enable:!0,count:!0,min2read:!0},author_label:{enable:!1,auto:!1,list:[]},code_block:{copy:!0,style:"mac",highlight_theme:{light:"github",dark:"vs2015"},font:{enable:!1,family:null,url:null}},toc:{enable:!0,max_depth:6,number:!1,expand:!0,init_open:!0},copyright:{enable:!0,default:"CC BY-NC-ND"},lazyload:!0,pangu_js:!1,recommendation:{enable:!1,title:"推荐阅读",limit:3,mobile_limit:2,placeholder:"/images/wallhaven-wqery6-light.webp",skip_dirs:[]}},colors:{primary:"#A31F34",secondary:null,default_mode:"light"},global:{fonts:{chinese:{enable:!1,family:null,url:null},english:{enable:!1,family:null,url:null},title:{enable:!1,family:null,url:null}},content_max_width:"1000px",sidebar_width:"210px",hover:{shadow:!0,scale:!1},scroll_progress:{bar:!1,percentage:!1},website_counter:{url:"https://cn.vercount.one/js",enable:!0,site_pv:!0,site_uv:!0,post_pv:!0},single_page:!0,preloader:{enable:"ture",custom_message:"Welcome to my little corner of chaos and brilliance!"},open_graph:{enable:!0,image:"/images/redefine-og.webp",description:"Hexo Theme Redefine, Redefine Your Hexo Journey."},google_analytics:{enable:!1,id:null}},home_banner:{enable:!0,style:"fixed",image:{light:"/images/light.jpeg",dark:"/images/dark.png"},title:"Resonant Journals",subtitle:{text:["Welcome to my little corner of chaos and brilliance!"],hitokoto:{enable:!1,show_author:!1,api:"https://v1.hitokoto.cn"},typing_speed:20,backing_speed:80,starting_delay:500,backing_delay:1500,loop:!1,smart_backspace:!0},text_color:{light:"#fff",dark:"#d1d1b6"},text_style:{title_size:"2.8rem",subtitle_size:"1.5rem",line_height:1.2},custom_font:{enable:!1,family:null,url:null},social_links:{enable:"ture",style:"default",links:{github:null,instagram:null,zhihu:null,twitter:null,email:"yuchen.starfall@outlook.com"},qrs:{weixin:null}}},plugins:{feed:{enable:!1},aplayer:{enable:!1,type:"fixed",audios:[{name:null,artist:null,url:null,cover:null,lrc:null}]},mermaid:{enable:!1,version:"11.4.1"}},version:"2.8.2",navbar:{auto_hide:"ture",color:{left:"#f78736",right:"#367df7",transparency:35},width:{home:"1200px",pages:"1000px"},links:{Home:{path:"/",icon:"fa-regular fa-house"},Archives:{path:"/archives",icon:"fa-regular fa-archive"},Categories:{icon:"fa-solid fa-folder",path:"/categories/"},Tags:{icon:"fa-solid fa-tags",path:"/tags/"}},search:{enable:!1,preload:!0}},page_templates:{friends_column:2,tags_style:"blur"},home:{sidebar:{enable:!0,position:"left",first_item:"menu",announcement:"Ephemeral Trails",show_on_mobile:!0,links:{Archives:{path:"/archives",icon:"fa-regular fa-archive"},Tags:{path:"/tags",icon:"fa-regular fa-tags"},Categories:{path:"/categories",icon:"fa-regular fa-folder"}}},article_date_format:"auto",excerpt_length:200,categories:{enable:!0,limit:3},tags:{enable:!0,limit:3}},footerStart:"2024/12/5 00:00:00"},window.lang_ago={second:"%s seconds ago",minute:"%s minutes ago",hour:"%s hours ago",day:"%s days ago",week:"%s weeks ago",month:"%s months ago",year:"%s years ago"},window.data={masonry:!0}</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/fontawesome.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/brands.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/solid.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/regular.min.css"><meta name="generator" content="Hexo 7.3.0"></head><body><div class="progress-bar-container"><span class="pjax-progress-bar"></span></div><main class="page-container" id="swup"><div class="main-content-container flex flex-col justify-between min-h-dvh"><div class="main-content-header"><header class="navbar-container px-6 md:px-12"><div class="navbar-content transition-navbar"><div class="left"><a class="logo-title" href="/">Steps Among the Stars</a></div><div class="right"><div class="desktop"><ul class="navbar-list"><li class="navbar-item"><a href="/"><i class="fa-regular fa-house fa-fw"></i> HOME</a></li><li class="navbar-item"><a href="/archives"><i class="fa-regular fa-archive fa-fw"></i> ARCHIVES</a></li><li class="navbar-item"><a href="/categories/"><i class="fa-solid fa-folder fa-fw"></i> CATEGORIES</a></li><li class="navbar-item"><a href="/tags/"><i class="fa-solid fa-tags fa-fw"></i> TAGS</a></li></ul></div><div class="mobile"><div class="icon-item navbar-bar"><div class="navbar-bar-middle"></div></div></div></div></div><div class="navbar-drawer h-dvh w-full absolute top-0 left-0 bg-background-color flex flex-col justify-between"><ul class="drawer-navbar-list flex flex-col px-4 justify-center items-start"><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/"><span>HOME </span><i class="fa-regular fa-house fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/archives"><span>ARCHIVES </span><i class="fa-regular fa-archive fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/categories/"><span>CATEGORIES </span><i class="fa-solid fa-folder fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/tags/"><span>TAGS </span><i class="fa-solid fa-tags fa-sm fa-fw"></i></a></li></ul><div class="statistics flex justify-around my-2.5"><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/tags"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">48</div><div class="label text-third-text-color text-sm">Tags</div></a><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/categories"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">7</div><div class="label text-third-text-color text-sm">Categories</div></a><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/archives"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">53</div><div class="label text-third-text-color text-sm">Posts</div></a></div></div><div class="window-mask"></div></header></div><div class="main-content-body transition-fade-up"><div class="main-content"><div class="post-page-container flex relative justify-between box-border w-full h-full"><div class="article-content-container"><div class="article-title relative w-full"><div class="w-full flex items-center pt-6 justify-start"><h1 class="article-title-regular text-second-text-color tracking-tight text-4xl md:text-6xl font-semibold px-2 sm:px-6 md:px-8 py-3">Pan-Genome of Wild and Cultivated Soybeans</h1></div></div><div class="article-header flex flex-row gap-2 items-center px-2 sm:px-6 md:px-8"><div class="avatar w-[46px] h-[46px] flex-shrink-0 rounded-medium border border-border-color p-[1px]"><img src="/images/profile.jpg"></div><div class="info flex flex-col justify-between"><div class="author flex items-center"><span class="name text-default-text-color text-lg font-semibold">Yuchen</span></div><div class="meta-info"><div class="article-meta-info"><span class="article-date article-meta-item"><i class="fa-regular fa-pen-fancy"></i>&nbsp; <span class="desktop">2025-06-22 10:56</span> <span class="mobile">2025-06-22 10:56</span> <span class="hover-info">Created</span> </span><span class="article-date article-meta-item"><i class="fa-regular fa-wrench"></i>&nbsp; <span class="desktop">2025-06-24 14:59:06</span> <span class="mobile">2025-06-24 14:59:06</span> <span class="hover-info">Updated</span> </span><span class="article-categories article-meta-item"><i class="fa-regular fa-folders"></i>&nbsp;<ul><li><a href="/categories/Literature/">Literature</a>&nbsp;</li></ul></span><span class="article-tags article-meta-item"><i class="fa-regular fa-tags"></i>&nbsp;<ul><li><a href="/tags/Article/">Article</a>&nbsp;</li><li>| <a href="/tags/Pangenome/">Pangenome</a>&nbsp;</li></ul></span><span class="article-pv article-meta-item"><i class="fa-regular fa-eye"></i>&nbsp;<span id="busuanzi_value_page_pv"></span></span></div></div></div></div><div class="article-content markdown-body px-2 sm:px-6 md:px-8 pb-8"><h2 id="Background"><a href="#Background" class="headerlink" title="Background"></a>Background</h2><h4 id="大豆基因组概述"><a href="#大豆基因组概述" class="headerlink" title="大豆基因组概述"></a>大豆基因组概述</h4><ul><li>古多倍化事件：</li></ul><p>​ 全基因组复制是高等真核生物进化过程中普遍存在的现象，使真核生物的基因数量加倍，为新物种适应和进化提供了丰富的遗传物质。WGD事件可产生典型的多倍体，包括自多倍体和异源多倍体。由于功能冗余，重复基因往往在 WGD 事件后发生分化（丢失）。重复基因对可能会有几种不同的进化命运：一个拷贝失去功能（假基因化）；一个拷贝获得新功能（新功能化）；两个拷贝分割其祖先的功能（亚功能化）；两个拷贝都保留其祖先的功能。[1]</p><p>​ 大豆（Glycine max）是典型的古多倍体，经历了两轮全基因组复制事件（分别发生在大约 5900 万年前和 1300 万年前），导致近75％的基因以多拷贝形式存在。相比之下，水稻、玉米等谷物仅经历一次古多倍体化事件。[2]</p><h4 id="泛基因组及其研究方法简述"><a href="#泛基因组及其研究方法简述" class="headerlink" title="泛基因组及其研究方法简述"></a>泛基因组及其研究方法简述</h4><ul><li>泛基因组</li></ul><p>​ 植物基因组非常多样化，以至于很大一部分基因组序列没有在个体之间共享。可变DNA序列与保守的核心序列构成了更复杂的泛基因组，该基因组代表了一个物种中所有非冗余DNA的集合。因此，泛基因组代表了一个物种中所有DNA序列多样性的集合。</p><ul><li><p>植物泛基因组可变序列背后的SV的主要驱动力 [3]</p><p>驱动植物中SV产生的四种主要驱动力：TE活性；非等基因同源重组（NAHR）；基因渐渗（genetic introgression）&#x2F;水平基因转移（HGT）；多倍体植物中的偏向性基因丢失&#x2F;基因分馏（fractionation）。</p><ul><li>TE活性：转位后，它们经常诱导中到大（数百个碱基到数十个千碱基）InDels和其他复杂的SV。由于其广泛存在，TE在几乎所有植物泛基因组中都具有特征，其中它们构成了可变序列的很大一部分。</li><li>非等基因同源重组（NAHR）：当非等位重复DNA序列在减数分裂过程中错位，导致缺失、重复和其他类型的SV时就会发生。</li><li>基因渐渗（genetic introgression）&#x2F;水平基因转移（HGT）：虽然TE和NAHR都可以生成SV，但它们很少能够直接引入新的DNA序列。相比之下，最初不存在于物种基因库中的DNA序列可以通过基因渐渗或HGT获得。</li><li>多倍体植物中的偏向性基因丢失&#x2F;基因分馏（fractionation）：多倍体植物基因组复制后的fractionation是基因和非编码序列PAV背后的另一个主要动力。来自一个亚基因组的重复基因副本也倾向于更频繁地丢失，从而产生一些广泛的PAV基因。</li></ul></li></ul><p>​ 总之，SV极大程度上产生了植物泛基因组的可变序列。因此，构建好泛基因组之后，鉴定SVs是非常重要的步骤，这为后续的<strong>基因组分析、进化研究以及农业生产</strong>奠定了重要的基础。本篇文章就是一个将<strong>鉴定得到的SVs与应用联系</strong>起来的典范！</p><ul><li>泛基因组研究方法</li></ul><p>​ 在全基因组识别后，与SV相关的序列在现有参考基因组中没有同源性，可以沿着线性基因组迭代堆叠或锚定，或者以图泛基因组的形式以纳入更多新颖的序列变体。但是，由于变异序列以线性附加到骨干基因组序列上难以表征变异在基因组上的精确顺序信息，因此此类泛基因组受到限制。而图泛基因组以节点（nodes）代表序列片段，边（edges）代表连接关系（如顺序、方向），因此可整合多个个体的变异信息，形成“分支”结构，更完整地反映物种的遗传多样性。所以，图泛基因组是表征、容纳泛基因组变异信息的相对更好的策略。</p><h4 id="重测序（Resequencing）-与基因组从头组装（De-Novo-Genome-Assembly）"><a href="#重测序（Resequencing）-与基因组从头组装（De-Novo-Genome-Assembly）" class="headerlink" title="重测序（Resequencing） 与基因组从头组装（De Novo Genome Assembly）"></a>重测序（Resequencing） 与基因组从头组装（De Novo Genome Assembly）</h4><ul><li><p>重测序：对已有参考基因组的物种进行个体或群体的基因组测序，通过与参考序列比对来鉴定变异（如SNP、InDel、SV等），依赖参考基因组，适用于<strong>大样本群体遗传分析</strong>。</p></li><li><p>基因组从头组装：在<strong>不依赖参考基因组</strong>的情况下，直接利用测序数据将短读长或长读长序列拼接成完整的基因组序列。适用于<strong>挖掘新的结构变异</strong>、新物种基因组解析等等。</p></li></ul><p>​ 在本篇文章中，重测序的2027个种质（另外871个序列信息来自前期研究）作为群体用于用于检测SNPs、小indels等遗传变异，构建群体遗传图谱；进行群体结构分析（如构建系统发育树）。26个代表性种质的从头组装基因组用于构建泛基因组以识别更多的结构变异，以利于后续应用。</p><h4 id="本研究使用的三个参考基因组"><a href="#本研究使用的三个参考基因组" class="headerlink" title="本研究使用的三个参考基因组"></a>本研究使用的三个参考基因组</h4><ul><li>Wm82：第一个栽培品种大豆参考基因组，美国主栽大豆品种；</li><li>ZH13：代表中国大豆品种 “中黄 13” 的基因组，是中国种植最广泛的黄豆品种；</li><li>W05：野生大豆参考基因组。</li></ul><h2 id="Research-Strategy"><a href="#Research-Strategy" class="headerlink" title="Research Strategy"></a>Research Strategy</h2><p>本文的研究思路如下图所示：</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-cell_pangenome_soybean/reserch_strategy.png"></p><h2 id="Methods"><a href="#Methods" class="headerlink" title="Methods"></a>Methods</h2><p>​ 有关本文章 Methods 部分的介绍，将根据上述 Research Strategy 的顺序进行，结合流程图食用效果更佳哦！</p><h3 id="1-Sequencing"><a href="#1-Sequencing" class="headerlink" title="1. Sequencing"></a>1. <strong>Sequencing</strong></h3><ul><li>26个代表性种质的多组学测序<ul><li>SMRT 测序 - PacBio</li><li>HiSeq 测序 - Illumina</li><li>光学 Mapping - BioNano Irys</li><li>Hi-C 测序</li></ul></li><li>2027个大豆种质的全基因组重测序<ul><li>HiSeq 测序 - Illumina</li></ul></li><li>RNA-Seq and miRNA-Seq<ul><li>样本采集：涵盖不同时期的9个样本（生长阶段V1的<strong>根 茎 幼叶</strong>、生长阶段 R1的成熟<strong>叶 花</strong>、生长阶段 R4的<strong>老叶</strong>、4 周前的<strong>豆荚</strong>和<strong>种子</strong>、6 周的<strong>种子</strong>、8 周的<strong>种子</strong>）。</li><li>RNA-Seq在NovaSeq 平台上进行测序；miRNA-Seq 文库在 Nextseq 500平台上进行测序。</li></ul></li></ul><h3 id="2-SNP-Calling-and-Phylogenetic-Analyses"><a href="#2-SNP-Calling-and-Phylogenetic-Analyses" class="headerlink" title="2. SNP Calling and Phylogenetic Analyses"></a>2. <strong>SNP Calling and Phylogenetic Analyses</strong></h3><p>将2898个种质与ZH13参考基因组比对，得到全基因组范围内的SNPs，并建立系统发育树，如下图所示：</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-cell_pangenome_soybean/M2.png"></p><h3 id="3-Genome-Assembly"><a href="#3-Genome-Assembly" class="headerlink" title="3. Genome Assembly"></a>3. <strong>Genome Assembly</strong></h3><p>26个代表性基因组的组装整合了多种测序方法，以达到染色体水平上的准确组装，组装流程如下图所示：</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-cell_pangenome_soybean/M3.png"></p><ul><li>组装质量的评估：BUSCO。</li></ul><h3 id="4-Gene-Annotation"><a href="#4-Gene-Annotation" class="headerlink" title="4. Gene Annotation"></a>4. <strong>Gene Annotation</strong></h3><p>本研究通过多重策略进行基因注释，包括从头预测、基于同源性的基因预测、基于RNA-Seq的基因预测，最后整合基因预测结果获得完整且准确的基因注释文件。流程如下图所示：</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-cell_pangenome_soybean/M4.png"></p><ul><li>注释完整性的评估：BUSCO</li></ul><h3 id="5-Core-and-Dispensable-Gene-Family-Clustering"><a href="#5-Core-and-Dispensable-Gene-Family-Clustering" class="headerlink" title="5. Core and Dispensable Gene Family Clustering"></a>5. <strong>Core and Dispensable Gene Family Clustering</strong></h3><ul><li><p>基因家族聚类</p><p>根据基因家族在27个种质中的分布范围，分为 core genes（在27个种质中均存在）, softcore genes（在25-26个种质中存在）, dispensable genes（在24-2个种质中存在）, private genes（仅在单个种质中存在）。</p></li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-cell_pangenome_soybean/M5.png"></p><ul><li>功能注释与富集分析</li></ul><table><thead><tr><th><strong>分析类型</strong></th><th><strong>工具</strong> <strong>&#x2F;</strong> <strong>数据库</strong></th><th><strong>参数</strong> <strong>&#x2F;</strong> <strong>目的</strong></th></tr></thead><tbody><tr><td>KEGG 通路分析</td><td>KOBAS 3.0</td><td>解析基因参与的代谢与信号通路</td></tr><tr><td>蛋白质结构域注释</td><td>InterProScan 5</td><td>识别基因的功能结构域（如 AP2、WRKY 等）</td></tr><tr><td>GO 注释</td><td>PANNZER2</td><td>注释基因的生物学过程、分子功能等</td></tr><tr><td>富集分析</td><td>R 包 ClusterProfiler v3.10.1</td><td>筛选显著富集的功能类别（如应激响应、代谢通路）</td></tr><tr><td>QTL 关联</td><td>SoyBase 数据库</td><td>关联基因与农艺性状相关 QTL 区域</td></tr></tbody></table><h3 id="6-Calculation-of-population-genetic-parameters"><a href="#6-Calculation-of-population-genetic-parameters" class="headerlink" title="6. Calculation of population genetic parameters"></a>6. <strong>Calculation of population genetic parameters</strong></h3><p>以 500kb 不重叠窗口为单位，计算 dN (非同义替换率)、dS (同义替换率)、核苷酸多样性 (π)。</p><ul><li>核苷酸多样性（π）：群体内序列差异度，公式：π &#x3D; ∑(x_i (1-x_i))，x_i 为等位基因频率；</li><li>dN&#x2F;dS &gt; 1：正选择（适应性进化）；</li><li>dN&#x2F;dS &#x3D; 1：中性进化；</li><li>dN&#x2F;dS &lt; 1：负选择（纯化选择）。</li></ul><h3 id="7-Synteny-Analysis-and-Repeat-Analysis"><a href="#7-Synteny-Analysis-and-Repeat-Analysis" class="headerlink" title="7. Synteny Analysis and Repeat Analysis"></a>7. <strong>Synteny Analysis and Repeat Analysis</strong></h3><ul><li>共线性分析是比较基因组学中的核心方法，用于研究不同物种或同一物种不同基因组间的染色体区域同源性和基因排列顺序的保守性。通过共线性分析，可以推断物种间的进化关系、染色体结构变异、基因功能保守性等。这里共线性分析得到的结果可以<strong>用于PAV、CNV等的检测</strong>。共线性的分析流程如下图所示：</li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-cell_pangenome_soybean/M7.png"></p><ul><li>重复序列的分析采用了一种结合从头结构分析和同源性比较的方法。如下图所示：</li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-cell_pangenome_soybean/M7.2.png"></p><h3 id="8-Structural-Variation-Identification"><a href="#8-Structural-Variation-Identification" class="headerlink" title="8. Structural Variation Identification"></a>8. <strong>Structural Variation Identification</strong></h3><ul><li>SNP 和 INDEL 的鉴定是使用 MUMUmer4 的 show-saps 进行的；</li><li>本文将<strong>大于50bp</strong> 的插入或者缺失变异定义为 PAV，本文使用 SVMU (structural variants from MUMmer) pipeline 进行PAV的鉴定。具体流程如下图所示：</li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-cell_pangenome_soybean/M8.png"></p><ul><li>CNV ( 拷贝数变异 ) 的检测</li></ul><p>首先过滤掉长度小于 100 bp 的共线块，之后将具有2个或更多相互重叠、序列一致性大于 90% 的独立共线块的序列区定义为CNV。</p><ul><li>易位和倒位的检测</li></ul><p>本文将结构变异大于等于1000 bp的变异用于易位和倒位的检测。基于MUMmer4 比对得到的非等位同源块，人工检查变异区相对于相邻块的位置和方向来确定易位和倒位。</p><ul><li><p>SVs的合并</p><p>以ZH13 作为参考基因组，SoyW01 作为初始数据集，再逐个添加其他样本中的独特SVs，最终得到一个非冗余性的SVs集合。</p></li></ul><h3 id="8-graph-based-genome-for-soybean"><a href="#8-graph-based-genome-for-soybean" class="headerlink" title="8. graph-based genome for soybean"></a>8. <strong>graph-based genome for soybean</strong></h3><ul><li><p>构建图泛基因组时，以ZH13基因组作为参考，将重复序列小于90%的非冗余结构变异保存在vcf文件中，之后，通过vg（v1.6.0）进行基于图泛基因组的构建。</p></li><li><p>为了对2898个种质中的结构变异进行基因分型，使用默认参数，通过vg工具包将每个种质材料的短读段比对到图泛基因组上。</p></li></ul><h2 id="Results-——-application"><a href="#Results-——-application" class="headerlink" title="Results —— application"></a>Results —— application</h2><p>如果你跟着 Research Strategy 一路顺下来的话，一定会发现似乎 Methods 部分并没有结束。是的！如上所述，本文章就如何应用泛基因组做出了很好的尝试，因此，我将以应用部分的 results 为重点，并将这部分用到的 methods 穿插其中，以便更好地梳理和理解泛基因组的应用。</p><p>PS：仅对结果做方法与应用上的梳理，不会详细解释每张图片，详情请自己去看原文哦！</p><h3 id="1-Sequence-Variations-and-Paleopolyploid"><a href="#1-Sequence-Variations-and-Paleopolyploid" class="headerlink" title="1. Sequence Variations and Paleopolyploid"></a>1. <strong>Sequence Variations and Paleopolyploid</strong></h3><p>由于全基因组复制（ WGD ） 会导致基因组内出现大量重复的共线性区块（paralogons），因此通过基因组比对、检测共线块的方式来进行全基因组复制研究。</p><h4 id="1-1-Method：Synteny-Analysis"><a href="#1-1-Method：Synteny-Analysis" class="headerlink" title="1.1 Method：Synteny Analysis"></a>1.1 Method：Synteny Analysis</h4><ul><li>序列处理与比对</li></ul><p>屏蔽重复 DNA 区域，仅保留预测基因区域；使用 MUMmer4 比对基因编码序列的翻译产物，如下图所示：</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-cell_pangenome_soybean/M9.png"></p><ul><li>共线链构建</li></ul><p>将预处理得到的、认为可能是近期全基因组复制产生的重复区域，基于基因顺序而不是基因坐标构建共线性链，排除非直系同源基因以提高灵敏度，具体流程如下图所示：</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-cell_pangenome_soybean/M9.2.png"></p><h4 id="1-2-Sequence-Variations-and-Paleopolyploid"><a href="#1-2-Sequence-Variations-and-Paleopolyploid" class="headerlink" title="1.2 Sequence Variations and Paleopolyploid"></a>1.2 Sequence Variations and Paleopolyploid</h4><p>以** Wm 82 基因组**的 WGD 信息为参考，分析结构变异与 WGD 的关系。</p><p>为什么用 Wm82 的WGD信息为参考呢？因为 Wm82 是首个完成测序的大豆基因组，其 WGD 注释（如共线性区块、同源基因对）被认为是大豆研究的金标准。</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-cell_pangenome_soybean/R1.1.png"></p><ul><li>Fig A、B可知，WGD 倾向于发生在基因富集区域，远离重复DNA区域；</li><li>Fig C 可知，WGD区域的核苷酸多样性显著低于非WGD区域；</li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-cell_pangenome_soybean/R1.2.png"></p><ul><li>Fig D 可知，WGD 区域含有更高比例的核心基因和软核心基因；非 WGD 区域含有更高比例的可分配和私有基因；</li><li>Fig E 可知，在非WGD区域中鉴定出更多的SVs；WGD区域的私有SV数量少于非WGD区域；</li><li>综上，表明基因组复制不仅抑制了基因的进化速率，而且作为一种重要的遗传力量，塑造了SV的进化。</li></ul><p>之后本文探究了PAV是否会影响核苷酸多样性并比较WGD和非WGD区域的核苷酸多样性。因此，从每个PAV两侧提取了1kb的序列，并划分为100bp的窗口，计算各窗口的核苷酸多样性（如 π 值）。结果如下图所示：</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-cell_pangenome_soybean/R1.3.png"></p><p>结果表明，全基因组复制降低了核苷酸多样性，但对紧密围绕PAV的核苷酸多样性影响不大。</p><p><strong>综上，本文利用泛基因组鉴定到的SV，揭示了WGD与SV的关系，发现了WGD作为重要的遗传力量塑造了SV的进化。</strong></p><h3 id="2-Gene-Structure-Variation-and-Gene-Fusion"><a href="#2-Gene-Structure-Variation-and-Gene-Fusion" class="headerlink" title="2. Gene Structure Variation and Gene Fusion"></a>2. <strong>Gene Structure Variation and Gene Fusion</strong></h3><p>基因融合（Gene Fusion）是基因组中一种重要的结构变异，指两个或多个原本独立的基因通过<strong>染色体断裂、重排</strong>等机制连接成一个新的嵌合基因。<strong>下述条件B的判定逻辑就基于造成基因融合的机制</strong>。</p><h4 id="2-1-Method：Whole-Genome-Duplication-WGD-Investigation"><a href="#2-1-Method：Whole-Genome-Duplication-WGD-Investigation" class="headerlink" title="2.1 Method：Whole Genome Duplication (WGD) Investigation"></a>2.1 Method：Whole Genome Duplication (WGD) Investigation</h4><ul><li><p>候选基因对筛选</p><ul><li>序列比对：将两两基因组的 cDNA 序列进行 BLAST 比对；</li><li>融合事件判定：满足条件 A 或 B 的基因对被列为<strong>候选</strong>融合事件<ul><li>条件 A：基因组 A 中的单个基因与基因组 B 中的2 个或多个相邻分离基因匹配；</li><li>条件 B：这些分离基因所在区域在基因组 A 中无共线性基因（即该区域存在结构变异）；</li></ul></li></ul></li><li><p>从候选基因中进一步筛选</p><ul><li>蛋白质编码潜力筛选：排除开放阅读框（ORF）长度 &lt; 100 个氨基酸的 cDNA 序列；</li><li>匹配长度筛选：若分离基因与融合基因的匹配长度 &lt; 50%，则排除该候选对；</li><li>基因长度比例筛选：若融合基因长度 &lt; 分离基因总长度的 60%，则排除；</li><li>表达证据筛选：排除无 RNA-seq 转录数据支持的基因对。</li></ul></li></ul><p>通过上述两个步骤我们就筛选得到了发生基因融合的基因单元。</p><h4 id="2-2-Gene-Structure-Variation-and-Gene-Fusion"><a href="#2-2-Gene-Structure-Variation-and-Gene-Fusion" class="headerlink" title="2.2 Gene Structure Variation and Gene Fusion"></a>2.2 Gene Structure Variation and Gene Fusion</h4><p>本文构建的高质量泛基因组，使得更多的基因融合能够被更加准确地检测到。比如，E3是大豆主要开花位点的重要基因，在自然种群中检测到多个等位基因。</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-cell_pangenome_soybean/R2.1.png"></p><p>如图所示，本文通过泛基因组注释得到的SVs信息，发现了E3基因的三类变体。并且结合 RNA-Seq 数据发现了 E3-tr 的一个转录本发生了基因融合现象。之后又通过设计不同引物进行了验证（这里未做展示），发现在泛基因组的一些种质中，确实存在这类基因融合现象。这说明了泛基因组鉴定基因融合的准确性。</p><h3 id="3-Contribution-of-Structural-Variations-in-Soybean-Domestication"><a href="#3-Contribution-of-Structural-Variations-in-Soybean-Domestication" class="headerlink" title="3. Contribution of Structural Variations in Soybean Domestication"></a>3. <strong>Contribution of Structural Variations in Soybean Domestication</strong></h3><p>之前的研究鉴定到了很多可能与大豆驯化相关的关键基因座，但是这些关键基因组的遗传变异是如何产生的还未得到很好的确定。本文构建高质量泛基因组所鉴定出的大量 SVs，能够让我们有机会清晰地阐述大豆的驯化历程。</p><p>例如，种皮色素沉着的改变是大豆驯化过程中明显的选择性状之一。几乎所有野生大豆都表现出黑色种皮颜色，而大多数栽培品种则表现出黄色种皮颜色。之前定义I基因座是导致种皮颜色从黑色变为无色的重要驯化基因座。有研究表明，该基因座与黄色大豆种皮中查尔酮合酶（CHS）基因表达降低有关，这是通过同源依赖的基因沉默实现的。近期一项研究表明，该基因沉默可能是由CHS簇的倒位和基因重复引起的。</p><p>因此，本文就利用所鉴定到的 SVs 去看导致种皮发生变化的基因层面的原因。</p><h4 id="3-1-Method：CHS-gene-unit-identification"><a href="#3-1-Method：CHS-gene-unit-identification" class="headerlink" title="3.1 Method：CHS gene unit identification"></a>3.1 Method：CHS gene unit identification</h4><p>CHS 基因在 8 号染色体的I位点变现出高水平的重复。为了获得所有 CHS 单元（基因和&#x2F;或假基因），首先使用 <strong>BLASTn</strong> 在 29 个大豆基因组中检索了从 SoyZH13_08G103000 (CHS4)的起始密码子到终止密码子的基因组序列，之后将相似度大于 90% 的选定为候选 CHS 单元。之后通过 MEGA6 构建围绕I位点 SNP 的 NJ 系统发育树。</p><h4 id="3-2-Contribution-of-Structural-Variations-in-Soybean-Domestication"><a href="#3-2-Contribution-of-Structural-Variations-in-Soybean-Domestication" class="headerlink" title="3.2 Contribution of Structural Variations in Soybean Domestication"></a>3.2 Contribution of Structural Variations in Soybean Domestication</h4><p><img lazyload src="/images/loading.svg" data-src="/img/img-cell_pangenome_soybean/R3.png"></p><p>如图我们可以发现 H3、H4、H5鉴定得到的变异符合之前的研究，即黄色种皮可能是由CHS簇的倒位和基因重复引起的。但是，构建的泛基因组又发现了一类不属于这类变异但种皮表现出黄色的 H2 类群，鉴定发现来自 H1 的23.4kb 序列复制并倒置到CHS基因簇中，从而导致了 H2 类群的产生。</p><h3 id="4-Structural-Variations-Affect-Gene-Expression-and-Are-Associated-with-Agronomic-Traits"><a href="#4-Structural-Variations-Affect-Gene-Expression-and-Are-Associated-with-Agronomic-Traits" class="headerlink" title="4. Structural Variations Affect Gene Expression and Are Associated with Agronomic Traits"></a>4. <strong>Structural Variations Affect Gene Expression and Are Associated with Agronomic Traits</strong></h3><p>基因表达可能受到基因结构变异的影响，进而导致农艺性状的改变。</p><h4 id="4-1-Method：Gene-Expression"><a href="#4-1-Method：Gene-Expression" class="headerlink" title="4.1 Method：Gene  Expression"></a>4.1 Method：Gene Expression</h4><ul><li>本文利用RNA-Seq得到的数据又进行了基因表达分析，以探究SVs对大豆农业性状的影响。流程如下图所示：</li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-cell_pangenome_soybean/M12.png"></p><ul><li><p>RPKM是一种用于量化 RNA 测序数据中基因表达水平的标准化方法，适用于不同样本间或不同基因长度间的表达量的比较。RPKM &#x3D; 比对到目标基因的reads数 &#x2F; （基因长度kb * 测序数据中总比对 reads数 百万）</p></li><li><p>之后又通过主成分分析（PCA）比较了26个种质中这些基因的表达水平，然后进行了结构变异与表达主成分分析数据之间的相关性分析。</p></li></ul><h4 id="4-2-Structural-Variations-Affect-Gene-Expression-and-Are-Associated-with-Agronomic-Traits"><a href="#4-2-Structural-Variations-Affect-Gene-Expression-and-Are-Associated-with-Agronomic-Traits" class="headerlink" title="4.2 Structural Variations Affect Gene Expression and Are Associated with Agronomic Traits"></a>4.2 Structural Variations Affect Gene Expression and Are Associated with Agronomic Traits</h4><p>作者发现 17696 个 SVs 位于&#x2F;围绕高可信度核心基因（基因及其 ±3kb 侧翼区域）。为了研究结构变异是否影响更多基因表达，进行了结构变异与表达主成分分析数据之间的相关性分析，发现了1021个基因结构变异与不同种质见的表达水平变化相关。</p><p>这使得我们能够识别出一些可能与重要性状相关的候选基因。比如，缺铁性黄化是石灰质土壤中大豆生产的常见问题。研究表明，有几个重要的数量性状基因座（QTL）与缺铁性黄化有关，其中一个位于14号染色体上。还在该 QTL 上发现了一个编码 Fe2+&#x2F;Zn2+ 调控转运蛋白的基因（SoyZH13_14G179600,）。</p><p>作者基于泛基因组发现的 SVs 检测到了在不同种质中在该QTL区域发生的重要缺失，如图所示。</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-cell_pangenome_soybean/R4.png"></p><p>并发现该 PAV 与周围的五个多态性紧密连锁。这些变异将种质分为两类，不含缺失（Hap-1）以及含有缺失（Hap-2）的群体。通过比较两类的基因表达谱，可以发现含有缺失的类群具有更高的 SoyZH13_14G179600 基因表达。之后作者又结合不同地区土壤的 pH 值，预测这两类群体的地理分布也不同，并统计学分析证明了这种分布：Hap-2 种质较多位于较高纬度地区，Hap-1 较多位于较低纬度地区。</p><p>这表明了14G179600 的遗传分化有助于大豆在铁吸收方面的适应性。</p><p>综上所述，构建具有高质量的泛基因组为后续的<strong>基因组分析、进化研究以及农业生产</strong>奠定了重要的基础。</p><h3 id="参考文献"><a href="#参考文献" class="headerlink" title="参考文献"></a>参考文献</h3><p>[1] Dynamics of cis-regulatory sequences and transcriptional divergence of duplicated genes in soybean</p><p>[2] Genome sequence of the palaeopolyploid soybean</p><p>[3] Plant pan-genomics and its applications</p></div><div class="post-copyright-info w-full my-8 px-2 sm:px-6 md:px-8"><div class="article-copyright-info-container"><ul><li><strong>Title:</strong> Pan-Genome of Wild and Cultivated Soybeans</li><li><strong>Author:</strong> Yuchen</li><li><strong>Created at :</strong> 2025-06-22 10:56:00</li><li><strong>Updated at :</strong> 2025-06-24 14:59:06</li><li><strong>Link:</strong> https://yuchen-starfall.github.io/2025/06/22/Pan-Genome of Wild and Cultivated Soybeans/</li><li><strong>License: </strong>This work is licensed under <a class="license" target="_blank" rel="noopener" href="https://creativecommons.org/licenses/by-nc-sa/4.0">CC BY-NC-SA 4.0</a>.</li></ul></div></div><ul class="post-tags-box text-lg mt-1.5 flex-wrap justify-center flex md:hidden"><li class="tag-item mx-0.5"><a href="/tags/Article/">#Article</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/Pangenome/">#Pangenome</a>&nbsp;</li></ul><div class="article-nav my-8 flex justify-between items-center px-2 sm:px-6 md:px-8"><div class="article-next border-border-color shadow-redefine-flat shadow-shadow-color-2 rounded-medium px-4 py-2 hover:shadow-redefine-flat-hover hover:shadow-shadow-color-2"><a class="next" rel="next" href="/2025/06/11/CAS%20Evaluate%20teacher&#39;s%20work/"><span class="title flex justify-center items-center"><span class="post-nav-title-item">CAS Evaluate teacher&#39;s work</span> <span class="post-nav-item">Next posts</span> </span><span class="right arrow-icon flex justify-center items-center"><i class="fa-solid fa-chevron-right"></i></span></a></div></div><div class="comment-container px-2 sm:px-6 md:px-8 pb-8"><div class="comments-container mt-10 w-full"><div id="comment-anchor" class="w-full h-2.5"></div><div class="comment-area-title w-full my-1.5 md:my-2.5 text-xl md:text-3xl font-bold">Comments</div><div id="giscus-container"></div><script data-swup-reload-script defer>async function loadGiscus(){const t={src:"https://giscus.app/client.js","data-repo":"Yuchen-starfall/Yuchen-starfall.github.io","data-repo-id":"R_kgDONY6r-Q","data-category":"Announcements# Github discussion category","data-category-id":"DIC_kwDONY6r-c4Ck8Zm","data-mapping":"pathname","data-strict":"0","data-reactions-enabled":"1","data-emit-metadata":"1","data-theme":"preferred_color_scheme","data-lang":"en","data-input-position":"bottom","data-loading":"lazy",crossorigin:"anonymous",async:!0},a=document.createElement("script");for(const e in t)a.setAttribute(e,t[e]);document.getElementById("giscus-container").appendChild(a)}{let t=setTimeout(()=>{loadGiscus(),clearTimeout(t)},1e3)}</script></div></div></div><div class="toc-content-container"><div class="post-toc-wrap"><div class="post-toc"><div class="toc-title">On this page</div><div class="page-title">Pan-Genome of Wild and Cultivated Soybeans</div><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#Background"><span class="nav-text">Background</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#%E5%A4%A7%E8%B1%86%E5%9F%BA%E5%9B%A0%E7%BB%84%E6%A6%82%E8%BF%B0"><span class="nav-text">大豆基因组概述</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#%E6%B3%9B%E5%9F%BA%E5%9B%A0%E7%BB%84%E5%8F%8A%E5%85%B6%E7%A0%94%E7%A9%B6%E6%96%B9%E6%B3%95%E7%AE%80%E8%BF%B0"><span class="nav-text">泛基因组及其研究方法简述</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#%E9%87%8D%E6%B5%8B%E5%BA%8F%EF%BC%88Resequencing%EF%BC%89-%E4%B8%8E%E5%9F%BA%E5%9B%A0%E7%BB%84%E4%BB%8E%E5%A4%B4%E7%BB%84%E8%A3%85%EF%BC%88De-Novo-Genome-Assembly%EF%BC%89"><span class="nav-text">重测序（Resequencing） 与基因组从头组装（De Novo Genome Assembly）</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#%E6%9C%AC%E7%A0%94%E7%A9%B6%E4%BD%BF%E7%94%A8%E7%9A%84%E4%B8%89%E4%B8%AA%E5%8F%82%E8%80%83%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="nav-text">本研究使用的三个参考基因组</span></a></li></ol></li></ol><li class="nav-item nav-level-2"><a class="nav-link" href="#Research-Strategy"><span class="nav-text">Research Strategy</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Methods"><span class="nav-text">Methods</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1-Sequencing"><span class="nav-text">1. Sequencing</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-SNP-Calling-and-Phylogenetic-Analyses"><span class="nav-text">2. SNP Calling and Phylogenetic Analyses</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-Genome-Assembly"><span class="nav-text">3. Genome Assembly</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#4-Gene-Annotation"><span class="nav-text">4. Gene Annotation</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#5-Core-and-Dispensable-Gene-Family-Clustering"><span class="nav-text">5. Core and Dispensable Gene Family Clustering</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#6-Calculation-of-population-genetic-parameters"><span class="nav-text">6. Calculation of population genetic parameters</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#7-Synteny-Analysis-and-Repeat-Analysis"><span class="nav-text">7. Synteny Analysis and Repeat Analysis</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#8-Structural-Variation-Identification"><span class="nav-text">8. Structural Variation Identification</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#8-graph-based-genome-for-soybean"><span class="nav-text">8. graph-based genome for soybean</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Results-%E2%80%94%E2%80%94-application"><span class="nav-text">Results —— application</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1-Sequence-Variations-and-Paleopolyploid"><span class="nav-text">1. Sequence Variations and Paleopolyploid</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#1-1-Method%EF%BC%9ASynteny-Analysis"><span class="nav-text">1.1 Method：Synteny Analysis</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#1-2-Sequence-Variations-and-Paleopolyploid"><span class="nav-text">1.2 Sequence Variations and Paleopolyploid</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-Gene-Structure-Variation-and-Gene-Fusion"><span class="nav-text">2. Gene Structure Variation and Gene Fusion</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#2-1-Method%EF%BC%9AWhole-Genome-Duplication-WGD-Investigation"><span class="nav-text">2.1 Method：Whole Genome Duplication (WGD) Investigation</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#2-2-Gene-Structure-Variation-and-Gene-Fusion"><span class="nav-text">2.2 Gene Structure Variation and Gene Fusion</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-Contribution-of-Structural-Variations-in-Soybean-Domestication"><span class="nav-text">3. Contribution of Structural Variations in Soybean Domestication</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#3-1-Method%EF%BC%9ACHS-gene-unit-identification"><span class="nav-text">3.1 Method：CHS gene unit identification</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#3-2-Contribution-of-Structural-Variations-in-Soybean-Domestication"><span class="nav-text">3.2 Contribution of Structural Variations in Soybean Domestication</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#4-Structural-Variations-Affect-Gene-Expression-and-Are-Associated-with-Agronomic-Traits"><span class="nav-text">4. Structural Variations Affect Gene Expression and Are Associated with Agronomic Traits</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#4-1-Method%EF%BC%9AGene-Expression"><span class="nav-text">4.1 Method：Gene Expression</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#4-2-Structural-Variations-Affect-Gene-Expression-and-Are-Associated-with-Agronomic-Traits"><span class="nav-text">4.2 Structural Variations Affect Gene Expression and Are Associated with Agronomic Traits</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E5%8F%82%E8%80%83%E6%96%87%E7%8C%AE"><span class="nav-text">参考文献</span></a></li></ol></li></div></div></div></div></div></div><div class="main-content-footer"><footer class="footer mt-5 py-5 h-auto text-base text-third-text-color relative border-t-2 border-t-border-color"><div class="info-container py-3 text-center"><div class="text-center">&copy; <span>2024</span> - 2025&nbsp;&nbsp;<i class="fa-solid fa-heart fa-beat" style="--fa-animation-duration:0.5s;color:#f54545"></i>&nbsp;&nbsp;<a href="/">Yuchen</a><p class="post-count space-x-0.5"><span>53 posts in total</span></p></div><script data-swup-reload-script src="https://cn.vercount.one/js"></script><div class="relative text-center lg:absolute lg:right-[20px] lg:top-1/2 lg:-translate-y-1/2 lg:text-right"><span id="busuanzi_container_site_uv" class="lg:!block"><span class="text-sm">VISITOR COUNT</span> <span id="busuanzi_value_site_uv"></span> </span><span id="busuanzi_container_site_pv" class="lg:!block"><span class="text-sm">TOTAL PAGE VIEWS</span> <span id="busuanzi_value_site_pv"></span></span></div><div class="relative text-center lg:absolute lg:left-[20px] lg:top-1/2 lg:-translate-y-1/2 lg:text-left"><span class="lg:block text-sm">POWERED BY <?xml version="1.0" encoding="utf-8"?><!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"><svg class="relative top-[2px] inline-block align-baseline" version="1.1" id="圖層_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="1rem" height="1rem" viewBox="0 0 512 512" enable-background="new 0 0 512 512" xml:space="preserve"><path fill="#0E83CD" d="M256.4,25.8l-200,115.5L56,371.5l199.6,114.7l200-115.5l0.4-230.2L256.4,25.8z M349,354.6l-18.4,10.7l-18.6-11V275H200v79.6l-18.4,10.7l-18.6-11v-197l18.5-10.6l18.5,10.8V237h112v-79.6l18.5-10.6l18.5,10.8V354.6z"/></svg><a target="_blank" class="text-base" href="https://hexo.io">Hexo</a></span> <span class="text-sm lg:block">THEME&nbsp;<a class="text-base" target="_blank" href="https://github.com/EvanNotFound/hexo-theme-redefine">Redefine v2.8.2</a></span></div><div>Blog up for <span class="odometer" id="runtime_days"></span> days <span class="odometer" id="runtime_hours"></span> hrs <span class="odometer" id="runtime_minutes"></span> Min <span class="odometer" id="runtime_seconds"></span> Sec</div><script data-swup-reload-script>try{function odometer_init(){document.querySelectorAll(".odometer").forEach(e=>{new Odometer({el:e,format:"( ddd).dd",duration:200})})}odometer_init()}catch(e){}</script></div></footer></div></div><div class="post-tools"><div class="post-tools-container"><ul class="article-tools-list"><li class="right-bottom-tools page-aside-toggle"><i class="fa-regular fa-outdent"></i></li><li class="go-comment"><i class="fa-regular fa-comments"></i></li></ul></div></div><div class="right-side-tools-container"><div class="side-tools-container"><ul class="hidden-tools-list"><li class="right-bottom-tools tool-font-adjust-plus flex justify-center items-center"><i class="fa-regular fa-magnifying-glass-plus"></i></li><li class="right-bottom-tools tool-font-adjust-minus flex justify-center items-center"><i class="fa-regular fa-magnifying-glass-minus"></i></li><li class="right-bottom-tools tool-dark-light-toggle flex justify-center items-center"><i class="fa-regular fa-moon"></i></li><li class="right-bottom-tools tool-scroll-to-top flex justify-center items-center"><i class="fa-regular fa-arrow-up"></i></li><li class="right-bottom-tools tool-scroll-to-bottom flex justify-center items-center"><i class="fa-regular fa-arrow-down"></i></li></ul><ul class="visible-tools-list"><li class="right-bottom-tools toggle-tools-list flex justify-center items-center"><i class="fa-regular fa-cog fa-spin"></i></li></ul></div></div><div class="image-viewer-container"><img src=""></div></main><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/Swup.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupSlideTheme.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupScriptsPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupProgressPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupScrollPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupPreloadPlugin.min.js"></script><script>const swup=new Swup({plugins:[new SwupScriptsPlugin({optin:!0}),new SwupProgressPlugin,new SwupScrollPlugin({offset:80}),new SwupSlideTheme({mainElement:".main-content-body"}),new SwupPreloadPlugin],containers:["#swup"]})</script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/imageViewer.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/utils.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/main.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/navbarShrink.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/scrollTopBottom.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/lightDarkSwitch.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/categoryList.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/codeBlock.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/lazyload.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/runtime.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/odometer.min.js"></script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/assets/odometer-theme-minimal.css"><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/Typed.min.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/typed.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/minimasonry.min.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/masonry.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/tocToggle.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/toc.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/tabs.js" data-swup-reload-script></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/moment-with-locales.min.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/essays.js" data-swup-reload-script></script></body></html>