<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><meta name="keywords" content="Hexo Theme Redefine"><meta name="author" content="Yuchen"><link rel="preconnect" href="https://fonts.googleapis.com"><link rel="preconnect" href="https://fonts.gstatic.com" crossorigin><link rel="preconnect" href="https://cdn.jsdelivr.net" crossorigin><link rel="canonical" href="http://example.com/2025/04/01/gwas statistics methods/"><meta name="robots" content="index,follow"><meta name="googlebot" content="index,follow"><meta name="revisit-after" content="1 days"><meta name="description" content="Hexo Theme Redefine, Redefine Your Hexo Journey."><meta property="og:type" content="article"><meta property="og:title" content="Statistical methods for population association studies"><meta property="og:url" content="http://example.com/2025/04/01/GWAS%20statistics%20methods/index.html"><meta property="og:site_name" content="Hexo"><meta property="og:description" content="Hexo Theme Redefine, Redefine Your Hexo Journey."><meta property="og:locale" content="en_US"><meta property="og:image" content="http://example.com/images/redefine-og.webp"><meta property="article:published_time" content="2025-04-01T13:57:00.000Z"><meta property="article:modified_time" content="2025-04-08T15:34:00.703Z"><meta property="article:author" content="John Doe"><meta property="article:tag" content="Biomedical Statistics"><meta property="article:tag" content="Model select"><meta property="article:tag" content="Article"><meta property="article:tag" content="GWAS"><meta property="article:tag" content="linkage disequilibrium（LD）"><meta name="twitter:card" content="summary"><meta name="twitter:image" content="http://example.com/images/redefine-og.webp"><link rel="icon" type="image/png" href="/images/profile.jpg" sizes="192x192"><link rel="apple-touch-icon" sizes="180x180" href="/images/profile.jpg"><meta name="theme-color" content="#A31F34"><link rel="shortcut icon" href="/images/profile.jpg"><title>Statistical methods for population association studies | Steps Among the Stars</title><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/Chillax/chillax.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/css/build/tailwind.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/GeistMono/geist-mono.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/Geist/geist.css"><script id="hexo-configurations">window.config={hostname:"example.com",root:"/",language:"en"},window.theme={articles:{style:{font_size:"16px",line_height:1.5,image_border_radius:"14px",image_alignment:"center",image_caption:!1,link_icon:!0,delete_mask:!1,title_alignment:"left",headings_top_spacing:{h1:"3.2rem",h2:"2.4rem",h3:"1.9rem",h4:"1.6rem",h5:"1.4rem",h6:"1.3rem"}},word_count:{enable:!0,count:!0,min2read:!0},author_label:{enable:!1,auto:!1,list:[]},code_block:{copy:!0,style:"mac",highlight_theme:{light:"github",dark:"vs2015"},font:{enable:!1,family:null,url:null}},toc:{enable:!0,max_depth:6,number:!1,expand:!0,init_open:!0},copyright:{enable:!0,default:"CC BY-NC-ND"},lazyload:!0,pangu_js:!1,recommendation:{enable:!1,title:"推荐阅读",limit:3,mobile_limit:2,placeholder:"/images/wallhaven-wqery6-light.webp",skip_dirs:[]}},colors:{primary:"#A31F34",secondary:null,default_mode:"light"},global:{fonts:{chinese:{enable:!1,family:null,url:null},english:{enable:!1,family:null,url:null},title:{enable:!1,family:null,url:null}},content_max_width:"1000px",sidebar_width:"210px",hover:{shadow:!0,scale:!1},scroll_progress:{bar:!1,percentage:!1},website_counter:{url:"https://cn.vercount.one/js",enable:!0,site_pv:!0,site_uv:!0,post_pv:!0},single_page:!0,preloader:{enable:"ture",custom_message:"Welcome to my little corner of chaos and brilliance!"},open_graph:{enable:!0,image:"/images/redefine-og.webp",description:"Hexo Theme Redefine, Redefine Your Hexo Journey."},google_analytics:{enable:!1,id:null}},home_banner:{enable:!0,style:"fixed",image:{light:"/images/light.jpeg",dark:"/images/dark.png"},title:"Resonant Journals",subtitle:{text:["Welcome to my little corner of chaos and brilliance!"],hitokoto:{enable:!1,show_author:!1,api:"https://v1.hitokoto.cn"},typing_speed:20,backing_speed:80,starting_delay:500,backing_delay:1500,loop:!1,smart_backspace:!0},text_color:{light:"#fff",dark:"#d1d1b6"},text_style:{title_size:"2.8rem",subtitle_size:"1.5rem",line_height:1.2},custom_font:{enable:!1,family:null,url:null},social_links:{enable:"ture",style:"default",links:{github:null,instagram:null,zhihu:null,twitter:null,email:"yuchen.starfall@outlook.com"},qrs:{weixin:null}}},plugins:{feed:{enable:!1},aplayer:{enable:!1,type:"fixed",audios:[{name:null,artist:null,url:null,cover:null,lrc:null}]},mermaid:{enable:!1,version:"11.4.1"}},version:"2.8.2",navbar:{auto_hide:"ture",color:{left:"#f78736",right:"#367df7",transparency:35},width:{home:"1200px",pages:"1000px"},links:{Home:{path:"/",icon:"fa-regular fa-house"},Archives:{path:"/archives",icon:"fa-regular fa-archive"},Categories:{icon:"fa-solid fa-folder",path:"/categories/"},Tags:{icon:"fa-solid fa-tags",path:"/tags/"}},search:{enable:!1,preload:!0}},page_templates:{friends_column:2,tags_style:"blur"},home:{sidebar:{enable:!0,position:"left",first_item:"menu",announcement:"Ephemeral Trails",show_on_mobile:!0,links:{Archives:{path:"/archives",icon:"fa-regular fa-archive"},Tags:{path:"/tags",icon:"fa-regular fa-tags"},Categories:{path:"/categories",icon:"fa-regular fa-folder"}}},article_date_format:"auto",excerpt_length:200,categories:{enable:!0,limit:3},tags:{enable:!0,limit:3}},footerStart:"2024/12/5 00:00:00"},window.lang_ago={second:"%s seconds ago",minute:"%s minutes ago",hour:"%s hours ago",day:"%s days ago",week:"%s weeks ago",month:"%s months ago",year:"%s years ago"},window.data={masonry:!0}</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/fontawesome.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/brands.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/solid.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/regular.min.css"><meta name="generator" content="Hexo 7.3.0"></head><body><div class="progress-bar-container"><span class="pjax-progress-bar"></span></div><main class="page-container" id="swup"><div class="main-content-container flex flex-col justify-between min-h-dvh"><div class="main-content-header"><header class="navbar-container px-6 md:px-12"><div class="navbar-content transition-navbar"><div class="left"><a class="logo-title" href="/">Steps Among the Stars</a></div><div class="right"><div class="desktop"><ul class="navbar-list"><li class="navbar-item"><a href="/"><i class="fa-regular fa-house fa-fw"></i> HOME</a></li><li class="navbar-item"><a href="/archives"><i class="fa-regular fa-archive fa-fw"></i> ARCHIVES</a></li><li class="navbar-item"><a href="/categories/"><i class="fa-solid fa-folder fa-fw"></i> CATEGORIES</a></li><li class="navbar-item"><a href="/tags/"><i class="fa-solid fa-tags fa-fw"></i> TAGS</a></li></ul></div><div class="mobile"><div class="icon-item navbar-bar"><div class="navbar-bar-middle"></div></div></div></div></div><div class="navbar-drawer h-dvh w-full absolute top-0 left-0 bg-background-color flex flex-col justify-between"><ul class="drawer-navbar-list flex flex-col px-4 justify-center items-start"><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/"><span>HOME </span><i class="fa-regular fa-house fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/archives"><span>ARCHIVES </span><i class="fa-regular fa-archive fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/categories/"><span>CATEGORIES </span><i class="fa-solid fa-folder fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/tags/"><span>TAGS </span><i class="fa-solid fa-tags fa-sm fa-fw"></i></a></li></ul><div class="statistics flex justify-around my-2.5"><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/tags"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">48</div><div class="label text-third-text-color text-sm">Tags</div></a><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/categories"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">7</div><div class="label text-third-text-color text-sm">Categories</div></a><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/archives"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">52</div><div class="label text-third-text-color text-sm">Posts</div></a></div></div><div class="window-mask"></div></header></div><div class="main-content-body transition-fade-up"><div class="main-content"><div class="post-page-container flex relative justify-between box-border w-full h-full"><div class="article-content-container"><div class="article-title relative w-full"><img src="/img/img-GWAS/4.png" alt="Statistical methods for population association studies" class="w-full h-60 sm:h-72 md:h-80 object-cover sm:rounded-t-large dark:brightness-75"><div class="w-full flex items-center absolute bottom-0 justify-start"><h1 class="article-title-cover text-center mx-6 my-6 text-second-text-color bg-background-color-transparent px-4 py-3 text-3xl sm:text-4xl md:text-5xl font-semibold backdrop-blur-lg rounded-xl border border-border-color">Statistical methods for population association studies</h1></div></div><div class="article-header flex flex-row gap-2 items-center px-2 sm:px-6 md:px-8"><div class="avatar w-[46px] h-[46px] flex-shrink-0 rounded-medium border border-border-color p-[1px]"><img src="/images/profile.jpg"></div><div class="info flex flex-col justify-between"><div class="author flex items-center"><span class="name text-default-text-color text-lg font-semibold">Yuchen</span></div><div class="meta-info"><div class="article-meta-info"><span class="article-date article-meta-item"><i class="fa-regular fa-pen-fancy"></i>&nbsp; <span class="desktop">2025-04-01 21:57</span> <span class="mobile">2025-04-01 21:57</span> <span class="hover-info">Created</span> </span><span class="article-date article-meta-item"><i class="fa-regular fa-wrench"></i>&nbsp; <span class="desktop">2025-04-08 23:34</span> <span class="mobile">2025-04-08 23:34</span> <span class="hover-info">Updated</span> </span><span class="article-categories article-meta-item"><i class="fa-regular fa-folders"></i>&nbsp;<ul><li><a href="/categories/Literature/">Literature</a>&nbsp;</li></ul></span><span class="article-tags article-meta-item"><i class="fa-regular fa-tags"></i>&nbsp;<ul><li><a href="/tags/Biomedical-Statistics/">Biomedical Statistics</a>&nbsp;</li><li>| <a href="/tags/Model-select/">Model select</a>&nbsp;</li><li>| <a href="/tags/Article/">Article</a>&nbsp;</li><li>| <a href="/tags/GWAS/">GWAS</a>&nbsp;</li><li>| <a href="/tags/linkage-disequilibrium%EF%BC%88LD%EF%BC%89/">linkage disequilibrium（LD）</a>&nbsp;</li></ul></span><span class="article-pv article-meta-item"><i class="fa-regular fa-eye"></i>&nbsp;<span id="busuanzi_value_page_pv"></span></span></div></div></div></div><div class="article-content markdown-body px-2 sm:px-6 md:px-8 pb-8"><blockquote><p>Title: A tutorial on statistical methods for population association studies</p><p>Excerpt:该综述介绍了早期GWAS研究中可用的的统计学工具。简要的介绍了GWAS研究核心的遗传学与统计学原理 ，并简要梳理了GWAS各个环节上会用到的基础的统计学原理与工具。对理解GWAS的检验原理有很大帮助，后续的GWAS检验方法基本是基于这些基本原理的扩展与补充，万变不离其宗。</p></blockquote><h2 id="Association-studies"><a href="#Association-studies" class="headerlink" title="Association studies"></a>Association studies</h2><p>即使在群体中个体是“无关”的，基因关联研究 (association studies) 仍然可以利用遗传标记 (marker alleles) 追踪潜在的致病变异。</p><p>祖先突变不仅会传给后代，还可能与周围的基因组区域<strong>连锁</strong>，使得一些遗传标记 (SNPs) 也会在病例组中更常见，使得标记可以作为<strong>遗传标志</strong>用于疾病研究。但是，随着时间推移和重组发生，这种连锁关系会减弱，只有最紧密相连的标记会保留。</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-GWAS/5.png"></p><p>常见疾病与常见变异 (common disease-common variant, CDCV) 假说，即许多复杂疾病可能是由频率较高但效应较小的变异所导致的。</p><h3 id="Spurious（虚假）associations-due-to-population-structure"><a href="#Spurious（虚假）associations-due-to-population-structure" class="headerlink" title="Spurious（虚假）associations due to population structure"></a>Spurious（虚假）associations due to population structure</h3><p>本质为病例与对照的遗传背景不匹配，导致偏差</p><div class="note-large default"><div class="notel-title rounded-t-lg p-3 font-bold text-lg flex flex-row gap-2 items-center"><i class="notel-icon fa-solid fa-info"></i><p>Example</p></div><div class="notel-content"><ul><li><strong>病例组</strong>：70%为北欧裔（白人），30%东亚裔</li><li><strong>对照组</strong>：30%北欧裔，70%东亚裔<ul><li>某个SNP在北欧人群中频率天然高（如乳糖耐受基因），与肺癌无关</li><li>结果：该SNP在病例组中频率显著高于对照组，被误判为”肺癌相关”</li></ul></li></ul></div></div><p>措施：</p><ul><li>病例对照严格匹配种族、地理来源</li><li>全基因组分析时默认包含<strong>PCA校正</strong></li><li>对显著信号进行跨群体验证</li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-GWAS/6.png"></p><h3 id="Gene-mapping"><a href="#Gene-mapping" class="headerlink" title="Gene mapping"></a>Gene mapping</h3><p>所有基因作图方法的核心思想是：疾病易感等位基因会与紧密连锁基因座的变异体共同代际传递。<strong>连锁研究</strong>通过已知家系直接追踪疾病表型与标记等位基因的代际传递，寻找提示标记与致病基因座连锁的关联性。</p><p>区别于association studies ，linkage analysis研究<strong>家族性遗传病</strong>，尤其当同一家族中有多位患者时。适用于<strong>家系数据</strong>，并且用于整合多家族<strong>罕见突变</strong>，不同与association studies的CDCV。</p><ul><li><strong>参数法</strong>：假设一个具体的疾病遗传模型（如显性&#x2F;隐性遗传），通过家谱追踪致病基因和标记基因的传递是否同步。<br><em>举例</em>：若某个标记基因总是和疾病同时出现在患者身上，则认为两者”连锁”。</li><li><strong>非参数法</strong>：不假设遗传模型，直接看患病亲属之间是否共享更多相同基因片段。</li></ul><h2 id="Preliminary-analyses"><a href="#Preliminary-analyses" class="headerlink" title="Preliminary analyses"></a>Preliminary analyses</h2><p>Data quality is of importance , data should be checked thoroughly</p><h3 id="HWE"><a href="#HWE" class="headerlink" title="HWE"></a>HWE</h3><p>HWE告诉我们群体中的<strong>等位基因频率以及基因型频率并不随世代的推移而变化</strong>。如果发现如果某个SNP显著偏离HWE，可能是<strong>基因分型错误</strong>导致的。可以剔除SNP位点，提高数据质量例如：</p><ul><li><strong>实验误差</strong>：测序或基因芯片可能会错误地识别某些等位基因。</li><li><strong>杂合子误判</strong>：某些实验技术可能错误地将<strong>杂合子（Aa）标记为纯合子（AA 或 aa）</strong>，导致基因型频率与HWE期望值不符。</li><li><strong>PCR扩增失败</strong>：如果PCR扩增区域包含突变，某些等位基因可能无法被正确检测。</li></ul><p>同时，HEW还可以用于发现<strong>种群分层</strong>现象。HWE假设种群是<strong>随机交配</strong>的，但在实际研究中，样本可能来自<strong>遗传背景不同的亚群</strong>（例如不同的民族或地理区域）。如果某个SNP在不同亚群的等位基因频率不同，合并后的数据就可能表现出HWE偏离。</p><p>在某些情况下，HWE偏离可能不是数据错误，而是<strong>生物学信号</strong>，表明该SNP可能与<strong>疾病或自然选择</strong>相关。因此在GWAS数据处理中，通常在<strong>对照组</strong>中进行HWE检验，以剔除可能的错误位点，并在<strong>病例组</strong>中分析HWE偏离，以寻找可能的疾病关联SNP。</p><div class="note-large default"><div class="notel-title rounded-t-lg p-3 font-bold text-lg flex flex-row gap-2 items-center"><i class="notel-icon fa-solid fa-info"></i><p>methods</p></div><div class="notel-content"><p>比较观测基因型频数与HWE下的<strong>期望频数</strong>（$E_{AA}&#x3D;p^2N,E_{Aa}&#x3D;2pqN,E_{aa}&#x3D;q^2N$）。</p><p>$\chi^2&#x3D;\sum\frac{(O-E)^2}{E}$,k个等位基因自由度$df&#x3D;\frac{k(k-1)}{2}$</p><p>若频数过低（如罕见变异），卡方近似不准确，需改用<strong>费舍尔精确检验</strong>。</p></div></div><h4 id="Log-QQ-quantile–quantile-P-value-plot"><a href="#Log-QQ-quantile–quantile-P-value-plot" class="headerlink" title="Log QQ (quantile–quantile) P-value plot"></a>Log QQ (quantile–quantile) P-value plot</h4><ul><li><strong>目的</strong>：评估多个SNP的HWE检验P值整体分布是否符合期望（即零假设成立时P值应服从均匀分布）。<ul><li><strong>分位数（Quantile）</strong>：指将数据按大小排序后，处于特定百分比位置的值</li><li>将观测数据的分位数与理论分布的分位数一一对应比较。</li></ul></li><li><strong>绘制方法</strong>：<ul><li>横轴（X）：理论分位数，即 $−log10(i&#x2F;(L+1))$，L为SNP总数，i为第i小的P值排序。</li><li>纵轴（Y）：观测分位数，即$ −log10(Pi)$。</li></ul></li><li><strong>正常情况</strong>：散点沿Y&#x3D;X<em>Y</em>&#x3D;<em>X</em>线分布。</li><li><strong>异常信号</strong>：<ul><li><strong>整体右偏</strong>（散点高于对角线）：大量SNP偏离HWE，提示群体分层或系统性基因分型错误。</li><li><strong>个别离群点</strong>：特定SNP显著偏离，需检查分型质量或潜在生物学因素（如选择压力）。</li></ul></li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-GWAS/7.png"></p><h3 id="Missing-genotype-data"><a href="#Missing-genotype-data" class="headerlink" title="Missing genotype data"></a>Missing genotype data</h3><p>Replacing missing genotypes with predicted values that are based on the observed genotypes at neighbouring SNPs. – 连锁紧密区域，相邻SNP基因型存在关联性</p><p>填补方法分类</p><ul><li><strong>单次填补</strong><ul><li><strong>最大似然估计</strong>：基于统计学模型，计算最可能的基因型（如概率最高的基因型）。</li></ul></li><li><strong>多重填补</strong><ul><li>生成多个可能的填补结果（如随机抽样），通过多次分析后合并结果，反映填补的不确定性。</li></ul></li><li><strong>热卡填补</strong>（‘hot-deck’ approaches）<ul><li>从其他基因型相似的个体中直接复制缺失值（如邻近位点匹配的个体）。</li></ul></li><li><strong>回归模型</strong>（regression models）<ul><li>利用邻近SNP的基因型构建回归模型，预测缺失值。</li></ul></li></ul><p>检测填补是否存在偏倚，将“是否缺失”本身编码为变量（1&#x3D;存在，0&#x3D;缺失），检验其与表型（如病例&#x2F;对照）的关联性。若显著相关，说明缺失存在偏倚。</p><h3 id="Haplotype-and-genotype-data"><a href="#Haplotype-and-genotype-data" class="headerlink" title="Haplotype and genotype data"></a>Haplotype and genotype data</h3><ol><li>单倍型（Haplotype） vs. 基因型（Genotype）<ul><li><strong>基因型</strong>：个体在某个SNP位点的两个等位基因组合（如AA、AG、GG），不区分来自父方或母方。</li><li><strong>单倍型</strong>：由多个紧密连锁的SNP组成的染色体片段，每条染色体（父源或母源）对应一个单倍型。例如，若某个体在三个相邻SNP位点的基因型为AG、CT、GA，其单倍型可能为父源<code>A-C-G</code>和母源<code>G-T-A</code>的组合。</li></ul></li><li><strong>直接实验测定</strong><ul><li><strong>实验室单倍型分型</strong>：通过分子生物学技术（如长片段测序）直接确定单倍型，成本高昂且技术复杂。</li><li><strong>家系分析</strong>：通过检测父母或子女的基因型推断未知单倍型（需家系样本，不适用于无亲缘关系的群体研究）。</li></ul></li><li><strong>统计推断方法</strong><ul><li><strong>原理</strong>：基于群体中连锁SNP的单倍型多样性有限的假设（低重组区域中，常见单倍型种类少）。因此在强关联环境下推断的结果可信度更高</li><li><strong>常用软件</strong>：<ul><li><strong>SNPHAP</strong>：简单快速，适合初步分析</li><li><strong>PHASE</strong>：准确性高，但计算资源消耗大</li><li><strong>FASTPHASE</strong>：兼顾速度与准确性，接近PHASE的精度但效率显著提升</li></ul></li></ul></li></ol><h3 id="Measures-of-LD-and-estimates-of-recombination-rates"><a href="#Measures-of-LD-and-estimates-of-recombination-rates" class="headerlink" title="Measures of LD and estimates of recombination rates"></a>Measures of LD and estimates of recombination rates</h3><p>Measures of LD：D’,r^2（D’揭示历史重组，r²指导实际功效）。具体参考LD的post</p><p><strong>核心机制</strong>：重组是破坏LD的主要生物学过程，重组率高的区域LD衰减更快。</p><ul><li><strong>PHASE</strong>：通过单倍型相位推断重组事件，精度高但计算成本大。</li><li><strong>LDHAT</strong>：基于群体遗传学模型，估计局部重组率。</li><li><strong>HOTSPOTTER</strong>：专门检测重组热点。</li><li><strong>实证发现</strong>：重组率在细粒度上高度异质（如1 kb内差异可达百倍）。</li></ul><h3 id="Tagging-SNP"><a href="#Tagging-SNP" class="headerlink" title="Tagging SNP"></a>Tagging SNP</h3><ul><li><strong>定义</strong>：从大量SNP中选择最小子集，使其尽可能保留原始SNP集的遗传变异信息。 – International HapMap Project</li><li><strong>核心原理</strong>：利用连锁不平衡（LD）结构，通过少数标记SNP（tag SNP）代表与其强相关（如r² &gt; 0.9）的其他SNP。</li></ul><p><strong>复杂优化方法</strong></p><ul><li><strong>聚类算法</strong>：将SNP按LD结构分组，每组选一个代表SNP（如基于最大r²或覆盖最多SNP）。</li><li><strong>多态性最大化</strong>：选择能够解释最多遗传变异的SNP组合（参考文献32）。</li><li><strong>统计方法依赖性</strong>：标记策略需与后续分析方法匹配（如单倍型分析需保留块内关键SNP）。</li></ul><p><strong>局限性</strong></p><ul><li><strong>仅适用于常见变异</strong>：tag SNP难以有效代表低频或稀有变异（MAF &lt; 5%）。</li><li><strong>群体特异性</strong>：基于某一群体（如欧洲人群）选择的tag SNP可能在其他群体（如非洲人群）中效果不佳。</li><li><strong>LD动态性</strong>：不同人群的LD结构差异可能影响标记效能（需通过本地数据验证）。</li></ul><h2 id="Tests-of-association-single-SNP"><a href="#Tests-of-association-single-SNP" class="headerlink" title="Tests of association: single SNP"></a>Tests of association: single SNP</h2><p>底层逻辑：检测某个SNP的基因型分布是否在病例组与对照组间存在显著差异，或是否与表型（连续型或分类变量）显著相关。</p><h3 id="Methods"><a href="#Methods" class="headerlink" title="Methods"></a>Methods</h3><ol><li><p><strong>基础检验方法</strong></p><ul><li><p><strong>2×3列联表检验</strong></p><ul><li><strong>Pearson卡方检验(chi-square test)（2自由度）</strong>：比较病例与对照组中三种基因型（AA、Aa、aa）分布的差异，无需遗传模型假设。</li><li><strong>Fisher精确检验</strong>：适用于小样本或稀有等位基因（低MAF）场景，计算更严格但计算量更大。</li></ul></li><li><table><thead><tr><th align="left"></th><th align="left">AA</th><th align="left">Aa</th><th align="left">aa</th></tr></thead><tbody><tr><td align="left"><strong>病例组</strong></td><td align="left">120</td><td align="left">80</td><td align="left">30</td></tr><tr><td align="left"><strong>对照组</strong></td><td align="left">150</td><td align="left">100</td><td align="left">50</td></tr></tbody></table></li></ul></li><li><p><strong>遗传模型导向的优化检验</strong></p><ul><li><strong>加性模型（Armitage趋势检验）</strong><ul><li><strong>假设</strong>：杂合子（Aa）风险介于两种纯合子（AA与aa）之间（线性趋势）。</li><li><strong>方法</strong>：将基因型编码为0（AA）、1（Aa）、2（aa），进行趋势卡方检验（1自由度）。<ul><li>计算期望，得到方差，$\chi^2$统计量检验</li></ul></li><li><strong>局限</strong>：需假设哈迪-温伯格平衡（HWE），且对显性&#x2F;隐性模型敏感度低。</li></ul></li><li><strong>显性&#x2F;隐性模型检验</strong><ul><li><strong>显性模型</strong>：合并Aa与aa（或AA）为一组，采用2×2卡方检验（如Aa+aa vs. AA）。</li><li><strong>隐性模型</strong>：合并AA与Aa为一组（如AA+Aa vs. aa）。</li></ul></li></ul></li><li><p><strong>进阶优化</strong></p><ul><li><strong>贝叶斯方法</strong><ul><li><strong>核心思想</strong>：整合先验知识（如加性模型更可能），计算后验概率评估关联强度。</li></ul></li></ul><blockquote><p>Lunn, D. J., Whittaker, J. C. &amp; Best, N. A Bayesian toolkit for genetic association studies. Genet. Epidemiol. 30, 231–247 (2006).</p></blockquote><ul><li><strong>多重检验校正</strong><ul><li><strong>Bonferroni校正</strong>：简单但保守（尤其SNP间存在LD时）。</li><li><strong>错误发现率（FDR）</strong>：更灵活，适用于探索性研究</li></ul></li></ul></li></ol><h3 id="Capture-continuou-variation-sassociation-methods-–-Regression-mods-vs-ANOVA"><a href="#Capture-continuou-variation-sassociation-methods-–-Regression-mods-vs-ANOVA" class="headerlink" title="Capture continuou variation sassociation methods –  Regression mods vs. ANOVA"></a>Capture continuou variation sassociation methods – Regression mods vs. ANOVA</h3><p>注意：连续性变量</p><ul><li><p><strong>线性回归（Linear Regression）</strong></p><ul><li><p><strong>目标</strong>：检验SNP基因型对连续表型（如身高、血糖）的影响是否符合线性关系。</p></li><li><p><strong>模型公式</strong>：</p><p>表型&#x3D;β0+β1×基因型编码+误差</p></li><li><p><strong>基因型编码方式</strong>：</p><ul><li><strong>加性模型</strong>：AA&#x3D;0，Aa&#x3D;1，aa&#x3D;2（假设每增加一个突变等位基因，表型变化β₁单位）。</li><li><strong>显性模型</strong>：AA&#x3D;0，Aa&#x3D;1，aa&#x3D;1（假设只要有突变等位基因，效应即最大）。</li><li><strong>隐性模型</strong>：AA&#x3D;0，Aa&#x3D;0，aa&#x3D;1（假设必须两个突变等位基因才有效应）。</li></ul></li></ul></li></ul><ul><li><p><strong>方差分析（ANOVA）</strong></p><ul><li><p><strong>目标</strong>：检验不同基因型组（AA、Aa、aa）的表型均值是否存在差异，不假设特定遗传模式。<strong>当遗传模式未知或非线性时，捕捉任意形式的关联</strong></p></li><li><p><strong>模型公式</strong>：</p><p>表型&#x3D;μ+α基因型+误差（μ为总均值，α为各基因型组的效应）</p></li><li><p><strong>自由度（df&#x3D;2）</strong>：比较三组的均值差异，需估计两个参数（如α_Aa和α_aa）。</p></li></ul></li></ul><h3 id="Logistic-Regression"><a href="#Logistic-Regression" class="headerlink" title="Logistic Regression"></a>Logistic Regression</h3><p>检验某个SNP的基因型是否与二元表型（病例&#x2F;对照）显著关联，同时可调整协变量（如年龄、性别）。</p><p>$logit(P(Y&#x3D;1))&#x3D;β_0+β_1\times基因型编码+β_2\times协变量+⋯$</p><ul><li>Y&#x3D;1：病例（患病）；Y&#x3D;0：对照（健康）。</li><li>P(Y&#x3D;1)：个体患病的概率。</li><li>β1：SNP的效应量（对数优势比，log odds ratio）。</li></ul><h3 id="Score-tests"><a href="#Score-tests" class="headerlink" title="Score tests"></a>Score tests</h3><p>得分检验是一种基于 <strong>似然函数导数（得分函数）</strong> 的统计假设检验方法，用于在原假设（H₀）下快速评估某个参数是否显著，而无需拟合备择假设（H₁）的完整模型。</p><h3 id="Ordered-categorical-outcomes"><a href="#Ordered-categorical-outcomes" class="headerlink" title="Ordered categorical outcomes"></a>Ordered categorical outcomes</h3><p>保留分类变量的顺序信息（疾病分级等信息）建模</p><h2 id="Dealing-with-population-stratification"><a href="#Dealing-with-population-stratification" class="headerlink" title="Dealing with population stratification"></a>Dealing with population stratification</h2><h3 id="Genomic-control-（GC）"><a href="#Genomic-control-（GC）" class="headerlink" title="Genomic control （GC）"></a>Genomic control （GC）</h3><p>利用大量与表型无关的 <strong>“无效SNP（Null SNPs）”</strong> 估计群体分层引起的检验统计量膨胀（λ），并对候选SNP的统计量进行校正。（假设无群体分层时，统计量应服从χ²分布）</p><h3 id="Structured-Association"><a href="#Structured-Association" class="headerlink" title="Structured Association"></a>Structured Association</h3><p>假设样本来自若干隐性亚群体（K个），通过模型（如Structure软件）估计个体属于各亚群体的概率（Q矩阵），在关联分析中控制亚群体效应。</p><h3 id="Principal-Component-Analysis-PCA"><a href="#Principal-Component-Analysis-PCA" class="headerlink" title="Principal Component Analysis, PCA"></a>Principal Component Analysis, PCA</h3><p>通过PCA提取全基因组SNP的主成分（PCs），反映样本的祖先差异，将前几个PC作为协变量纳入模型</p><h3 id="Mixed-Linear-Model-MLM"><a href="#Mixed-Linear-Model-MLM" class="headerlink" title="Mixed Linear Model, MLM"></a>Mixed Linear Model, MLM</h3><p>通过 <strong>亲缘关系矩阵（Kinship Matrix）</strong> 建模样本间的遗传相关性，校正群体分层和隐性亲缘关系。</p><h2 id="Tests-of-association-multiple-SNP"><a href="#Tests-of-association-multiple-SNP" class="headerlink" title="Tests of association: multiple SNP"></a>Tests of association: multiple SNP</h2><p>在候选基因或LD区块内，同时分析多个SNP的联合效应，捕捉单SNP分析可能忽略的交互信息或连锁不平衡（LD）结构。</p></div><div class="post-copyright-info w-full my-8 px-2 sm:px-6 md:px-8"><div class="article-copyright-info-container"><ul><li><strong>Title:</strong> Statistical methods for population association studies</li><li><strong>Author:</strong> Yuchen</li><li><strong>Created at :</strong> 2025-04-01 21:57:00</li><li><strong>Updated at :</strong> 2025-04-08 23:34:00</li><li><strong>Link:</strong> https://yuchen-starfall.github.io/2025/04/01/GWAS statistics methods/</li><li><strong>License: </strong>This work is licensed under <a class="license" target="_blank" rel="noopener" href="https://creativecommons.org/licenses/by-nc-sa/4.0">CC BY-NC-SA 4.0</a>.</li></ul></div></div><ul class="post-tags-box text-lg mt-1.5 flex-wrap justify-center flex md:hidden"><li class="tag-item mx-0.5"><a href="/tags/Biomedical-Statistics/">#Biomedical Statistics</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/Model-select/">#Model select</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/Article/">#Article</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/GWAS/">#GWAS</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/linkage-disequilibrium%EF%BC%88LD%EF%BC%89/">#linkage disequilibrium（LD）</a>&nbsp;</li></ul><div class="article-nav my-8 flex justify-between items-center px-2 sm:px-6 md:px-8"><div class="article-prev border-border-color shadow-redefine-flat shadow-shadow-color-2 rounded-medium px-4 py-2 hover:shadow-redefine-flat-hover hover:shadow-shadow-color-2"><a class="prev" rel="prev" href="/2025/04/08/Analysis%20of%20Population%20Genetic%20Structure/"><span class="left arrow-icon flex justify-center items-center"><i class="fa-solid fa-chevron-left"></i> </span><span class="title flex justify-center items-center"><span class="post-nav-title-item">Analysis of Population Genetic Structure</span> <span class="post-nav-item">Prev posts</span></span></a></div><div class="article-next border-border-color shadow-redefine-flat shadow-shadow-color-2 rounded-medium px-4 py-2 hover:shadow-redefine-flat-hover hover:shadow-shadow-color-2"><a class="next" rel="next" href="/2025/03/25/Haplotype%20estimation%20and%20linkage%20disequilibrium%20analysis/"><span class="title flex justify-center items-center"><span class="post-nav-title-item">Haplotype estimation and linkage disequilibrium(LD) analysis</span> <span class="post-nav-item">Next posts</span> </span><span class="right arrow-icon flex justify-center items-center"><i class="fa-solid fa-chevron-right"></i></span></a></div></div><div class="comment-container px-2 sm:px-6 md:px-8 pb-8"><div class="comments-container mt-10 w-full"><div id="comment-anchor" class="w-full h-2.5"></div><div class="comment-area-title w-full my-1.5 md:my-2.5 text-xl md:text-3xl font-bold">Comments</div><div id="giscus-container"></div><script data-swup-reload-script defer>async function loadGiscus(){const t={src:"https://giscus.app/client.js","data-repo":"Yuchen-starfall/Yuchen-starfall.github.io","data-repo-id":"R_kgDONY6r-Q","data-category":"Announcements# Github discussion category","data-category-id":"DIC_kwDONY6r-c4Ck8Zm","data-mapping":"pathname","data-strict":"0","data-reactions-enabled":"1","data-emit-metadata":"1","data-theme":"preferred_color_scheme","data-lang":"en","data-input-position":"bottom","data-loading":"lazy",crossorigin:"anonymous",async:!0},a=document.createElement("script");for(const e in t)a.setAttribute(e,t[e]);document.getElementById("giscus-container").appendChild(a)}{let t=setTimeout(()=>{loadGiscus(),clearTimeout(t)},1e3)}</script></div></div></div><div class="toc-content-container"><div class="post-toc-wrap"><div class="post-toc"><div class="toc-title">On this page</div><div class="page-title">Statistical methods for population association studies</div><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#Association-studies"><span class="nav-text">Association studies</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Spurious%EF%BC%88%E8%99%9A%E5%81%87%EF%BC%89associations-due-to-population-structure"><span class="nav-text">Spurious（虚假）associations due to population structure</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Gene-mapping"><span class="nav-text">Gene mapping</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Preliminary-analyses"><span class="nav-text">Preliminary analyses</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#HWE"><span class="nav-text">HWE</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Log-QQ-quantile%E2%80%93quantile-P-value-plot"><span class="nav-text">Log QQ (quantile–quantile) P-value plot</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Missing-genotype-data"><span class="nav-text">Missing genotype data</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Haplotype-and-genotype-data"><span class="nav-text">Haplotype and genotype data</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Measures-of-LD-and-estimates-of-recombination-rates"><span class="nav-text">Measures of LD and estimates of recombination rates</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Tagging-SNP"><span class="nav-text">Tagging SNP</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Tests-of-association-single-SNP"><span class="nav-text">Tests of association: single SNP</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Methods"><span class="nav-text">Methods</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Capture-continuou-variation-sassociation-methods-%E2%80%93-Regression-mods-vs-ANOVA"><span class="nav-text">Capture continuou variation sassociation methods – Regression mods vs. ANOVA</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Logistic-Regression"><span class="nav-text">Logistic Regression</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Score-tests"><span class="nav-text">Score tests</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Ordered-categorical-outcomes"><span class="nav-text">Ordered categorical outcomes</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Dealing-with-population-stratification"><span class="nav-text">Dealing with population stratification</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Genomic-control-%EF%BC%88GC%EF%BC%89"><span class="nav-text">Genomic control （GC）</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Structured-Association"><span class="nav-text">Structured Association</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Principal-Component-Analysis-PCA"><span class="nav-text">Principal Component Analysis, PCA</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Mixed-Linear-Model-MLM"><span class="nav-text">Mixed Linear Model, MLM</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Tests-of-association-multiple-SNP"><span class="nav-text">Tests of association: multiple SNP</span></a></li></ol></div></div></div></div></div></div><div class="main-content-footer"><footer class="footer mt-5 py-5 h-auto text-base text-third-text-color relative border-t-2 border-t-border-color"><div class="info-container py-3 text-center"><div class="text-center">&copy; <span>2024</span> - 2025&nbsp;&nbsp;<i class="fa-solid fa-heart fa-beat" style="--fa-animation-duration:0.5s;color:#f54545"></i>&nbsp;&nbsp;<a href="/">Yuchen</a><p class="post-count space-x-0.5"><span>52 posts in total</span></p></div><script data-swup-reload-script src="https://cn.vercount.one/js"></script><div class="relative text-center lg:absolute lg:right-[20px] lg:top-1/2 lg:-translate-y-1/2 lg:text-right"><span id="busuanzi_container_site_uv" class="lg:!block"><span class="text-sm">VISITOR COUNT</span> <span id="busuanzi_value_site_uv"></span> </span><span id="busuanzi_container_site_pv" class="lg:!block"><span class="text-sm">TOTAL PAGE VIEWS</span> <span id="busuanzi_value_site_pv"></span></span></div><div class="relative text-center lg:absolute lg:left-[20px] lg:top-1/2 lg:-translate-y-1/2 lg:text-left"><span class="lg:block text-sm">POWERED BY <?xml version="1.0" encoding="utf-8"?><!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"><svg class="relative top-[2px] inline-block align-baseline" version="1.1" id="圖層_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="1rem" height="1rem" viewBox="0 0 512 512" enable-background="new 0 0 512 512" xml:space="preserve"><path fill="#0E83CD" d="M256.4,25.8l-200,115.5L56,371.5l199.6,114.7l200-115.5l0.4-230.2L256.4,25.8z M349,354.6l-18.4,10.7l-18.6-11V275H200v79.6l-18.4,10.7l-18.6-11v-197l18.5-10.6l18.5,10.8V237h112v-79.6l18.5-10.6l18.5,10.8V354.6z"/></svg><a target="_blank" class="text-base" href="https://hexo.io">Hexo</a></span> <span class="text-sm lg:block">THEME&nbsp;<a class="text-base" target="_blank" href="https://github.com/EvanNotFound/hexo-theme-redefine">Redefine v2.8.2</a></span></div><div>Blog up for <span class="odometer" id="runtime_days"></span> days <span class="odometer" id="runtime_hours"></span> hrs <span class="odometer" id="runtime_minutes"></span> Min <span class="odometer" id="runtime_seconds"></span> Sec</div><script data-swup-reload-script>try{function odometer_init(){document.querySelectorAll(".odometer").forEach(e=>{new Odometer({el:e,format:"( ddd).dd",duration:200})})}odometer_init()}catch(e){}</script></div></footer></div></div><div class="post-tools"><div class="post-tools-container"><ul class="article-tools-list"><li class="right-bottom-tools page-aside-toggle"><i class="fa-regular fa-outdent"></i></li><li class="go-comment"><i class="fa-regular fa-comments"></i></li></ul></div></div><div class="right-side-tools-container"><div class="side-tools-container"><ul class="hidden-tools-list"><li class="right-bottom-tools tool-font-adjust-plus flex justify-center items-center"><i class="fa-regular fa-magnifying-glass-plus"></i></li><li class="right-bottom-tools tool-font-adjust-minus flex justify-center items-center"><i class="fa-regular fa-magnifying-glass-minus"></i></li><li class="right-bottom-tools tool-dark-light-toggle flex justify-center items-center"><i class="fa-regular fa-moon"></i></li><li class="right-bottom-tools tool-scroll-to-top flex justify-center items-center"><i class="fa-regular fa-arrow-up"></i></li><li class="right-bottom-tools tool-scroll-to-bottom flex justify-center items-center"><i class="fa-regular fa-arrow-down"></i></li></ul><ul class="visible-tools-list"><li class="right-bottom-tools toggle-tools-list flex justify-center items-center"><i class="fa-regular fa-cog fa-spin"></i></li></ul></div></div><div class="image-viewer-container"><img src=""></div></main><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/Swup.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupSlideTheme.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupScriptsPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupProgressPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupScrollPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupPreloadPlugin.min.js"></script><script>const swup=new Swup({plugins:[new SwupScriptsPlugin({optin:!0}),new SwupProgressPlugin,new SwupScrollPlugin({offset:80}),new SwupSlideTheme({mainElement:".main-content-body"}),new SwupPreloadPlugin],containers:["#swup"]})</script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/imageViewer.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/utils.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/main.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/navbarShrink.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/scrollTopBottom.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/lightDarkSwitch.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/categoryList.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/codeBlock.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/lazyload.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/runtime.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/odometer.min.js"></script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/assets/odometer-theme-minimal.css"><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/Typed.min.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/typed.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/minimasonry.min.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/masonry.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/tocToggle.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/toc.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/tabs.js" data-swup-reload-script></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/moment-with-locales.min.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/essays.js" data-swup-reload-script></script></body></html>