<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><meta name="keywords" content="Hexo Theme Redefine"><meta name="author" content="Yuchen"><link rel="preconnect" href="https://fonts.googleapis.com"><link rel="preconnect" href="https://fonts.gstatic.com" crossorigin><link rel="preconnect" href="https://cdn.jsdelivr.net" crossorigin><link rel="canonical" href="http://example.com/2025/06/10/gwas - plink/"><meta name="robots" content="index,follow"><meta name="googlebot" content="index,follow"><meta name="revisit-after" content="1 days"><meta name="description" content="Hexo Theme Redefine, Redefine Your Hexo Journey."><meta property="og:type" content="article"><meta property="og:title" content="GWAS -- PLINK"><meta property="og:url" content="http://example.com/2025/06/10/GWAS%20-%20PLINK/index.html"><meta property="og:site_name" content="Hexo"><meta property="og:description" content="Hexo Theme Redefine, Redefine Your Hexo Journey."><meta property="og:locale" content="en_US"><meta property="og:image" content="http://example.com/images/redefine-og.webp"><meta property="article:published_time" content="2025-06-10T07:32:00.000Z"><meta property="article:modified_time" content="2025-06-23T17:02:03.692Z"><meta property="article:author" content="John Doe"><meta property="article:tag" content="Biomedical Statistics"><meta property="article:tag" content="GWAS"><meta property="article:tag" content="linkage disequilibrium（LD）"><meta property="article:tag" content="Tools"><meta name="twitter:card" content="summary"><meta name="twitter:image" content="http://example.com/images/redefine-og.webp"><link rel="icon" type="image/png" href="/images/profile.jpg" sizes="192x192"><link rel="apple-touch-icon" sizes="180x180" href="/images/profile.jpg"><meta name="theme-color" content="#A31F34"><link rel="shortcut icon" href="/images/profile.jpg"><title>GWAS -- PLINK | Steps Among the Stars</title><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/Chillax/chillax.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/css/build/tailwind.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/GeistMono/geist-mono.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/Geist/geist.css"><script id="hexo-configurations">window.config={hostname:"example.com",root:"/",language:"en"},window.theme={articles:{style:{font_size:"16px",line_height:1.5,image_border_radius:"14px",image_alignment:"center",image_caption:!1,link_icon:!0,delete_mask:!1,title_alignment:"left",headings_top_spacing:{h1:"3.2rem",h2:"2.4rem",h3:"1.9rem",h4:"1.6rem",h5:"1.4rem",h6:"1.3rem"}},word_count:{enable:!0,count:!0,min2read:!0},author_label:{enable:!1,auto:!1,list:[]},code_block:{copy:!0,style:"mac",highlight_theme:{light:"github",dark:"vs2015"},font:{enable:!1,family:null,url:null}},toc:{enable:!0,max_depth:6,number:!1,expand:!0,init_open:!0},copyright:{enable:!0,default:"CC BY-NC-ND"},lazyload:!0,pangu_js:!1,recommendation:{enable:!1,title:"推荐阅读",limit:3,mobile_limit:2,placeholder:"/images/wallhaven-wqery6-light.webp",skip_dirs:[]}},colors:{primary:"#A31F34",secondary:null,default_mode:"light"},global:{fonts:{chinese:{enable:!1,family:null,url:null},english:{enable:!1,family:null,url:null},title:{enable:!1,family:null,url:null}},content_max_width:"1000px",sidebar_width:"210px",hover:{shadow:!0,scale:!1},scroll_progress:{bar:!1,percentage:!1},website_counter:{url:"https://cn.vercount.one/js",enable:!0,site_pv:!0,site_uv:!0,post_pv:!0},single_page:!0,preloader:{enable:"ture",custom_message:"Welcome to my little corner of chaos and brilliance!"},open_graph:{enable:!0,image:"/images/redefine-og.webp",description:"Hexo Theme Redefine, Redefine Your Hexo Journey."},google_analytics:{enable:!1,id:null}},home_banner:{enable:!0,style:"fixed",image:{light:"/images/light.jpeg",dark:"/images/dark.png"},title:"Resonant Journals",subtitle:{text:["Welcome to my little corner of chaos and brilliance!"],hitokoto:{enable:!1,show_author:!1,api:"https://v1.hitokoto.cn"},typing_speed:20,backing_speed:80,starting_delay:500,backing_delay:1500,loop:!1,smart_backspace:!0},text_color:{light:"#fff",dark:"#d1d1b6"},text_style:{title_size:"2.8rem",subtitle_size:"1.5rem",line_height:1.2},custom_font:{enable:!1,family:null,url:null},social_links:{enable:"ture",style:"default",links:{github:null,instagram:null,zhihu:null,twitter:null,email:"yuchen.starfall@outlook.com"},qrs:{weixin:null}}},plugins:{feed:{enable:!1},aplayer:{enable:!1,type:"fixed",audios:[{name:null,artist:null,url:null,cover:null,lrc:null}]},mermaid:{enable:!1,version:"11.4.1"}},version:"2.8.2",navbar:{auto_hide:"ture",color:{left:"#f78736",right:"#367df7",transparency:35},width:{home:"1200px",pages:"1000px"},links:{Home:{path:"/",icon:"fa-regular fa-house"},Archives:{path:"/archives",icon:"fa-regular fa-archive"},Categories:{icon:"fa-solid fa-folder",path:"/categories/"},Tags:{icon:"fa-solid fa-tags",path:"/tags/"}},search:{enable:!1,preload:!0}},page_templates:{friends_column:2,tags_style:"blur"},home:{sidebar:{enable:!0,position:"left",first_item:"menu",announcement:"Ephemeral Trails",show_on_mobile:!0,links:{Archives:{path:"/archives",icon:"fa-regular fa-archive"},Tags:{path:"/tags",icon:"fa-regular fa-tags"},Categories:{path:"/categories",icon:"fa-regular fa-folder"}}},article_date_format:"auto",excerpt_length:200,categories:{enable:!0,limit:3},tags:{enable:!0,limit:3}},footerStart:"2024/12/5 00:00:00"},window.lang_ago={second:"%s seconds ago",minute:"%s minutes ago",hour:"%s hours ago",day:"%s days ago",week:"%s weeks ago",month:"%s months ago",year:"%s years ago"},window.data={masonry:!0}</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/fontawesome.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/brands.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/solid.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/regular.min.css"><meta name="generator" content="Hexo 7.3.0"></head><body><div class="progress-bar-container"><span class="pjax-progress-bar"></span></div><main class="page-container" id="swup"><div class="main-content-container flex flex-col justify-between min-h-dvh"><div class="main-content-header"><header class="navbar-container px-6 md:px-12"><div class="navbar-content transition-navbar"><div class="left"><a class="logo-title" href="/">Steps Among the Stars</a></div><div class="right"><div class="desktop"><ul class="navbar-list"><li class="navbar-item"><a href="/"><i class="fa-regular fa-house fa-fw"></i> HOME</a></li><li class="navbar-item"><a href="/archives"><i class="fa-regular fa-archive fa-fw"></i> ARCHIVES</a></li><li class="navbar-item"><a href="/categories/"><i class="fa-solid fa-folder fa-fw"></i> CATEGORIES</a></li><li class="navbar-item"><a href="/tags/"><i class="fa-solid fa-tags fa-fw"></i> TAGS</a></li></ul></div><div class="mobile"><div class="icon-item navbar-bar"><div class="navbar-bar-middle"></div></div></div></div></div><div class="navbar-drawer h-dvh w-full absolute top-0 left-0 bg-background-color flex flex-col justify-between"><ul class="drawer-navbar-list flex flex-col px-4 justify-center items-start"><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/"><span>HOME </span><i class="fa-regular fa-house fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/archives"><span>ARCHIVES </span><i class="fa-regular fa-archive fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/categories/"><span>CATEGORIES </span><i class="fa-solid fa-folder fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/tags/"><span>TAGS </span><i class="fa-solid fa-tags fa-sm fa-fw"></i></a></li></ul><div class="statistics flex justify-around my-2.5"><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/tags"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">48</div><div class="label text-third-text-color text-sm">Tags</div></a><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/categories"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">7</div><div class="label text-third-text-color text-sm">Categories</div></a><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/archives"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">53</div><div class="label text-third-text-color text-sm">Posts</div></a></div></div><div class="window-mask"></div></header></div><div class="main-content-body transition-fade-up"><div class="main-content"><div class="post-page-container flex relative justify-between box-border w-full h-full"><div class="article-content-container"><div class="article-title relative w-full"><div class="w-full flex items-center pt-6 justify-start"><h1 class="article-title-regular text-second-text-color tracking-tight text-4xl md:text-6xl font-semibold px-2 sm:px-6 md:px-8 py-3">GWAS -- PLINK</h1></div></div><div class="article-header flex flex-row gap-2 items-center px-2 sm:px-6 md:px-8"><div class="avatar w-[46px] h-[46px] flex-shrink-0 rounded-medium border border-border-color p-[1px]"><img src="/images/profile.jpg"></div><div class="info flex flex-col justify-between"><div class="author flex items-center"><span class="name text-default-text-color text-lg font-semibold">Yuchen</span></div><div class="meta-info"><div class="article-meta-info"><span class="article-date article-meta-item"><i class="fa-regular fa-pen-fancy"></i>&nbsp; <span class="desktop">2025-06-10 15:32</span> <span class="mobile">2025-06-10 15:32</span> <span class="hover-info">Created</span> </span><span class="article-date article-meta-item"><i class="fa-regular fa-wrench"></i>&nbsp; <span class="desktop">2025-06-24 01:02:03</span> <span class="mobile">2025-06-24 01:02:03</span> <span class="hover-info">Updated</span> </span><span class="article-categories article-meta-item"><i class="fa-regular fa-folders"></i>&nbsp;<ul><li><a href="/categories/Tools/">Tools</a>&nbsp;</li></ul></span><span class="article-tags article-meta-item"><i class="fa-regular fa-tags"></i>&nbsp;<ul><li><a href="/tags/Biomedical-Statistics/">Biomedical Statistics</a>&nbsp;</li><li>| <a href="/tags/GWAS/">GWAS</a>&nbsp;</li><li>| <a href="/tags/linkage-disequilibrium%EF%BC%88LD%EF%BC%89/">linkage disequilibrium（LD）</a>&nbsp;</li><li>| <a href="/tags/Tools/">Tools</a>&nbsp;</li></ul></span><span class="article-pv article-meta-item"><i class="fa-regular fa-eye"></i>&nbsp;<span id="busuanzi_value_page_pv"></span></span></div></div></div></div><div class="article-content markdown-body px-2 sm:px-6 md:px-8 pb-8"><p>全基因组关联分析（Genome-Wide Association Study，GWAS），是以连锁不平衡（Linkage Disequilibrium，LD）为基础，利用基因组中数以百万计的单核苷酸多态性（Single Nucleotide Polymorphism，SNP）作为分子遗传标记，与可观测的表型在群体水平上的统计学分析方法。而<strong>PLINK</strong> 是一个开源的全基因组关联分析工具，由 Shaun Purcell 等人开发，最初发布于 2007 年，广泛应用于人类和植物、动物等群体的基因组学研究中</p><h2 id="PLINK-format"><a href="#PLINK-format" class="headerlink" title="PLINK format"></a>PLINK format</h2><p>We have 3 parts of information:</p><ol><li>Individual information</li><li>Variant information</li><li>Genotype matrix</li></ol><p>And there are different ways (format sets) to represent this information in PLINK1.9 and PLINK2:</p><ol><li>ped &#x2F; map</li><li>fam &#x2F; bim &#x2F; bed</li><li>psam &#x2F; pvar &#x2F; pgen</li></ol><p><img lazyload src="/images/loading.svg" data-src="/img/img-GWAS/8.png"></p><h3 id="ped-map"><a href="#ped-map" class="headerlink" title="ped &#x2F; map"></a>ped &#x2F; map</h3><p><code>.ped</code> (PLINK&#x2F;MERLIN&#x2F;Haploview text pedigree + genotype table)</p><p>Original standard text format for sample pedigree information and genotype calls.Contains no header line, and one line per sample with 2V+6 fields where V is the number of variants. The first six fields are the same as those in a .fam file. The seventh and eighth fields are allele calls for the first variant in the .map file (‘0’ &#x3D; no call); the 9th and 10th are allele calls for the second variant; and so on.</p><p><code>cut -d &quot; &quot; -f 1-16 1KG.EAS.auto.snp.norm.nodup.split.rare002.common015.missing.ped | head</code></p><p><strong>以空格分隔</strong></p><div class="code-container" data-rel="Plaintext"><figure class="iseeu highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">0 HG00403 0 0 0 -9 G G T T A A G A C C</span><br><span class="line">0 HG00404 0 0 0 -9 G G T T A A G A T C</span><br><span class="line">0 HG00406 0 0 0 -9 G G T T A A G A T C</span><br><span class="line">0 HG00407 0 0 0 -9 G G T T A A A A C C</span><br><span class="line">0 HG00409 0 0 0 -9 G G T T A A G A C C</span><br><span class="line">0 HG00410 0 0 0 -9 G G T T A A G A C C</span><br><span class="line">0 HG00419 0 0 0 -9 G G T T A A A A T C</span><br><span class="line">0 HG00421 0 0 0 -9 G G T T A A G A C C</span><br><span class="line">0 HG00422 0 0 0 -9 G G T T A A G A C C</span><br><span class="line">0 HG00428 0 0 0 -9 G G T T A A G A C C</span><br><span class="line">0 HG00436 0 0 0 -9 G G A T G A A A C C</span><br><span class="line">0 HG00437 0 0 0 -9 C G T T A A G A C C</span><br><span class="line">0 HG00442 0 0 0 -9 G G T T A A G A C C</span><br><span class="line">0 HG00443 0 0 0 -9 G G T T A A G A C C</span><br><span class="line">0 HG00445 0 0 0 -9 G G T T A A G A C C</span><br><span class="line">0 HG00446 0 0 0 -9 C G T T A A G A T C</span><br></pre></td></tr></table></figure></div><p><code>.map</code> (PLINK text fileset variant information file)</p><p>Variant information file accompanying a .ped text pedigree + genotype table. A text file with no header line, and one line per variant with the following 3-4 fields:</p><ul><li><p>Chromosome code. PLINK 1.9 also permits contig names here, but most older programs do not.</p></li><li><p>Variant identifier</p></li><li><p>Position in morgans or centimorgans (optional; also safe to use dummy value of ‘0’)</p></li><li><p>Base-pair coordinate</p><p><code>head 1KG.EAS.auto.snp.norm.nodup.split.rare002.common015.missing.map</code></p><div class="code-container" data-rel="Plaintext"><figure class="iseeu highlight plaintext"><figcaption><span>1:13273:G:C 0 13273</span></figcaption><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">1       1:14599:T:A     0       14599</span><br><span class="line">1       1:14604:A:G     0       14604</span><br><span class="line">1       1:14930:A:G     0       14930</span><br><span class="line">1       1:69897:T:C     0       69897</span><br><span class="line">1       1:86331:A:G     0       86331</span><br><span class="line">1       1:91581:G:A     0       91581</span><br><span class="line">1       1:122872:T:G    0       122872</span><br><span class="line">1       1:135163:C:T    0       135163</span><br><span class="line">1       1:233473:C:G    0       233473</span><br></pre></td></tr></table></figure></div></li></ul><p>Reference: <a class="link" target="_blank" rel="noopener" href="https://www.cog-genomics.org/plink/1.9/formats">https://www.cog-genomics.org/plink/1.9/formats<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a></p><h3 id="bed-fam-bim"><a href="#bed-fam-bim" class="headerlink" title="bed &#x2F; fam &#x2F; bim"></a>bed &#x2F; fam &#x2F; bim</h3><ul><li>bed&#x2F;fam&#x2F;bim formats are the <strong>binary</strong> implementation of ped&#x2F;map formats.</li><li>bed&#x2F;bim&#x2F;fam files contain the <strong>same information as</strong> ped&#x2F;map but are much smaller in size.</li></ul><p>example “<code>.fam</code>“</p><p><code>head 1KG.EAS.auto.snp.norm.nodup.split.rare002.common015.missing.fam</code></p><pre><code>0 HG00403 0 0 0 -9
0 HG00404 0 0 0 -9
0 HG00406 0 0 0 -9
0 HG00407 0 0 0 -9
0 HG00409 0 0 0 -9
0 HG00410 0 0 0 -9
0 HG00419 0 0 0 -9
0 HG00421 0 0 0 -9
0 HG00422 0 0 0 -9
0 HG00428 0 0 0 -9
<div class="code-container" data-rel="Plaintext"><figure class="iseeu highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line">example &quot;`.bim`&quot;</span><br><span class="line"></span><br><span class="line">` head 1KG.EAS.auto.snp.norm.nodup.split.rare002.common015.missing.bim`</span><br><span class="line"></span><br><span class="line">    1       1:13273:G:C     0       13273   C       G</span><br><span class="line">    1       1:14599:T:A     0       14599   A       T</span><br><span class="line">    1       1:14604:A:G     0       14604   G       A</span><br><span class="line">    1       1:14930:A:G     0       14930   G       A</span><br><span class="line">    1       1:69897:T:C     0       69897   C       T</span><br><span class="line">    1       1:86331:A:G     0       86331   G       A</span><br><span class="line">    1       1:91581:G:A     0       91581   A       G</span><br><span class="line">    1       1:122872:T:G    0       122872  G       T</span><br><span class="line">    1       1:135163:C:T    0       135163  T       C</span><br><span class="line">    1       1:233473:C:G    0       233473  G       C</span><br></pre></td></tr></table></figure></div>
</code></pre><p>example “<code>.bed</code>“<br>“Primary representation of genotype calls at biallelic variants<br>The first three bytes should be 0x6c, 0x1b, and 0x01 in that order.<br>The rest of the file is a sequence of V blocks of N&#x2F;4 (rounded up) bytes each, where V is the number of variants and N is the number of samples. The first block corresponds to the first marker in the .bim file, etc.”</p><p><code>hexdump -C 1KG.EAS.auto.snp.norm.nodup.split.rare002.common015.missing.bed | head</code></p><div class="code-container" data-rel="Plaintext"><figure class="iseeu highlight plaintext"><figcaption><span>6c 1b 01 ff ff bf bf ff ff ff ef fb ff ff ff fe |l...............|</span></figcaption><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">00000010  ff ff ff ff fb ff bb ff  ff fb af ff ff fe fb ff  |................|</span><br><span class="line">00000020  ff ff ff fe ff ff ff ff  ff bf ff ff ef ff ff ef  |................|</span><br><span class="line">00000030  bb ff ff ff ff ff ff ff  fa ff ff ff ff ff ff ff  |................|</span><br><span class="line">00000040  ff ff ff fb ff ff ff ff  ff ff ff ff ff ff ff ef  |................|</span><br><span class="line">00000050  ff ff ff fb fe ef fe ff  ff ff ff eb ff ff fe fe  |................|</span><br><span class="line">00000060  ff ff fe ff bf ff fa fb  fb eb be ff ff 3b ff be  |.............;..|</span><br><span class="line">00000070  fe be bf ef fe ff ef ee  ff ff bf ea fe bf fe ff  |................|</span><br><span class="line">00000080  bf ff ff ef ff ff ff ff  ff fa ff ff eb ff ff ff  |................|</span><br><span class="line">00000090  ff ff fb fe af ff bf ff  ff ff ff ff ff ff ff ff  |................|```</span><br></pre></td></tr></table></figure></div><div class="note-large default"><div class="notel-title rounded-t-lg p-3 font-bold text-lg flex flex-row gap-2 items-center"><i class="notel-icon fa-solid fa-info"></i><p>Note</p></div><div class="notel-content"><p><code>hexdump</code> 是一个在 Unix&#x2F;Linux 系统中用于<strong>查看文件二进制内容</strong>的命令行工具。它将文件内容以十六进制（hex）、十进制、八进制或 ASCII 格式显示出来，主要用于分析二进制文件、调试数据或检查文件结构。</p></div></div><p>Reference: [<a class="link" target="_blank" rel="noopener" href="https://www.cog-genomics.org/plink/1.9/formats">https://www.cog-genomics.org/plink/1.9/formats<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a></p><h3 id="Imputation-dosage"><a href="#Imputation-dosage" class="headerlink" title="Imputation dosage"></a>Imputation dosage</h3><ul><li><strong>基因型填充 (Imputation)</strong>：利用参考面板（如千人基因组计划）推测未分型或低质量SNP的基因型。</li><li><strong>剂量值 (Dosage)</strong>：表示某个等位基因（通常是次要等位 A1）的期望拷贝数，范围是 <code>[0, 2]</code>（连续值）。<ul><li>例：剂量值&#x3D;1.8 → 该个体携带 A1 的概率很高（接近2个拷贝）。</li></ul></li><li><strong>基因型概率 (Genotype Probabilities)</strong>：指某个位点上三种基因型（<em>A1A1</em>, <em>A1A2</em>, <em>A2A2</em>）的概率（3个值的向量，和为1）。<ul><li>例：<code>(0.01, 0.98, 0.01)</code> → 极大概率是杂合子 (<em>A1A2</em>)。</li></ul></li></ul><hr><ul><li><p>bgen &#x2F; bgi</p><ul><li><p>存储 <strong>基因型概率（3个概率值）</strong> 或 <strong>剂量值</strong>，是填充数据的<strong>黄金标准格式</strong>，支持概率的完整信息。</p></li><li></li></ul><table><thead><tr><th align="left">文件扩展名</th><th align="left">内容</th></tr></thead><tbody><tr><td align="left"><code>.bgen</code></td><td align="left"><strong>主数据文件</strong>（二进制）：存储样本ID、位点信息和基因型概率向量。</td></tr><tr><td align="left"><code>.bgi</code></td><td align="left"><strong>索引文件</strong>（二进制）：加速按染色体&#x2F;位置查询位点，类似数据库索引。</td></tr></tbody></table><ul><li><p><strong>注意</strong>：剂量值可通过概率计算得出（<code>Dosage = 0*P(AA) + 1*P(AB) + 2*P(BB)</code>），但<strong>逆向转换会丢失信息</strong>（剂量值无法还原为原始概率）。</p></li><li><p>Reference: <a class="link" target="_blank" rel="noopener" href="https://www.well.ox.ac.uk/~gav/bgen_format/">https://www.well.ox.ac.uk/~gav&#x2F;bgen_format&#x2F;<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a></p></li></ul></li><li><p>pgen,psam,pvar</p><ul><li><p>PLINK 2.0 的高效二进制格式，<strong>专为快速分析优化</strong>，但仅存储<strong>剂量值</strong>（不存储概率）</p></li><li><table><thead><tr><th align="left">文件扩展名</th><th align="left">内容</th><th align="left">类比PLINK1.0</th></tr></thead><tbody><tr><td align="left"><code>.pgen</code></td><td align="left"><strong>基因型数据</strong>（二进制）：存储剂量值（连续值）。</td><td align="left">替代 <code>.bed</code></td></tr><tr><td align="left"><code>.psam</code></td><td align="left"><strong>样本信息文件</strong>（文本）：存储样本ID、性别、表型等。</td><td align="left">替代 <code>.fam</code></td></tr><tr><td align="left"><code>.pvar</code></td><td align="left"><strong>位点信息文件</strong>（文本）：存储染色体、位置、等位基因等。</td><td align="left">替代 <code>.bim</code></td></tr></tbody></table></li><li><p>Reference: <a class="link" target="_blank" rel="noopener" href="https://www.cog-genomics.org/plink/2.0/formats#pgen">https://www.cog-genomics.org/plink/2.0/formats#pgen<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a></p></li></ul></li></ul><p>NOTE: <code>pgen</code> only saved the dosage for each individual (a scalar ranged from 0 to 2). It could not been converted back to the genotype probability (a vector of length 3) or allele probability (a matrix of dimension 2 x 2) saved in <code>bgen</code>.</p><h3 id="Summary"><a href="#Summary" class="headerlink" title="Summary"></a>Summary</h3><p><img lazyload src="/images/loading.svg" data-src="/img/img-GWAS/9.png"></p><h2 id="PLINK-tutorial"><a href="#PLINK-tutorial" class="headerlink" title="PLINK tutorial"></a>PLINK tutorial</h2><h3 id="Download-PLINK-1-9-2-0-and-then-unzip"><a href="#Download-PLINK-1-9-2-0-and-then-unzip" class="headerlink" title="Download PLINK 1.9 &#x2F; 2.0 and then unzip"></a>Download PLINK 1.9 &#x2F; 2.0 and then unzip</h3><p>Download PLINK1.9 and PLINK2 from the following webpage to the corresponding directories:</p><ul><li>PLINK1.9 : <a class="link" target="_blank" rel="noopener" href="https://www.cog-genomics.org/plink/">https://www.cog-genomics.org/plink/<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a></li><li>PLINK2 : <a class="link" target="_blank" rel="noopener" href="https://www.cog-genomics.org/plink/2.0/">https://www.cog-genomics.org/plink/2.0/<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a></li></ul><h3 id="Add-paths-to-environment-path"><a href="#Add-paths-to-environment-path" class="headerlink" title="Add paths to environment path"></a>Add paths to environment path</h3><p>Then add <code>~/tools/bin/</code> to the environment path.</p><h3 id="Download-the-genotype-data"><a href="#Download-the-genotype-data" class="headerlink" title="Download the genotype data"></a>Download the genotype data</h3><p>Next, we need to download the sample genotype data. The way to create the sample data is described [here].(<a class="link" target="_blank" rel="noopener" href="https://cloufield.github.io/GWASTutorial/01_Dataset/">https://cloufield.github.io/GWASTutorial/01_Dataset/<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a>)<br>This dataset contains 504 EAS individuals from 1000 Genome Project Phase 3v5 with around 1 million variants. (<a class="link" target="_blank" rel="noopener" href="https://www.dropbox.com/scl/fi/41ep8xbdccp9xw5epim19/1KG.EAS.auto.snp.norm.nodup.split.rare002.common015.missing.zip?rlkey=tklapxwypeg79b1sx03o6ycs7&dl=1">https://www.dropbox.com/scl/fi/41ep8xbdccp9xw5epim19/1KG.EAS.auto.snp.norm.nodup.split.rare002.common015.missing.zip?rlkey=tklapxwypeg79b1sx03o6ycs7&amp;dl=1<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a>)</p><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line">&gt; <span class="built_in">head</span> 1KG.EAS.auto.snp.norm.nodup.split.rare002.common015.missing.bim</span><br><span class="line">1	1:14930:A:G	0	14930	G	A</span><br><span class="line">1	1:15774:G:A	0	15774	A	G</span><br><span class="line">1	1:15777:A:G	0	15777	G	A</span><br><span class="line">1	1:57292:C:T	0	57292	T	C</span><br><span class="line">1	1:77874:G:A	0	77874	A	G</span><br><span class="line">1	1:87360:C:T	0	87360	T	C</span><br><span class="line">1	1:92917:T:A	0	92917	A	T</span><br><span class="line">1	1:104186:T:C	0	104186	C	T</span><br><span class="line">1	1:125271:C:T	0	125271	T	C</span><br><span class="line">1	1:232449:G:A	0	232449	A	G</span><br><span class="line">&gt; <span class="built_in">head</span> 1KG.EAS.auto.snp.norm.nodup.split.rare002.common015.missing.fam</span><br><span class="line">HG00403	HG00403	0	0	0	-9</span><br><span class="line">HG00404	HG00404	0	0	0	-9</span><br><span class="line">HG00406	HG00406	0	0	0	-9</span><br><span class="line">HG00407	HG00407	0	0	0	-9</span><br><span class="line">HG00409	HG00409	0	0	0	-9</span><br><span class="line">HG00410	HG00410	0	0	0	-9</span><br><span class="line">HG00419	HG00419	0	0	0	-9</span><br><span class="line">HG00421	HG00421	0	0	0	-9</span><br><span class="line">HG00422	HG00422	0	0	0	-9</span><br><span class="line">HG00428	HG00428	0	0	0	-9</span><br></pre></td></tr></table></figure></div><div class="note-large default"><div class="notel-title rounded-t-lg p-3 font-bold text-lg flex flex-row gap-2 items-center"><i class="notel-icon fa-solid fa-info"></i><p>Note</p></div><div class="notel-content"><p>BIM 文件包含<strong>所有 SNP 标记（遗传变异位点）的信息</strong>，每行对应一个 SNP。它有 6 列：</p><ol><li><strong>染色体编号</strong>：<code>1</code> 表示 1 号染色体</li><li><strong>SNP 标识符</strong>：SNP 的唯一 ID（格式通常是 <code>染色体:位置:等位基因1:等位基因2</code>）</li><li><strong>遗传距离（cM）</strong>：通常为 <code>0</code>（表示未使用）</li><li><strong>物理位置（bp）</strong>：SNP 在染色体上的具体位置</li><li><strong>等位基因 1（次要等位）</strong>：通常是变异频率较低的等位基因</li><li><strong>等位基因 2（主要等位）</strong>：通常是变异频率较高的等位基因</li></ol></div></div><div class="note-large default"><div class="notel-title rounded-t-lg p-3 font-bold text-lg flex flex-row gap-2 items-center"><i class="notel-icon fa-solid fa-info"></i><p>Note</p></div><div class="notel-content"><p>FAM 文件包含<strong>样本（个体）信息</strong>，每行对应一个研究个体。它有 6 列：</p><ol><li><strong>家系 ID</strong>：个体所属家系的标识符<ul><li>例：<code>HG00403</code>（来自 1000 Genomes Project 的样本）</li></ul></li><li><strong>个体 ID</strong>：个体的唯一标识符</li><li><strong>父亲 ID</strong>：<code>0</code> 表示未知（无父亲信息）</li><li><strong>母亲 ID</strong>：<code>0</code> 表示未知（无母亲信息）</li><li><strong>性别</strong>：<code>0</code> 表示未知（或未记录）<ul><li><code>1</code> &#x3D; 男性, <code>2</code> &#x3D; 女性</li></ul></li><li><strong>表型值</strong>：<code>-9</code> 表示缺失<ul><li>通常 <code>1</code> &#x3D; 对照组, <code>2</code> &#x3D; 病例组（如患病）</li></ul></li></ol></div></div><h3 id="QC"><a href="#QC" class="headerlink" title="QC"></a>QC</h3><table><thead><tr><th>QC step</th><th>Option in PLINK</th><th>Commonly used threshold to exclude</th></tr></thead><tbody><tr><td>SNP missing rate</td><td><code>--geno</code>, <code>--missing</code></td><td>missing rate &gt; 0.01 (0.02, or 0.05)</td></tr><tr><td>Sample missing rate</td><td><code>--mind</code>, <code>--missing</code></td><td>missing rate &gt; 0.01 (0.02, or 0.05)</td></tr><tr><td>Minor allele frequency</td><td><code>--freq</code>, <code>--maf</code></td><td>maf &lt; 0.01</td></tr><tr><td>Sample Relatedness</td><td><code>--genome</code></td><td>pi_hat &gt; 0.2 to exclude second-degree relatives</td></tr><tr><td>Hardy-Weinberg equilibrium</td><td><code>--hwe</code>,<code>--hardy</code></td><td>hwe &lt; 1e-6</td></tr><tr><td>Inbreeding F coefficient</td><td><code>--het</code></td><td>outside of 3 SD from the mean</td></tr></tbody></table><h4 id="Missing-rate-call-rate"><a href="#Missing-rate-call-rate" class="headerlink" title="Missing rate (call rate)"></a>Missing rate (call rate)</h4><p>The first thing we want to know is the missing rate of our data. Usually, we need to check the missing rate of samples and SNPs to decide a threshold to exclude low-quality samples and SNPs. (<a class="link" target="_blank" rel="noopener" href="https://www.cog-genomics.org/plink/1.9/basic_stats#missing">https://www.cog-genomics.org/plink/1.9/basic_stats#missing<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a>)</p><p>识别并剔除低质量的样本和SNP是数据预处理中最重要的环节之一。这是确保后续分析结果可靠性的基础。</p><ul><li><strong>Sample missing rate</strong>: the proportion of missing values for an individual across all SNPs.<ul><li>$ Sample\ Missing\ Rate_{ (j) } = {{N_{missing\ SNPs\ for\ j}}\over{M}} = 1 - Call\ Rate_{sample, j} $</li><li>如果一个样本在100万个SNP位点上有50,000个位点没有成功分型（数据缺失），那么该样本的缺失率 &#x3D; 50,000 &#x2F; 1,000,000 &#x3D; 5%，其检出率 &#x3D; 95%。</li></ul></li><li><strong>SNP missing rate</strong>: the proportion of missing values for a SNP across all samples.<ul><li>$ SNP\ Missing\ Rate_{ (i) } = {{N_{missing\ samples\ at\ i}}\over{N}} = 1 - Call\ Rate_{SNP, i} $</li><li>如果一个SNP位点在1000个样本中有30个样本没有成功分型，那么该SNP的缺失率 &#x3D; 30 &#x2F; 1000 &#x3D; 3%，其检出率 &#x3D; 97%。</li></ul></li></ul><div class="code-container" data-rel="Plaintext"><figure class="iseeu highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">plink \</span><br><span class="line">	--bfile $&#123;genotypeFile&#125; \</span><br><span class="line">	--missing \</span><br><span class="line">	--out plink_results</span><br><span class="line">	</span><br><span class="line"># genotypeFile 是bed,bim,fam文件的前缀名字</span><br></pre></td></tr></table></figure></div><p>This code will generate two files <code>plink_results.imiss</code> and <code>plink_results.lmiss</code>, which contain the missing rate information for samples and SNPs respectively.</p><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># missing rate for each sample</span></span><br><span class="line"><span class="built_in">head</span> plink_results.imiss</span><br><span class="line">    FID       IID MISS_PHENO   N_MISS   N_GENO   F_MISS</span><br><span class="line">HG00403   HG00403          Y    10020  1235116 0.008113</span><br><span class="line">HG00404   HG00404          Y     9192  1235116 0.007442</span><br><span class="line">HG00406   HG00406          Y    15751  1235116  0.01275</span><br><span class="line">HG00407   HG00407          Y    14653  1235116  0.01186</span><br><span class="line">HG00409   HG00409          Y     5667  1235116 0.004588</span><br><span class="line">HG00410   HG00410          Y     6066  1235116 0.004911</span><br><span class="line">HG00419   HG00419          Y    20000  1235116  0.01619</span><br><span class="line">HG00421   HG00421          Y    17542  1235116   0.0142</span><br><span class="line">HG00422   HG00422          Y    18608  1235116  0.01507</span><br></pre></td></tr></table></figure></div><ul><li>FID (Family ID)：样本所属家族的标识符</li><li>IID (Individual ID)：样本个体的唯一标识符</li><li>MISS_PHENO：表型数据是否缺失</li><li>Y：表型缺失</li><li>N：表型存在</li><li>N_MISS：该样本缺失的SNP数量</li><li>N_GENO：该样本理论上应检测的SNP总数</li><li>F_MISS：样本缺失率 &#x3D; N_MISS &#x2F; N_GENO（最重要的指标）</li></ul><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># missing rate for each SNP</span></span><br><span class="line"><span class="built_in">head</span> plink_results.lmiss</span><br><span class="line"> CHR              SNP   N_MISS   N_GENO   F_MISS</span><br><span class="line">   1      1:14930:A:G        2      504 0.003968</span><br><span class="line">   1      1:15774:G:A        3      504 0.005952</span><br><span class="line">   1      1:15777:A:G        3      504 0.005952</span><br><span class="line">   1      1:57292:C:T        6      504   0.0119</span><br><span class="line">   1      1:77874:G:A        3      504 0.005952</span><br><span class="line">   1      1:87360:C:T        1      504 0.001984</span><br><span class="line">   1      1:92917:T:A        7      504  0.01389</span><br><span class="line">   1     1:104186:T:C        3      504 0.005952</span><br><span class="line">   1     1:125271:C:T        2      504 0.003968</span><br></pre></td></tr></table></figure></div><ul><li><strong>SNP 位置</strong>：1 号染色体 14930 位置（A→G 突变）</li><li><strong>缺失情况</strong>：在 504 次检测中（约 252 个样本 ×2），有 2 次未成功分型</li><li><strong>缺失率</strong>：0.003968 ≈ <strong>0.4%</strong>（质量极好）</li></ul><h4 id="Allele-Frequency"><a href="#Allele-Frequency" class="headerlink" title="Allele Frequency"></a>Allele Frequency</h4><p>One of the most important statistics of SNPs is their frequency in a certain population. Many downstream analyses are based on investigating differences in allele frequencies.</p><p>Usually, variants can be categorized into 3 groups based on their Minor Allele Frequency (MAF):</p><ol><li><strong>Common variants</strong> : MAF&gt;&#x3D;0.05</li><li><strong>Low-frequency variants</strong> : 0.01&lt;&#x3D;MAF&lt;0.05</li><li><strong>Rare variants</strong> : MAF&lt;0.01</li></ol><p>Suppose the reference allele(REF) is A and the alternative allele(ALT) is B for a certain SNP. The posible genotypes are AA, AB and BB. In a population of N samples (2N alleles), $N = N_{AA} + 2 \times N_{AB} + N_{BB}$ :</p><ul><li>the number of A alleles: $N_A = 2 \times N_{AA} + N_{AB}$</li><li>the number of B alleles: $N_B = 2 \times N_{BB} + N_{AB}$</li></ul><p>So we can calculate the allele frequency:</p><ul><li>Reference Allele Frequency : AF_{REF}= {{N_A}\over{N_A + N_B}}$</li><li>Alternative Allele Frequency : $AF_{ALT}= {{N_B}\over{N_A + N_B}}$</li></ul><p>The MAF for this SNP in this specific population is defined as:</p>MAF = min( AF_{REF}, AF_{ALT} )$<div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">plink2 \</span><br><span class="line">	--bfile <span class="variable">$&#123;genotypeFile&#125;</span> \</span><br><span class="line">	--freq \</span><br><span class="line">	--out plink_results</span><br></pre></td></tr></table></figure></div><div class="code-container" data-rel="Plaintext"><figure class="iseeu highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">&gt; head plink_results.afreq</span><br><span class="line">#CHROM	ID	REF	ALT	PROVISIONAL_REF?	ALT_FREQS	OBS_CT</span><br><span class="line">1	1:14930:A:G	A	G	Y	0.413347	1004</span><br><span class="line">1	1:15774:G:A	G	A	Y	0.0279441	1002</span><br><span class="line">1	1:15777:A:G	A	G	Y	0.0738523	1002</span><br><span class="line">1	1:57292:C:T	C	T	Y	0.105422	996</span><br><span class="line">1	1:77874:G:A	G	A	Y	0.0199601	1002</span><br><span class="line">1	1:87360:C:T	C	T	Y	0.0228628	1006</span><br><span class="line">1	1:92917:T:A	T	A	Y	0.00301811	994</span><br><span class="line">1	1:104186:T:C	T	C	Y	0.500998	1002</span><br><span class="line">1	1:125271:C:T	C	T	Y	0.969124	1004</span><br></pre></td></tr></table></figure></div><ol><li><strong>#CHROM</strong>：染色体编号（如 1 表示 1 号染色体）</li><li><strong>ID</strong>：SNP 的唯一标识符（通常格式为 <code>染色体:位置:参考等位基因:变异等位基因</code>）</li><li><strong>REF</strong>：参考等位基因（基因组参考序列中的碱基）</li><li><strong>ALT</strong>：变异等位基因（样本中观察到的替代碱基）</li><li><strong>PROVISIONAL_REF?</strong>：参考等位基因是否临时指定<ul><li><code>Y</code>：是（PLINK 自动确定）</li><li><code>N</code>：否（来自原始数据）</li></ul></li><li><strong>ALT_FREQS</strong>：<strong>变异等位基因(ALT)的频率</strong>（范围 0-1，最重要的统计量）</li><li><strong>OBS_CT</strong>：用于计算频率的观察样本数（样本数×2，因每个样本有两条染色体）</li></ol><p>In PLINK1.9, the concept here is minor (A1) and major(A2) allele, while in PLINK2 it is the reference (REF) allele and the alternative (ALT) allele.</p><ul><li><strong>Major &#x2F; Minor</strong>: Major allele and minor allele are defined as the allele with the highest and lower(or the second highest for multiallelic variants) allele in a given population, respectively. So major and minor alleles for a certain SNP might be different in two independent populations. The range for MAF(minor allele frequencies) is [0,0.5].</li><li><strong>Ref &#x2F; Alt</strong>: The reference (REF) and alternative (ALT) alleles are simply determined by the allele on a reference genome. If we use the same reference genome, the reference(REF) and alternative(ALT) alleles will be the same across populations. The reference allele could be major or minor in different populations. The range for alternative allele frequency is [0,1], since it could be the major allele or the minor allele in a given population.</li></ul><h4 id="Hardy-Weinberg-equilibrium-exact-test"><a href="#Hardy-Weinberg-equilibrium-exact-test" class="headerlink" title="Hardy-Weinberg equilibrium exact test"></a>Hardy-Weinberg equilibrium exact test</h4><p>For SNP QC, besides checking the missing rate, we also need to check if the SNP is in Hardy-Weinberg equilibrium:</p><p><code>--hardy</code> will perform Hardy-Weinberg equilibrium exact test for each variant. Variants with low P value usually suggest genotyping errors, or indicate evolutionary selection for these variants.</p><p>The following command can calculate the Hardy-Weinberg equilibrium exact test statistics for all SNPs. (<a class="link" target="_blank" rel="noopener" href="https://www.cog-genomics.org/plink/1.9/basic_stats#hardy">https://www.cog-genomics.org/plink/1.9/basic_stats#hardy<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a>)</p>P(N_{AB} = n_{AB} | N, n_A) = {{2^{n_{AB}}}N!\over{n_{AA}!n_{AB}!n_{BB}!}} \times {{n_A!n_B!}\over{(2N)!}} $<p>To compute the Hardy-Weinberg equilibrium exact test statistics, we will sum up the probabilities of all configurations with probability equal to or less than the observed configuration :</p>P_{HWE} = \sum_{n^{*}_AB} I[P(N_{AB} = n_{AB} | N, n_A) \geqq P(N_{AB} = n^{*}_{AB} | N, n_A)] \times P(N_{AB} = n^{*}_{AB} | N, n_A) $<p>$I(x)$ is the indicator function. If x is true, $I(x) &#x3D; 1$; otherwise, $I(x) &#x3D; 0$.</p><p>Reference : Wigginton, J. E., Cutler, D. J., &amp; Abecasis, G. R. (2005). A note on exact tests of Hardy-Weinberg equilibrium. The American Journal of Human Genetics, 76(5), 887-893. <a class="link" target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1199378/">Link<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a></p><p>Calculate the Hardy-Weinberg equilibrium exact test statistics for a single SNP using <strong>Python</strong></p><p>This code is converted from <a class="link" target="_blank" rel="noopener" href="https://github.com/jeremymcrae/snphwe/blob/master/src/snp_hwe.cpp">here<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a> (Jeremy McRae) to python. Orginal citation: Wigginton, JE, Cutler, DJ, and Abecasis, GR (2005) A Note on Exact Tests of Hardy-Weinberg Equilibrium. AJHG 76: 887-893</p><div class="code-container" data-rel="Python"><figure class="iseeu highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">def</span> <span class="title function_">snphwe</span>(<span class="params">obs_hets: <span class="built_in">int</span>, obs_hom1: <span class="built_in">int</span>, obs_hom2: <span class="built_in">int</span></span>) -&gt; <span class="built_in">float</span>:</span><br><span class="line"><span class="string">&quot;&quot;&quot;Calculate Hardy-Weinberg equilibrium exact test for a variant</span></span><br><span class="line"><span class="string"></span></span><br><span class="line"><span class="string">Args:</span></span><br><span class="line"><span class="string">    obs_hets (int): observed heterozygous number</span></span><br><span class="line"><span class="string">    obs_hom1 (int): first observed homozygous number</span></span><br><span class="line"><span class="string">    obs_hom2 (int): second observed homozygous number</span></span><br><span class="line"><span class="string"></span></span><br><span class="line"><span class="string">Returns:</span></span><br><span class="line"><span class="string">    float: Hardy-Weinberg equilibrium exact test p-value.</span></span><br><span class="line"><span class="string">    </span></span><br><span class="line"><span class="string">    - P &gt; 0.05: No significant deviation from Hardy-Weinberg equilibrium (HWE).</span></span><br><span class="line"><span class="string">    - P ≤ 0.05: Significant deviation from HWE, which could be due to factors such as population stratification, inbreeding, genotyping errors, or natural selection.</span></span><br><span class="line"><span class="string">&quot;&quot;&quot;</span></span><br><span class="line">obs_minor_homs = <span class="built_in">min</span>(obs_hom1, obs_hom2)</span><br><span class="line">obs_mijor_homs = <span class="built_in">max</span>(obs_hom1, obs_hom2)</span><br><span class="line"></span><br><span class="line">rare = <span class="number">2</span> * obs_minor_homs + obs_hets</span><br><span class="line">genotypes = obs_hets + obs_mijor_homs + obs_minor_homs</span><br><span class="line"></span><br><span class="line">probs = [<span class="number">0.0</span> <span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">range</span>(rare +<span class="number">1</span>)]</span><br><span class="line"></span><br><span class="line">mid = rare * (<span class="number">2</span> * genotypes - rare) // (<span class="number">2</span> * genotypes)</span><br><span class="line"><span class="keyword">if</span> mid % <span class="number">2</span> != rare%<span class="number">2</span>:</span><br><span class="line">    mid += <span class="number">1</span></span><br><span class="line"></span><br><span class="line">probs[mid] = <span class="number">1.0</span></span><br><span class="line">sum_p = <span class="number">1</span> <span class="comment">#probs[mid]</span></span><br><span class="line"></span><br><span class="line">curr_homr = (rare - mid) // <span class="number">2</span></span><br><span class="line">curr_homc = genotypes - mid - curr_homr</span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> curr_hets <span class="keyword">in</span> <span class="built_in">range</span>(mid, <span class="number">1</span>, -<span class="number">2</span>):</span><br><span class="line">    probs[curr_hets - <span class="number">2</span>] = probs[curr_hets] * curr_hets * (curr_hets - <span class="number">1.0</span>)/ (<span class="number">4.0</span> * (curr_homr + <span class="number">1.0</span>) * (curr_homc + <span class="number">1.0</span>))</span><br><span class="line">    sum_p+= probs[curr_hets - <span class="number">2</span>]</span><br><span class="line">    curr_homr += <span class="number">1</span></span><br><span class="line">    curr_homc += <span class="number">1</span></span><br><span class="line"></span><br><span class="line">curr_homr = (rare - mid) // <span class="number">2</span></span><br><span class="line">curr_homc = genotypes - mid - curr_homr</span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> curr_hets <span class="keyword">in</span> <span class="built_in">range</span>(mid, rare-<span class="number">1</span>, <span class="number">2</span>):</span><br><span class="line">    probs[curr_hets + <span class="number">2</span>] = probs[curr_hets] * <span class="number">4.0</span> * curr_homr * curr_homc/ ((curr_hets + <span class="number">2.0</span>) * (curr_hets + <span class="number">1.0</span>))</span><br><span class="line">    sum_p += probs[curr_hets + <span class="number">2</span>]</span><br><span class="line">    curr_homr -= <span class="number">1</span></span><br><span class="line">    curr_homc -= <span class="number">1</span></span><br><span class="line"></span><br><span class="line">target = probs[obs_hets]</span><br><span class="line">p_hwe = <span class="number">0.0</span></span><br><span class="line"><span class="keyword">for</span> p <span class="keyword">in</span> probs:</span><br><span class="line">    <span class="keyword">if</span> p &lt;= target :</span><br><span class="line">        p_hwe += p / sum_p  </span><br><span class="line"></span><br><span class="line"><span class="keyword">return</span> <span class="built_in">min</span>(p_hwe,<span class="number">1</span>)</span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> __name__ == <span class="string">&#x27;__main__&#x27;</span>:</span><br><span class="line"><span class="comment"># For an example, we had 502 samples. </span></span><br><span class="line"><span class="comment"># At position 1:14930:A:G, we count:</span></span><br><span class="line"><span class="comment">#     + 407 heterozygous genotypes (signed 0|1 or 1|0).</span></span><br><span class="line"><span class="comment">#     + 4 homozygous genotypes AA (signed 0|0)</span></span><br><span class="line"><span class="comment">#     + 91 homozygous genotypes GG (signed 1|1)</span></span><br><span class="line"><span class="built_in">print</span>(snphwe(<span class="number">407</span>,<span class="number">4</span>,<span class="number">91</span>))</span><br></pre></td></tr></table></figure></div><p>Calculate the Hardy-Weinberg equilibrium exact test statistics using <strong>PLINK</strong></p><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">plink2 \</span><br><span class="line">	--bfile <span class="variable">$&#123;genotypeFile&#125;</span> \</span><br><span class="line">	--hardy \</span><br><span class="line">	--out plink_results </span><br></pre></td></tr></table></figure></div><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># head plink_results.hwe</span></span><br><span class="line"><span class="comment">#CHROM	ID	A1	AX	HOM_A1_CT	HET_A1_CT	TWO_AX_CT	O(HET_A1)	E(HET_A1)	P</span></span><br><span class="line">1	1:14930:A:G	A	G	91	407	4	0.810757	0.484982	4.86444e-61</span><br><span class="line">1	1:15774:G:A	G	A	473	28	0	0.0558882	0.0543265	1</span><br><span class="line">1	1:15777:A:G	A	G	428	72	1	0.143713	0.136796	0.505316</span><br><span class="line">1	1:57292:C:T	C	T	396	99	3	0.198795	0.188616	0.339285</span><br><span class="line">1	1:77874:G:A	G	A	481	20	0	0.0399202	0.0391234	1</span><br><span class="line">1	1:87360:C:T	C	T	480	23	0	0.0457256	0.0446802	1</span><br><span class="line">1	1:92917:T:A	T	A	494	3	0	0.00603622	0.006018	1</span><br><span class="line">1	1:104186:T:C	T	C	74	352	75	0.702595	0.499998	6.41791e-20</span><br><span class="line">1	1:125271:C:T	C	T	1	29	472	0.0577689	0.0598463	0.379814</span><br></pre></td></tr></table></figure></div><ol><li><strong>CHR</strong>：染色体编号，这里是第1号染色体。</li><li><strong>SNP</strong>：单核苷酸多态性（SNP）的标识符，格式为<code>染色体:位置:参考等位基因:变异等位基因</code>。例如，<code>1:14930:A:G</code>表示在第1号染色体的14930位置，参考等位基因是A，变异等位基因是G。</li><li><strong>TEST</strong>：检验类型，<code>ALL(NP)</code>表示对所有个体（非缺失）进行检验。</li><li><strong>A1</strong>：等位基因1（通常为次要等位基因或关注等位基因）。</li><li><strong>A2</strong>：等位基因2（通常为主要等位基因）。</li><li><strong>GENO</strong>：基因型计数，格式为<code>纯合A1/杂合/纯合A2</code>的个体数。例如，<code>4/407/91</code>表示有4个纯合A1（GG）、407个杂合（GA）和91个纯合A2（AA）个体。</li><li><strong>O(HET)</strong>：观察到的杂合子频率，计算为杂合子数除以总样本数。例如，<code>407 / (4 + 407 + 91) ≈ 0.8108</code>。</li><li><strong>E(HET)</strong>：期望的杂合子频率，根据哈迪-温伯格平衡公式计算（<code>2 * p * q</code>，其中p和q是等位基因频率）。</li><li><strong>P</strong>：HWE检验的p值，表示观察到的基因型频率与期望频率的偏离程度。例如，<code>4.864e-61</code>表示极显著偏离HWE，而<code>1</code>或<code>0.5053</code>表示不显著偏离。</li></ol><h4 id="Applying-filters"><a href="#Applying-filters" class="headerlink" title="Applying filters"></a>Applying filters</h4><p>Previously we calculated the basic statistics using PLINK. But when performing certain analyses, we just want to <strong>exclude the bad-quality samples or SNPs</strong> instead of calculating the statistics for all samples and SNPs.</p><p>In this case we can apply the following filters for example:</p><ul><li><code>--maf 0.01</code> : exlcude snps with maf&lt;0.01</li><li><code>--geno 0.02</code> :filters out all variants with missing rates exceeding 0.02</li><li><code>--mind 0.02</code> :filters out all samples with missing rates exceeding 0.02</li><li><code>--hwe 1e-6</code> : filters out all variants which have Hardy-Weinberg equilibrium exact test p-value below the provided threshold. NOTE: With case&#x2F;control data, cases and missing phenotypes are normally ignored. (see <a class="link" target="_blank" rel="noopener" href="https://www.cog-genomics.org/plink/1.9/filter#hwe">https://www.cog-genomics.org/plink/1.9/filter#hwe<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a>)</li></ul><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">./plink2 </span><br><span class="line">	--bfile <span class="variable">$&#123;genotypeFile&#125;</span> \</span><br><span class="line">	--maf 0.01 \</span><br><span class="line">	--geno 0.02 \</span><br><span class="line">	--mind 0.02 \</span><br><span class="line">	--hwe 1e-6 \</span><br><span class="line">	--make-bed \</span><br><span class="line">	--out qc_filtered_data</span><br></pre></td></tr></table></figure></div><p>This command generates five outputs: <code>qc_filtered_data.bed</code> , <code>qc_filtered_data.bim</code>, <code>qc_filtered_data.fam</code>, <code>qc_filtered_data.log</code>, <code>qc_filtered_data.mindrem.id</code></p><h4 id="LD-Pruning"><a href="#LD-Pruning" class="headerlink" title="LD Pruning"></a>LD Pruning</h4><p>There is often strong Linkage disequilibrium(LD) among SNPs, for some analysis we don’t need all SNPs and we need to remove the redundant SNPs to <strong>avoid bias in genetic estimations</strong>. For example, for relatedness estimation, we will use only LD-Pruned SNP set.</p><p>We can use <code>--indep-pairwise 50 5 0.2</code> to filter out those in strong LD and keep only the independent SNPs. （<strong>shift the window</strong>）</p><ul><li>consider a window of <code>50</code> SNPs</li><li>calculate LD between each pair of SNPs in the window and remove one of a pair of SNPs if the LD is greater than <code>5</code></li><li>shift the window <code>0.2</code> SNPs forward and repeat the procedure.</li></ul><p>Please check <a class="link" target="_blank" rel="noopener" href="https://www.cog-genomics.org/plink/1.9/ld#indep">https://www.cog-genomics.org/plink/1.9/ld#indep<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a> for details.</p><p><strong>连锁不平衡（Linkage Disequilibrium, LD）</strong> 指的是基因组中相邻SNP之间的非随机关联性（即某些SNP的等位基因倾向于共同遗传，而非独立分离）。<br>当SNP之间存在强LD时，它们提供的遗传信息是高度冗余的。如果直接使用所有SNP进行分析，可能会导致以下问题：</p><ol><li><strong>统计偏差</strong>：冗余SNP会过度代表某些基因组区域，影响关联分析的权重。</li><li><strong>多重检验问题</strong>：增加不必要的检验次数，降低统计功效。</li><li><strong>计算负担</strong>：尤其是大型GWAS或机器学习中，冗余数据会显著增加计算成本。</li></ol><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">plink2 \</span><br><span class="line">	--bfile <span class="variable">$&#123;genotypeFile&#125;</span> \</span><br><span class="line">	--maf 0.01 \</span><br><span class="line">	--geno 0.02 \</span><br><span class="line">	--mind 0.02 \</span><br><span class="line">	--hwe 1e-6 \</span><br><span class="line">	--indep-pairwise 50 5 0.2 \</span><br><span class="line">	--out plink_results</span><br></pre></td></tr></table></figure></div><p>This command generates two outputs: <code>plink_results.prune.in</code> and <code>plink_results.prune.out</code><br><code>plink_results.prune.in</code> is the independent set of SNPs we will use in the following analysis.</p><h4 id="Inbreeding-F-coefficient"><a href="#Inbreeding-F-coefficient" class="headerlink" title="Inbreeding F coefficient"></a>Inbreeding F coefficient</h4><p>Next, we can check the heterozygosity F of samples (<a class="link" target="_blank" rel="noopener" href="https://www.cog-genomics.org/plink/1.9/basic_stats#ibc">https://www.cog-genomics.org/plink/1.9/basic_stats#ibc<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a>) :</p><p><code>-het</code> option will compute observed and expected autosomal homozygous genotype counts for each sample. Usually, we need to exclude individuals with high or low heterozygosity coefficients, which suggests that the sample might be contaminated.</p>F = {{O(HOM) - E(HOM)}\over{ M - E(HOM)}}$<ul><li>$E(HOM)$ :Expected Homozygous Genotype Count</li><li>$O(HOM)$ :Observed Homozygous Genotype Count</li><li>M : Number of SNPs</li></ul><p>High F may indicate a relatively high level of <strong>inbreeding</strong>.</p><p>Low F may suggest the sample DNA was contaminated. &#x2F; Complete hybridization</p><div class="note-large default"><div class="notel-title rounded-t-lg p-3 font-bold text-lg flex flex-row gap-2 items-center"><i class="notel-icon fa-solid fa-info"></i><p>warning</p></div><div class="notel-content"><p>Performing LD-pruning beforehand since these calculations do not take LD into account.</p></div></div><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">plink2 \</span><br><span class="line">	--bfile <span class="variable">$&#123;genotypeFile&#125;</span> \</span><br><span class="line">  --extract plink_results.prune.in \</span><br><span class="line">	--het \</span><br><span class="line">	--out plink_results</span><br></pre></td></tr></table></figure></div><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">head</span> plink_results.het</span><br><span class="line"></span><br><span class="line">    FID       IID       O(HOM)       E(HOM)        N(NM)            F</span><br><span class="line">HG00403   HG00403       180222    1.796e+05       217363      0.01698</span><br><span class="line">HG00404   HG00404       180127    1.797e+05       217553      0.01023</span><br><span class="line">HG00406   HG00406       178891    1.789e+05       216533   -0.0001138</span><br><span class="line">HG00407   HG00407       178992     1.79e+05       216677   -0.0008034</span><br><span class="line">HG00409   HG00409       179918    1.801e+05       218045    -0.006049</span><br><span class="line">HG00410   HG00410       179782    1.801e+05       218028    -0.009268</span><br><span class="line">HG00419   HG00419       178362    1.783e+05       215849     0.001315</span><br><span class="line">HG00421   HG00421       178222    1.785e+05       216110    -0.008288</span><br><span class="line">HG00422   HG00422       178316    1.784e+05       215938      -0.0022</span><br></pre></td></tr></table></figure></div><p>A commonly used method is to exclude samples with heterozygosity F deviating more than 3 standard deviations (SD) from the mean. Some studies used a fixed value such as +-0.15 or +-0.2.</p><div class="note-large default"><div class="notel-title rounded-t-lg p-3 font-bold text-lg flex flex-row gap-2 items-center"><i class="notel-icon fa-solid fa-info"></i><p>warning</p></div><div class="notel-content"><p>Usually we will use only <a href="#ld-pruning">LD-pruned SNPs</a> for the calculation of F.</p></div></div><p>Here we use +-0.1 as the $F_{het}$ threshold for convenience.</p><ul><li><strong>F ≈ 0</strong>：正常样本（符合随机交配预期）。</li><li><strong>F &gt; 0</strong>：可能近亲繁殖。</li><li><strong>F &lt; 0</strong>：可能杂合性过高（需检查是否污染或混样）。</li><li><strong>排除极端 F 值样本</strong> 可提高后续分析的可靠性。</li></ul><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># only one sample</span></span><br><span class="line">awk <span class="string">&#x27;NR&gt;1 &amp;&amp; $6&gt;0.1 || $6&lt;-0.1 &#123;print $1,$2&#125;&#x27;</span> plink_results.het &gt; high_het.sample</span><br></pre></td></tr></table></figure></div><h4 id="Sample-SNP-filtering-extract-exclude-keep-remove"><a href="#Sample-SNP-filtering-extract-exclude-keep-remove" class="headerlink" title="Sample &amp; SNP filtering (extract&#x2F;exclude&#x2F;keep&#x2F;remove)"></a>Sample &amp; SNP filtering (extract&#x2F;exclude&#x2F;keep&#x2F;remove)</h4><p>Sometimes we will use only a subset of samples or SNPs included the original dataset.<br>In this case, we can use <code>--extract</code> or <code>--exclude</code> to select or exclude SNPs from analysis, <code>--keep</code> or <code>--remove</code> to select or exclude samples.</p><p>For <code>--keep</code> or <code>--remove</code>, the input is the filename of a sample FID and IID file.<br>For <code>--extract</code> or <code>--exclude</code>, the input is the filename of an SNP list file.</p><table><thead><tr><th align="left">选项</th><th align="left">用途</th><th align="left">输入文件格式</th></tr></thead><tbody><tr><td align="left"><code>--keep &lt;file&gt;</code></td><td align="left"><strong>保留</strong>指定的样本</td><td align="left">两列：<code>FID IID</code>（无表头）</td></tr><tr><td align="left"><code>--remove &lt;file&gt;</code></td><td align="left"><strong>排除</strong>指定的样本</td><td align="left">两列：<code>FID IID</code>（无表头）</td></tr><tr><td align="left"><code>--extract &lt;file&gt;</code></td><td align="left"><strong>保留</strong>指定的 SNP</td><td align="left">一列：<code>SNP_ID</code>（无表头）</td></tr><tr><td align="left"><code>--exclude &lt;file&gt;</code></td><td align="left"><strong>排除</strong>指定的 SNP</td><td align="left">一列：<code>SNP_ID</code>（无表头）</td></tr></tbody></table><p>example 筛选出snp pruning中的数据</p><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">head</span> plink_results.prune.in</span><br><span class="line"></span><br><span class="line">1:15774:G:A</span><br><span class="line">1:15777:A:G</span><br><span class="line">1:77874:G:A</span><br><span class="line">1:87360:C:T</span><br><span class="line">1:125271:C:T</span><br><span class="line">1:232449:G:A</span><br><span class="line">1:533113:A:G</span><br><span class="line">1:565697:A:G</span><br><span class="line">1:566933:A:G</span><br><span class="line">1:567092:T:C</span><br><span class="line"></span><br><span class="line">./plink2 --bfile 1KG.EAS.auto.snp.norm.nodup.split.rare002.common015.missing --extract qc_filtered_data.prune.in  --make-bed --out mydata_snps_filtered</span><br></pre></td></tr></table></figure></div><p>This command generates four outputs: <code>mydata_snps_filtered.bed</code> , <code>mydata_snps_filtered.bim</code>, <code>mydata_snps_filtered.fam</code> and <code>mydata_snps_filtered.log</code>.</p><h4 id="IBD-PI-HAT-kinship-coefficient"><a href="#IBD-PI-HAT-kinship-coefficient" class="headerlink" title="IBD &#x2F; PI_HAT &#x2F; kinship coefficient"></a>IBD &#x2F; PI_HAT &#x2F; kinship coefficient</h4><p><code>--genome</code> will estimate IBS&#x2F;IBD. Usually, for this analysis, we need to prune our data first since the strong LD will cause bias in the results.<br><strong>(This step is computationally intensive)</strong></p><p>Combined with the <code>--extract</code>, we can run.</p><ul><li><strong>IBD（Identity by Descent，同源血统）</strong><ul><li><strong>定义</strong>：指两个个体在某位点上拥有<strong>共同祖先</strong>遗传的相同等位基因（非独立突变产生）。</li></ul></li><li><strong>PI_HAT（π，IBD比例估计值）</strong><ul><li>PLINK中定义的亲缘关系系数，表示两个个体基因组中<strong>IBD比例的平均值</strong>，范围<code>[0,1]</code>。</li></ul></li><li><strong>Kinship Coefficient（亲缘系数，ϕ）</strong><ul><li><strong>定义</strong>：两个个体随机选取的等位基因<strong>IBD的概率</strong>，范围<code>[0, 0.5]</code>。</li></ul></li></ul><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">plink1.9 \</span><br><span class="line">	--bfile <span class="variable">$&#123;genotypeFile&#125;</span> \</span><br><span class="line">	--extract plink_results.prune.in \</span><br><span class="line">	--genome \</span><br><span class="line">	--out plink_results</span><br></pre></td></tr></table></figure></div><p>PI_HAT is the IBD estimation. Please check <a class="link" target="_blank" rel="noopener" href="https://www.cog-genomics.org/plink/1.9/ibd">https://www.cog-genomics.org/plink/1.9/ibd<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a> for more details.</p><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">head</span> plink_results.genome</span><br><span class="line"></span><br><span class="line">    FID1     IID1     FID2     IID2 RT    EZ      Z0      Z1      Z2  PI_HAT PHE       DST     PPC   RATIO</span><br><span class="line">HG00403  HG00403  HG00404  HG00404 UN    NA  1.0000  0.0000  0.0000  0.0000  -1  0.858562  0.3679  1.9774</span><br><span class="line">HG00403  HG00403  HG00406  HG00406 UN    NA  0.9805  0.0044  0.0151  0.0173  -1  0.858324  0.8183  2.0625</span><br><span class="line">HG00403  HG00403  HG00407  HG00407 UN    NA  0.9790  0.0000  0.0210  0.0210  -1  0.857794  0.8034  2.0587</span><br><span class="line">HG00403  HG00403  HG00409  HG00409 UN    NA  0.9912  0.0000  0.0088  0.0088  -1  0.857024  0.2637  1.9578</span><br><span class="line">HG00403  HG00403  HG00410  HG00410 UN    NA  0.9699  0.0235  0.0066  0.0184  -1  0.858194  0.6889  2.0335</span><br><span class="line">HG00403  HG00403  HG00419  HG00419 UN    NA  1.0000  0.0000  0.0000  0.0000  -1  0.857643  0.8597  2.0745</span><br><span class="line">HG00403  HG00403  HG00421  HG00421 UN    NA  0.9773  0.0218  0.0010  0.0118  -1  0.857276  0.2186  1.9484</span><br><span class="line">HG00403  HG00403  HG00422  HG00422 UN    NA  0.9880  0.0000  0.0120  0.0120  -1  0.857224  0.8277  2.0652</span><br><span class="line">HG00403  HG00403  HG00428  HG00428 UN    NA  0.9801  0.0069  0.0130  0.0164  -1  0.858162  0.9812  2.1471</span><br></pre></td></tr></table></figure></div><table><thead><tr><th align="left">列名</th><th align="left">含义</th><th align="left">解释</th></tr></thead><tbody><tr><td align="left"><strong>FID1, IID1</strong></td><td align="left">个体1的家系ID和个体ID</td><td align="left">用于标识第一个样本</td></tr><tr><td align="left"><strong>FID2, IID2</strong></td><td align="left">个体2的家系ID和个体ID</td><td align="left">用于标识第二个样本</td></tr><tr><td align="left"><strong>RT</strong> (Relationship Type)</td><td align="left">关系类型</td><td align="left"><code>UN</code> &#x3D; Unknown（未知，需依赖基因型推断）</td></tr><tr><td align="left"><strong>EZ</strong> (Error Flag)</td><td align="left">错误标记</td><td align="left"><code>NA</code> &#x3D; 无错误</td></tr><tr><td align="left"><strong>Z0</strong></td><td align="left">IBD0 比例</td><td align="left">两个个体<strong>无共享等位基因</strong>的比例</td></tr><tr><td align="left"><strong>Z1</strong></td><td align="left">IBD1 比例</td><td align="left">两个个体<strong>共享1个等位基因</strong>的比例</td></tr><tr><td align="left"><strong>Z2</strong></td><td align="left">IBD2 比例</td><td align="left">两个个体<strong>共享2个等位基因</strong>的比例</td></tr><tr><td align="left"><strong>PI_HAT</strong></td><td align="left">亲缘关系系数</td><td align="left"><code>PI_HAT = (Z1/2) + Z2</code>，范围 <code>[0, 1]</code></td></tr><tr><td align="left"><strong>PHE</strong> (Phenotype)</td><td align="left">表型信息</td><td align="left"><code>-1</code> &#x3D; 缺失</td></tr><tr><td align="left"><strong>DST, PPC, RATIO</strong></td><td align="left">辅助统计量</td><td align="left">通常可忽略</td></tr></tbody></table><div class="note-large default"><div class="notel-title rounded-t-lg p-3 font-bold text-lg flex flex-row gap-2 items-center"><i class="notel-icon fa-solid fa-info"></i><p>warning</p></div><div class="notel-content"><p>PLINK2 uses KING-robust kinship estimator, which is more robust in the presence of population substructure. See <a class="link" target="_blank" rel="noopener" href="https://www.cog-genomics.org/plink/2.0/distance#make_king">here<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a>.</p><p>Manichaikul, A., Mychaleckyj, J. C., Rich, S. S., Daly, K., Sale, M., &amp; Chen, W. M. (2010). Robust relationship inference in genome-wide association studies. Bioinformatics, 26(22), 2867-2873.</p></div></div><h4 id="LD-calculation"><a href="#LD-calculation" class="headerlink" title="LD calculation"></a>LD calculation</h4><p>We can also use our data to estimate the LD between a pair of SNPs.</p><p>Details on LD can be found <a class="link" target="_blank" rel="noopener" href="https://cloufield.github.io/GWASTutorial/19_ld/">here<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a></p><p><code>--chr</code> option in PLINK allows us to include SNPs on a specific chromosome.<br>To calculate LD r2 for SNPs on chr22 , we can run:</p><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">plink2 \</span><br><span class="line">	--bfile <span class="variable">$&#123;genotypeFile&#125;</span> \</span><br><span class="line">	--chr 22 \</span><br><span class="line">	--r2 \</span><br><span class="line">	--out plink_results</span><br></pre></td></tr></table></figure></div><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">head</span> plink_results.ld</span><br><span class="line"></span><br><span class="line"> CHR_A         BP_A             SNP_A  CHR_B         BP_B             SNP_B           R2</span><br><span class="line">22     16069141   22:16069141:C:G     22     16071624   22:16071624:A:G     0.771226</span><br><span class="line">22     16069784   22:16069784:A:T     22     16149743   22:16149743:T:A     0.217197</span><br><span class="line">22     16069784   22:16069784:A:T     22     16150589   22:16150589:C:A     0.224992</span><br><span class="line">22     16069784   22:16069784:A:T     22     16159060   22:16159060:G:A       0.2289</span><br><span class="line">22     16149743   22:16149743:T:A     22     16150589   22:16150589:C:A     0.965109</span><br><span class="line">22     16149743   22:16149743:T:A     22     16152606   22:16152606:T:C     0.692157</span><br><span class="line">22     16149743   22:16149743:T:A     22     16159060   22:16159060:G:A     0.721796</span><br><span class="line">22     16149743   22:16149743:T:A     22     16193549   22:16193549:C:T     0.336477</span><br><span class="line">22     16149743   22:16149743:T:A     22     16212542   22:16212542:C:T     0.442424</span><br></pre></td></tr></table></figure></div><p>R2是连锁不平衡系数，反应两个SNP的关联性强弱，范围 [0~1]。</p><h4 id="Data-management-make-bed-recode"><a href="#Data-management-make-bed-recode" class="headerlink" title="Data management (make-bed&#x2F;recode)"></a>Data management (make-bed&#x2F;recode)</h4><p>By far the input data we use is in binary form, but sometimes we may want the text version. To convert the formats, we can run:</p><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#extract the 1000 samples with the pruned SNPs, and make a bed file.</span></span><br><span class="line">plink2 \</span><br><span class="line">	--bfile <span class="variable">$&#123;genotypeFile&#125;</span> \</span><br><span class="line">	--extract plink_results.prune.in \</span><br><span class="line">	--make-bed \</span><br><span class="line">	--out plink_1000_pruned</span><br><span class="line"></span><br><span class="line"><span class="comment">#convert the bed/bim/fam to ped/map</span></span><br><span class="line">plink2 \</span><br><span class="line">	--bfile plink_1000_pruned \</span><br><span class="line">	--recode \</span><br><span class="line">	--out plink_1000_pruned</span><br></pre></td></tr></table></figure></div><h4 id="Apply-all-the-filters-to-obtain-a-clean-dataset"><a href="#Apply-all-the-filters-to-obtain-a-clean-dataset" class="headerlink" title="Apply all the filters to obtain a clean dataset"></a>Apply all the filters to obtain a clean dataset</h4><p>We can then apply the filters and remove samples with high $F_{het}$ to get a clean dataset for later use.</p><div class="code-container" data-rel="Plaintext"><figure class="iseeu highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">plink2 \</span><br><span class="line">	--bfile $&#123;genotypeFile&#125; \</span><br><span class="line">	--maf 0.01 \</span><br><span class="line">	--geno 0.02 \</span><br><span class="line">  --mind 0.02 \</span><br><span class="line">  --hwe 1e-6 \</span><br><span class="line">  --remove high_het.sample \</span><br><span class="line">  --keep-allele-order \</span><br><span class="line">  --make-bed \</span><br><span class="line">  --out sample_data.clean</span><br></pre></td></tr></table></figure></div><h3 id="Principle-component-analysis-PCA"><a href="#Principle-component-analysis-PCA" class="headerlink" title="Principle component analysis (PCA)"></a>Principle component analysis (PCA)</h3><p>PCA aims to find the <strong>orthogonal directions of maximum variance</strong> and project the data onto a new subspace with equal or fewer dimensions than the original one.<br><img width="600" alt="image" lazyload src="/images/loading.svg" data-src="https://github.com/Cloufield/GWASTutorial/assets/40289485/ee6bccfd-cf65-4126-88b1-047fc835b69b"></p><p>The red arrow shows the first principal component axis (PC1) and the blue arrow shows the second principal component axis (PC2). The two axes are <strong>orthogonal</strong>.</p><img width="600" alt="image" lazyload src="/images/loading.svg" data-src="https://github.com/Cloufield/GWASTutorial/assets/40289485/124b8c3d-0f83-4936-ab08-342efd29660a"><p>PCA is by far the most commonly used dimension reduction approach used in population genetics which could identify the difference in ancestry among the sample individuals.<br>The population outliers should be excluded from the samples used in GWAS to avoid bias caused by population stratification.<br>For GWAS, we also need to include top PCs to adjust for the population stratification.</p><p>Please read the following paper on how we apply PCA to genetic data:<br>Price, A., Patterson, N., Plenge, R. et al. Principal components analysis corrects for stratification in genome-wide association studies. Nat Genet 38, 904–909 (2006). <a class="link" target="_blank" rel="noopener" href="https://doi.org/10.1038/ng1847">https://doi.org/10.1038/ng1847<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a> <a class="link" target="_blank" rel="noopener" href="https://www.nature.com/articles/ng1847">https://www.nature.com/articles/ng1847<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a></p><h4 id="Pipeline"><a href="#Pipeline" class="headerlink" title="Pipeline"></a>Pipeline</h4><img width="600" alt="image" lazyload src="/images/loading.svg" data-src="https://github.com/Cloufield/GWASTutorial/assets/40289485/6a5880c7-10bd-4fac-a364-12ab14171f72"><h4 id="Prepareation"><a href="#Prepareation" class="headerlink" title="Prepareation"></a>Prepareation</h4><p>For PCA, we first <strong>exclude SNPs in high-LD or HLA</strong> regions from the genotype data.</p><blockquote><p>“The reason why we want to exclude such high-LD or HLA regions”<br>- Price, A. L., Weale, M. E., Patterson, N., Myers, S. R., Need, A. C., Shianna, K. V., Ge, D., Rotter, J. I., Torres, E., Taylor, K. D., Goldstein, D. B., &amp; Reich, D. (2008). Long-range LD can confound genome scans in admixed populations. American journal of human genetics, 83(1), 132–139. <a class="link" target="_blank" rel="noopener" href="https://doi.org/10.1016/j.ajhg.2008.06.005">https://doi.org/10.1016/j.ajhg.2008.06.005<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a></p></blockquote><p>use <code>high-ld.txt</code> to extract all SNPs that are located in the regions described in the file using the code as follows:</p><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plink2 --bfile <span class="variable">$&#123;plinkFile&#125;</span> --make-set high-ld.txt --write-set --out hild</span><br></pre></td></tr></table></figure></div><div class="code-container" data-rel="Plaintext"><figure class="iseeu highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">head high-ld.txt</span><br><span class="line"></span><br><span class="line">1	48000000	52000000	highld</span><br><span class="line">2	86000000	100500000	highld</span><br><span class="line">2	134500000	138000000	highld</span><br><span class="line">2	183000000	190000000	highld</span><br><span class="line">3	47500000	50000000	highld</span><br><span class="line">3	83500000	87000000	highld</span><br><span class="line">3	89000000	97500000	highld</span><br><span class="line">5	44500000	50500000	highld</span><br><span class="line">5	98000000	100500000	highld</span><br><span class="line">5	129000000	132000000	highld</span><br><span class="line">5	135500000	138500000	highld</span><br><span class="line">6	25000000	35000000	highld</span><br></pre></td></tr></table></figure></div><div class="code-container" data-rel="Plaintext"><figure class="iseeu highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">head hild.set</span><br><span class="line"></span><br><span class="line">highld</span><br><span class="line">1:48000156:C:G</span><br><span class="line">1:48002096:C:G</span><br><span class="line">1:48003081:T:C</span><br><span class="line">1:48004776:C:T</span><br><span class="line">1:48006500:A:G</span><br><span class="line">1:48006546:C:T</span><br><span class="line">1:48008102:T:G</span><br><span class="line">1:48009994:C:T</span><br><span class="line">1:48009997:C:A</span><br></pre></td></tr></table></figure></div><p>For downstream analysis, we can exclude these SNPs using <code>--exclude hild.set</code>.</p><h4 id="PAC-step"><a href="#PAC-step" class="headerlink" title="PAC step"></a>PAC step</h4><ol><li>LD-Pruning (<a class="link" target="_blank" rel="noopener" href="https://www.cog-genomics.org/plink/2.0/ld#indep">https://www.cog-genomics.org/plink/2.0/ld#indep<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a>)</li><li>Removing relatives from calculating PCs (usually 2-degree) (<a class="link" target="_blank" rel="noopener" href="https://www.cog-genomics.org/plink/2.0/distance#king_cutoff">https://www.cog-genomics.org/plink/2.0/distance#king_cutoff<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a>)</li><li>Running PCA using un-related samples and independent SNPs (<a class="link" target="_blank" rel="noopener" href="https://www.cog-genomics.org/plink/2.0/strat#pca">https://www.cog-genomics.org/plink/2.0/strat#pca<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a>)</li><li>Projecting to all samples (<a class="link" target="_blank" rel="noopener" href="https://www.cog-genomics.org/plink/2.0/score#pca_project">https://www.cog-genomics.org/plink/2.0/score#pca_project<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a>)</li></ol><p>(why removing high-LD regions) Price, A. L., Weale, M. E., Patterson, N., Myers, S. R., Need, A. C., Shianna, K. V., Ge, D., Rotter, J. I., Torres, E., Taylor, K. D., Goldstein, D. B., &amp; Reich, D. (2008). Long-range LD can confound genome scans in admixed populations. American journal of human genetics, 83(1), 132–139. <a class="link" target="_blank" rel="noopener" href="https://doi.org/10.1016/j.ajhg.2008.06.005">https://doi.org/10.1016/j.ajhg.2008.06.005<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a></p><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br></pre></td><td class="code"><pre><span class="line">plinkFile=<span class="string">&quot;&quot;</span> <span class="comment">#please set this to your own path</span></span><br><span class="line">outPrefix=<span class="string">&quot;plink_results&quot;</span></span><br><span class="line">threadnum=8</span><br><span class="line">hildset = hild.set </span><br><span class="line"></span><br><span class="line"><span class="comment"># LD-pruning, excluding high-LD and HLA regions</span></span><br><span class="line">plink2 \</span><br><span class="line">	--bfile <span class="variable">$&#123;plinkFile&#125;</span> \</span><br><span class="line">	--maf 0.01 \</span><br><span class="line">	--threads <span class="variable">$&#123;threadnum&#125;</span> \</span><br><span class="line">	--exclude <span class="variable">$&#123;hildset&#125;</span> \ </span><br><span class="line">	--indep-pairwise 500 50 0.2 \</span><br><span class="line">	--out <span class="variable">$&#123;outPrefix&#125;</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># Remove related samples using king-cuttoff</span></span><br><span class="line">plink2 \</span><br><span class="line">	--bfile <span class="variable">$&#123;plinkFile&#125;</span> \</span><br><span class="line">	--extract <span class="variable">$&#123;outPrefix&#125;</span>.prune.in \</span><br><span class="line">	--king-cutoff 0.0884 \</span><br><span class="line">	-threads <span class="variable">$&#123;threadnum&#125;</span> \</span><br><span class="line">	--out <span class="variable">$&#123;outPrefix&#125;</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># PCA after pruning and removing related samples</span></span><br><span class="line">plink2 \</span><br><span class="line">	--bfile <span class="variable">$&#123;plinkFile&#125;</span> \</span><br><span class="line">	--keep <span class="variable">$&#123;outPrefix&#125;</span>.king.cutoff.in.id \</span><br><span class="line">	--extract <span class="variable">$&#123;outPrefix&#125;</span>.prune.in \</span><br><span class="line">	--freq counts \</span><br><span class="line">	--threads <span class="variable">$&#123;threadnum&#125;</span> \</span><br><span class="line">	--pca approx allele-wts 10 \     </span><br><span class="line">	--out <span class="variable">$&#123;outPrefix&#125;</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># Projection (related and unrelated samples)</span></span><br><span class="line">plink2 \</span><br><span class="line">	--bfile <span class="variable">$&#123;plinkFile&#125;</span> \</span><br><span class="line">	--threads <span class="variable">$&#123;threadnum&#125;</span> \</span><br><span class="line">	--read-freq <span class="variable">$&#123;outPrefix&#125;</span>.acount \</span><br><span class="line">	--score <span class="variable">$&#123;outPrefix&#125;</span>.eigenvec.allele 2 6 header-read no-mean-imputation variance-standardize \</span><br><span class="line">	--score-col-nums 7-16 \</span><br><span class="line">	--out <span class="variable">$&#123;outPrefix&#125;</span>_projected</span><br></pre></td></tr></table></figure></div><p>info “<code>--pca</code> and <code>--pca approx</code>“</p><p>For step 3, please note that <code>approx</code> flag is <strong>only recommended for analysis of &gt;5000 samples</strong>. (It was applied in the sample code anyway because in real analysis you usually have a much larger sample size, though the sample size of our data is just ~500)</p><p>After step 3, the <code>allele-wts 10</code> modifier requests an additional one-line-per-allele <code>.eigenvec.allele</code> file with the first <code>10 PCs</code> expressed as allele weights instead of sample weights.</p><p>We will get the <code>plink_results.eigenvec.allele</code> file, which will be used to project onto all samples along with an allele count <code>plink_results.acount</code> file.</p><p>In the projection, <code>score $&#123;outPrefix&#125;.eigenvec.allele 2 5</code> sets the <code>ID</code> (2nd column) and <code>A1</code> (5th column), <code>score-col-nums 6-15</code> sets the first 10 PCs to be projected. Please check <a class="link" target="_blank" rel="noopener" href="https://www.cog-genomics.org/plink/2.0/score#pca_project">https://www.cog-genomics.org/plink/2.0/score#pca_project<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a> for more details on the projection.</p><div class="note-large default"><div class="notel-title rounded-t-lg p-3 font-bold text-lg flex flex-row gap-2 items-center"><i class="notel-icon fa-solid fa-info"></i><p>warning</p></div><div class="notel-content"><p>warning “Please check the content of your <code>.eigenvec.allele</code> file”<br>Using recent plink2 versions, there are some minor changes in the output format.<br><code>A1</code> is the 6th column, and the <code>score-col-nums</code> should be <code>7-16</code><br>Please adjust the column number in your script accordingly.</p></div></div><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">head</span> plink_results.eigenvec.allele</span><br><span class="line"></span><br><span class="line"><span class="comment">#CHROM  ID      REF     ALT     PROVISIONAL_REF?        A1      PC1     PC2     PC3     PC4     PC5     PC6     PC7PC8      PC9     PC10</span></span><br><span class="line">    1       1:15774:G:A     G       A       Y       G       0.57834 -1.03002        0.744557        -0.161887       0.389223    -0.0514592      0.133195        -0.0336162      -0.846376       0.0542876</span><br><span class="line">    1       1:15774:G:A     G       A       Y       A       -0.57834        1.03002 -0.744557       0.161887        -0.389223   0.0514592       -0.133195       0.0336162       0.846376        -0.0542876</span><br><span class="line">    1       1:15777:A:G     A       G       Y       A       -0.585215       0.401872        -0.393071       -1.79583   0.89579  -0.700882       -0.103729       -0.694495       -0.007313       0.513223</span><br><span class="line">    1       1:15777:A:G     A       G       Y       G       0.585215        -0.401872       0.393071        1.79583 -0.89579    0.700882        0.103729        0.694495        0.007313        -0.513223</span><br><span class="line">    1       1:57292:C:T     C       T       Y       C       -0.123768       0.912046        -0.353606       -0.220148  -0.893017        -0.374505       -0.141002       -0.249335       0.625097        0.206104</span><br><span class="line">    </span><br><span class="line">    </span><br><span class="line"><span class="built_in">head</span> plink_results.acount</span><br><span class="line"></span><br><span class="line"><span class="comment">#CHROM  ID      REF     ALT     PROVISIONAL_REF?        ALT_CTS OBS_CT</span></span><br><span class="line">1       1:15774:G:A     G       A       Y       28      994</span><br><span class="line">1       1:15777:A:G     A       G       Y       73      994</span><br><span class="line">1       1:57292:C:T     C       T       Y       104     988</span><br><span class="line">1       1:77874:G:A     G       A       Y       19      994</span><br><span class="line">1       1:87360:C:T     C       T       Y       23      998</span><br><span class="line">1       1:125271:C:T    C       T       Y       967     996</span><br><span class="line">1       1:232449:G:A    G       A       Y       185     996</span><br><span class="line">1       1:533113:A:G    A       G       Y       129     992</span><br><span class="line">1       1:565697:A:G    A       G       Y       334     996</span><br></pre></td></tr></table></figure></div><p>Eventually, we will get the PCA results for all samples.</p><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">head</span> plink_results_projected.sscore</span><br><span class="line"></span><br><span class="line"> <span class="comment">#FID    IID     ALLELE_CT       NAMED_ALLELE_DOSAGE_SUM PC1_AVG PC2_AVG PC3_AVG PC4_AVG PC5_AVG PC6_AVG PC7_AVG PC8_AVG     PC9_AVG PC10_AVG</span></span><br><span class="line">    HG00403 HG00403 390256  390256  0.00290265      -0.0248649      0.0100408       0.00957591      0.00694349      -0.00222251 0.0082228       -0.00114937     0.00335249      0.00437471</span><br><span class="line">    HG00404 HG00404 390696  390696  -0.000141221    -0.027965       0.025389        -0.00582538     -0.00274707     0.00658501  0.0113803       0.0077766       0.0159976       0.0178927</span><br><span class="line">    HG00406 HG00406 388524  388524  0.00707397      -0.0315445      -0.00437011     -0.0012621      -0.0114932      -0.00539483 -0.00620153     0.00452379      -0.000870627    -0.00227979</span><br><span class="line">    HG00407 HG00407 388808  388808  0.00683977      -0.025073       -0.00652723     0.00679729      -0.0116 -0.0102328 0.0139572        0.00618677      0.0138063       0.00825269</span><br></pre></td></tr></table></figure></div><h4 id="Plotting-the-PCs"><a href="#Plotting-the-PCs" class="headerlink" title="Plotting the PCs"></a>Plotting the PCs</h4><p>You can now create scatterplots of the PCs using R or Python.</p><p>For plotting using Python:<br><a class="link" target="_blank" rel="noopener" href="https://github.com/Cloufield/GWASTutorial/blob/main/05_PCA/plot_PCA.ipynb">plot_PCA.ipynb<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a></p><p>Note : We only used a small proportion of all available variants. This figure only very roughly shows the population structure in East Asia.</p><p>Requirements:</p><ul><li>python&gt;3</li><li>numpy,pandas,seaborn,matplotlib</li></ul><h4 id="PCA-UMAP-optional"><a href="#PCA-UMAP-optional" class="headerlink" title="PCA-UMAP (optional)"></a>PCA-UMAP (optional)</h4><p>We can also apply another non-linear dimension reduction algorithm called UMAP to the PCs to further identify the local structures. (PCA-UMAP)</p><p>For more details, please check:</p><ul><li><a class="link" target="_blank" rel="noopener" href="https://umap-learn.readthedocs.io/en/latest/index.html">https://umap-learn.readthedocs.io/en/latest/index.html<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a></li></ul><p>An example of PCA and PCA-UMAP for population genetics:</p><ul><li>Sakaue, S., Hirata, J., Kanai, M., Suzuki, K., Akiyama, M., Lai Too, C., … &amp; Okada, Y. (2020). Dimensionality reduction reveals fine-scale structure in the Japanese population with consequences for polygenic risk prediction. Nature communications, 11(1), 1-11.</li><li>(UMAP in population genetics) Diaz-Papkovich, A., Anderson-Trocmé, L. &amp; Gravel, S. A review of UMAP in population genetics. J Hum Genet 66, 85–91 (2021). <a class="link" target="_blank" rel="noopener" href="https://doi.org/10.1038/s10038-020-00851-4">https://doi.org/10.1038/s10038-020-00851-4<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a> <a class="link" target="_blank" rel="noopener" href="https://www.nature.com/articles/s10038-020-00851-4">https://www.nature.com/articles/s10038-020-00851-4<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a></li></ul><h3 id="Association-analysis"><a href="#Association-analysis" class="headerlink" title="Association analysis"></a>Association analysis</h3><h4 id="Overview"><a href="#Overview" class="headerlink" title="Overview"></a>Overview</h4><img width="900" alt="image" lazyload src="/images/loading.svg" data-src="https://github.com/Cloufield/GWASTutorial/assets/40289485/1e4ab229-6f5a-44a0-9850-81244944b969"><h4 id="Model-election"><a href="#Model-election" class="headerlink" title="Model election"></a>Model election</h4><p>To test the association between a phenotype and genotypes, we need to group the genotypes based on genetic models.</p><p>There are three basic genetic models:</p><ul><li>Additive model (ADD)</li><li>Dominant model (DOM)</li><li>Recessive model (REC)</li></ul><p>For example, suppose we have a biallelic SNP whose reference allele is A and the alternative allele is G.</p><p>There are three possible genotypes for this SNP: AA, AG, and GG.</p><p>This table shows how we group different genotypes under each genetic model for association tests using linear or logistic regressions.</p><table><thead><tr><th>Genetic models</th><th>AA</th><th>AG</th><th>GG</th></tr></thead><tbody><tr><td>Additive model</td><td>0</td><td>1</td><td>2</td></tr><tr><td>Dominant model</td><td>0</td><td>1</td><td>1</td></tr><tr><td>Recessive model</td><td>0</td><td>0</td><td>1</td></tr></tbody></table><p><strong>若群体结构简单</strong>：PLINK的显性&#x2F;隐性模型可针对性检验非加性效应。因此，admixture展示出结构分层，使用EMMAX. (<a class="link" target="_blank" rel="noopener" href="https://github.com/ShawnWx2019/emmax-pipeline">https://github.com/ShawnWx2019/emmax-pipeline<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a>)</p><p>A simple way to test association is to use the 2x2 or 2x3 contingency table. For dominant and recessive models, Chi-square tests are performed using the 2x2 table. For the additive model, Cochran-Armitage trend tests are performed for the 2x3 table. However, the non-parametric tests do not adjust for the bias caused by other covariates like sex, age and so forth.</p><img width="900" alt="image" lazyload src="/images/loading.svg" data-src="https://github.com/Cloufield/GWASTutorial/assets/40289485/f2b0f941-9405-4ebe-8e89-aef608eb9bb1"><h4 id="Association-testing-basics"><a href="#Association-testing-basics" class="headerlink" title="Association testing basics"></a>Association testing basics</h4><p>For quantitative traits, we can employ a simple <strong>linear regression model</strong> to test associations:</p><p>$$<br>y &#x3D; G\beta_G + X\beta_X + e<br>$$</p><ul><li>$G$ is the genotype matrix.</li><li>$\beta_G$ is the effect size for variants.</li><li>$X$ and $\beta_X$ are covariates and their effects.</li><li>$e$ is the error term.</li></ul><img width="900" alt="image" lazyload src="/images/loading.svg" data-src="https://github.com/Cloufield/GWASTutorial/assets/40289485/d5092f82-c5b5-4f08-873e-c2942955e634"><p>For binary traits, we can utilize the logistic regression model to test associations:</p><p>$$<br>logit(p) &#x3D; G\beta_G + X\beta_X + e<br>$$</p><img width="900" alt="image" lazyload src="/images/loading.svg" data-src="https://github.com/Cloufield/GWASTutorial/assets/40289485/1b902ad6-db06-44f7-9215-43b13a1a4fea"><h4 id="Files-preparation"><a href="#Files-preparation" class="headerlink" title="Files preparation"></a>Files preparation</h4><p>To perform genome-wide association tests, usually, we need the following files:</p><ul><li><strong>Genotype file</strong> (or dosage file) : usually in PLINK format, VCF format, or BGEN format.</li><li><strong>Phenotype file</strong> : plain text file.</li><li><strong>Covariate file</strong> (optional): plain text file. Commonly used covariates include age, sex, and top Principal Components.</li></ul><p>Phenotype file for a simulated binary trait; B1 is the phenotype name; 1 means the control, 2 means the case.</p><div class="code-container" data-rel="Txt"><figure class="iseeu highlight txt"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><span class="line">title=&quot;1kgeas_binary.txt&quot;</span><br><span class="line"></span><br><span class="line">FID IID B1</span><br><span class="line">HG00403 HG00403 1</span><br><span class="line">HG00404 HG00404 2</span><br><span class="line">HG00406 HG00406 1</span><br><span class="line">HG00407 HG00407 1</span><br><span class="line">HG00409 HG00409 2</span><br><span class="line">HG00410 HG00410 2</span><br><span class="line">HG00419 HG00419 1</span><br><span class="line">HG00421 HG00421 1</span><br><span class="line">HG00422 HG00422 1</span><br><span class="line"></span><br><span class="line">Covariate file (only top PCs calculated in the previous PCA section)</span><br><span class="line"></span><br><span class="line">title=&quot;plink_results_projected.sscore&quot;</span><br><span class="line"></span><br><span class="line">#FID    IID     ALLELE_CT       NAMED_ALLELE_DOSAGE_SUM PC1_AVG PC2_AVG PC3_AVG PC4_AVG PC5_AVG PC6_AVG PC7_AVG PC8_AVGPC9_AVG  PC10_AVG</span><br><span class="line">HG00403 HG00403 390256  390256  0.00290265      -0.0248649      -0.0100407      0.00957595      0.00694056      0.00222996      0.00823028      0.00116497      -0.00334937     0.00434627</span><br><span class="line">HG00404 HG00404 390696  390696  -0.000141221    -0.027965       -0.025389       -0.00582553     -0.00274711     -0.00657958     0.0113769       -0.00778919     -0.0159685      0.0180678</span><br><span class="line">HG00406 HG00406 388524  388524  0.00707397      -0.0315445      0.00437013      -0.00126195     -0.0114938      0.00538932      -0.00619657     -0.00454686     0.000969112     -0.00217617</span><br><span class="line">HG00407 HG00407 388808  388808  0.00683977      -0.025073       0.00652723      0.00679731      -0.0116001      0.0102403       0.0139674       -0.00621948     -0.013797       0.00827744</span><br><span class="line">HG00409 HG00409 391646  391646  0.000398695     -0.0290334      0.0189352       -0.00135996     0.0290464       -0.00941851     -0.0171911      0.01293 -0.0252628      0.0230819</span><br><span class="line">HG00410 HG00410 391600  391600  0.00277094      -0.0280021      0.0209991       -0.00799089     0.0318043       0.00283456      -0.0315157      0.000978664     -0.0133768      0.0356721</span><br><span class="line">HG00419 HG00419 387118  387118  0.00684154      -0.0326244      -0.00237159     0.0167284       -0.0119684      0.00795149      -0.0144241      -0.00716183     -0.0115059      0.0038652</span><br><span class="line">HG00421 HG00421 387720  387720  0.00157095      -0.0338115      0.00690542      0.0121058       0.00111448      -0.00531714     -0.00175494     0.00118513      -0.00391494     0.00414682</span><br><span class="line">HG00422 HG00422 387466  387466  0.00439167      -0.0332386      -0.000741482    0.0124843       -0.00362885     0.00342491      -0.0073205      -0.00939123     0.010718        0.00360906</span><br></pre></td></tr></table></figure></div><h4 id="Association-tests-using-PLINK"><a href="#Association-tests-using-PLINK" class="headerlink" title="Association tests using PLINK"></a>Association tests using PLINK</h4><div class="code-container" data-rel="Txt"><figure class="iseeu highlight txt"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">genotypeFile=&quot;../04_Data_QC/sample_data.clean&quot; # the clean dataset we generated in previous section</span><br><span class="line">phenotypeFile=&quot;../01_Dataset/1kgeas_binary.txt&quot; # the phenotype file</span><br><span class="line">covariateFile=&quot;../05_PCA/plink_results_projected.sscore&quot; # the PC score file</span><br></pre></td></tr></table></figure></div><div class="code-container" data-rel="Bash"><figure class="iseeu highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">covariateCols=6-10</span><br><span class="line">colName=<span class="string">&quot;B1&quot;</span></span><br><span class="line">threadnum=2</span><br><span class="line"></span><br><span class="line">plink2 \</span><br><span class="line">	--bfile <span class="variable">$&#123;genotypeFile&#125;</span> \</span><br><span class="line">	--pheno <span class="variable">$&#123;phenotypeFile&#125;</span> \</span><br><span class="line">	--pheno-name <span class="variable">$&#123;colName&#125;</span> \</span><br><span class="line">	--maf 0.01 \</span><br><span class="line">	--covar <span class="variable">$&#123;covariateFile&#125;</span> \</span><br><span class="line">	--covar-col-nums <span class="variable">$&#123;covariateCols&#125;</span> \</span><br><span class="line">	--glm hide-covar firth  firth-residualize single-prec-cc \</span><br><span class="line">	--threads <span class="variable">$&#123;threadnum&#125;</span> \</span><br><span class="line">	--out 1kgeas</span><br></pre></td></tr></table></figure></div><p>Association test results</p><div class="code-container" data-rel="Plaintext"><figure class="iseeu highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">    #CHROM  POS     ID      REF     ALT     PROVISIONAL_REF?        A1      OMITTED A1_FREQ TEST    OBS_CT  OR      LOG(OR)_SE  Z_STAT  P       ERRCODE</span><br><span class="line">1       15774   1:15774:G:A     G       A       Y       A       G       0.0282828       ADD     495     NA      NA NA       NA      FIRTH_CONVERGE_FAIL</span><br><span class="line">1       15777   1:15777:A:G     A       G       Y       G       A       0.0737374       ADD     495     NA      NA NA       NA      FIRTH_CONVERGE_FAIL</span><br><span class="line">1       57292   1:57292:C:T     C       T       Y       T       C       0.104675        ADD     492     NA      NA NA       NA      FIRTH_CONVERGE_FAIL</span><br><span class="line">1       77874   1:77874:G:A     G       A       Y       A       G       0.0191532       ADD     496     1.12228 0.46275     0.249299        0.80313 .</span><br><span class="line">1       87360   1:87360:C:T     C       T       Y       T       C       0.0231388       ADD     497     NA      NA NA       NA      FIRTH_CONVERGE_FAIL</span><br><span class="line">1       125271  1:125271:C:T    C       T       Y       C       T       0.0292339       ADD     496     1.53387 0.373358    1.1458  0.25188 .</span><br><span class="line">1       232449  1:232449:G:A    G       A       Y       A       G       0.185484        ADD     496     0.884097   0.168961 -0.729096       0.465943        .</span><br><span class="line">1       533113  1:533113:A:G    A       G       Y       G       A       0.129555        ADD     494     0.90593 0.196631    -0.50243        0.615365        .</span><br><span class="line">1       565697  1:565697:A:G    A       G       Y       G       A       0.334677        ADD     496     1.04653 0.15286     0.297509        0.766078        .</span><br></pre></td></tr></table></figure></div><div class="note-large default"><div class="notel-title rounded-t-lg p-3 font-bold text-lg flex flex-row gap-2 items-center"><i class="notel-icon fa-solid fa-info"></i><p>info</p></div><div class="notel-content"><p>Usually, other options are added to enhance the sumstats</p><ul><li>–keep xxx&#x2F;kiso2021&#x2F;for_plink2&#x2F;unrelated.sample.id	# Because the standard linear regression does not account for the relatedness, the kinship-pruned samples in last steps are suggested.</li><li>–mach-r2-filter 0.7 2.0	# It allows to use only the variants passed an (MaCH)Rsq filter. NOTE: when pgen file is used, the upper boundary should be 2.</li><li>–glm <strong>cols&#x3D;+a1freq,+machr2</strong> firth-fallback <strong>omit-ref</strong>	# The <code>cols=</code> requests the following columns in the sumstats: here are allele1 frequency and (MaCH)Rsq, <code>firth-fallback</code> will test the common variants without firth correction, which could improve the speed, <code>omit-ref</code> will force the ALTA1effect allele, otherwise the minor allele would be tested (see the above result, which ALT may not equal A1).</li><li>–covar-variance-standardize	# To normalize the covariates which may at a huge scale, like AGEAGE.</li><li>–covar-name AGE SEX PC1-PC20	# Instead of setting the index of columns, directly specify the column name.</li></ul></div></div><h4 id="Genomic-control"><a href="#Genomic-control" class="headerlink" title="Genomic control"></a>Genomic control</h4><p>Genomic control (GC) is a basic method for controlling for confounding factors including population stratification.</p><div class="note-large default"><div class="notel-title rounded-t-lg p-3 font-bold text-lg flex flex-row gap-2 items-center"><i class="notel-icon fa-solid fa-info"></i><p>info</p></div><div class="notel-content"><p>在GWAS中，<strong>群体分层</strong>（如样本来自不同祖先群体）会导致基因型与表型的虚假关联。例如：</p><ul><li>某SNP在群体A中频率高，而群体A恰好平均身高更高。</li><li>若不控制群体结构，该SNP可能被错误地认为与身高相关（假阳性）。</li></ul><p>Genomic Control 通过分析统计量的分布，检测是否存在系统性偏差（如群体分层、隐性亲缘关系等），并进行校正。</p></div></div><p>We will calculate the genomic control factor (lambda GC) to evaluate the inflation. The genomic control factor is calculated by dividing the <strong>median of observed Chi square statistics</strong> by the <strong>median of Chi square distribution with the degree of freedom being 1</strong> (which is approximately 0.455).</p><p>$$<br>\lambda_{GC} &#x3D; {median(\chi^{2}_{observed}) \over median(\chi^{2}_1)}<br>$$</p><p>Then, we can used the genomic control factor to correct observed Chi suqare statistics.</p><p>$$<br>\chi^{2}<em>{corrected} &#x3D; {\chi^{2}</em>{observed} \over \lambda_{GC}}<br>$$</p><p>Genomic inflation is based on the idea that most of the variants are not associated, thus no deviation between the observed and expected Chi square distribution, except the spikes at the end. However, if the trait is highly polygenic, this assumption may be violated.</p><p>Reference: Devlin, B., &amp; Roeder, K. (1999). Genomic control for association studies. Biometrics, 55(4), 997-1004.</p><h4 id="Significant-loci"><a href="#Significant-loci" class="headerlink" title="Significant loci"></a>Significant loci</h4><p>Loci that reached genome-wide significance threshold (<strong>P value &lt; 5e-8</strong>) :</p><div class="code-container" data-rel="Plaintext"><figure class="iseeu highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">SNPID	CHR	POS	EA	NEA	EAF	SE	Z	P	OR	N	STATUS	REF	ALT</span><br><span class="line">1:167562605:G:A	1	167562605	A	G	0.391481	0.159645	7.69462	1.419150e-14	3.415780	493	9999999	G	A</span><br><span class="line">2:55513738:C:T	2	55513738	C	T	0.376008	0.153159	-7.96244	1.686760e-15	0.295373	496	9999999	C	T</span><br><span class="line">7:134368632:T:G	7	134368632	G	T	0.138105	0.225526	6.89025	5.569440e-12	4.730010	496	9999999	T	G</span><br><span class="line">20:42758834:T:C	20	42758834	T	C	0.227273	0.184323	-7.76902	7.909780e-15	0.238829	495	9999999	T	C</span><br></pre></td></tr></table></figure></div><h4 id="Visualization"><a href="#Visualization" class="headerlink" title="Visualization"></a>Visualization</h4><p>To visualize the sumstats, we will create the Manhattan plot, QQ plot and regional plot.</p><p>Please check for codes : <a class="link" target="_blank" rel="noopener" href="https://cloufield.github.io/GWASTutorial/Visualization/">Visualization using gwaslab<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a></p><p><img lazyload src="/images/loading.svg" data-src="https://github.com/Cloufield/GWASTutorial/assets/40289485/40ad5aff-5ac5-4cd0-b9c2-077c0ce20e46" alt="image"></p><div class="note-large default"><div class="notel-title rounded-t-lg p-3 font-bold text-lg flex flex-row gap-2 items-center"><i class="notel-icon fa-solid fa-info"></i><p>tips</p></div><div class="notel-content"><p>Steps to create Manhattan plot</p><ol><li>sort the variants by genome coordinates.</li><li>map the genome coordinates of variants to the x axis.</li><li>convert P value to -log10(P).</li><li>create the scatter plot.</li></ol><p>Steps to create Q-Q plot</p><p>Suppose we have <code>n</code> variants in our sumstats, <strong>n &#x3D; 检验的SNP数量</strong>（即GWAS分析中纳入统计检验的遗传变异总数）。</p><ol><li>convert the <code>n</code> P value to -log10(P).</li><li>sort the -log10(P) values in asending order.</li><li>get <code>n</code> numbers from <code>(0,1)</code> with equal intervals.</li><li>convert the <code>n</code> numbers to -log10(P) and sort in ascending order.</li><li>create scatter plot using the sorted -log10(P) of sumstats as Y and sorted -log10(P) we generated as X.</li></ol></div></div><h3 id="Regional-plot"><a href="#Regional-plot" class="headerlink" title="Regional plot"></a>Regional plot</h3><p>Manhattan plot is very useful to check the overview of our sumstats. But if we want to check a specific genomic locus, we need a plot with finer resolution. This kind of plot is called a <strong>regional plot</strong>. It is basically the Manhattan plot of only a small region on the genome, with points colored by its LD r2 with the lead variant in this region.</p><p>Such a plot is especially helpful to understand the signal and loci, e.g., LD structure, independent signals, and genes.</p><p>The regional plot for the loci of 2:55513738:C:T.</p><p>Please check <a class="link" target="_blank" rel="noopener" href="https://cloufield.github.io/GWASTutorial/Visualization/">Visualization using gwaslab<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a></p><p><img lazyload src="/images/loading.svg" data-src="https://github.com/Cloufield/GWASTutorial/assets/40289485/2aacd0b4-4a4a-485b-97bd-8548679f19e0" alt="image"></p></div><div class="post-copyright-info w-full my-8 px-2 sm:px-6 md:px-8"><div class="article-copyright-info-container"><ul><li><strong>Title:</strong> GWAS -- PLINK</li><li><strong>Author:</strong> Yuchen</li><li><strong>Created at :</strong> 2025-06-10 15:32:00</li><li><strong>Updated at :</strong> 2025-06-24 01:02:03</li><li><strong>Link:</strong> https://yuchen-starfall.github.io/2025/06/10/GWAS - PLINK/</li><li><strong>License: </strong>This work is licensed under <a class="license" target="_blank" rel="noopener" href="https://creativecommons.org/licenses/by-nc-sa/4.0">CC BY-NC-SA 4.0</a>.</li></ul></div></div><ul class="post-tags-box text-lg mt-1.5 flex-wrap justify-center flex md:hidden"><li class="tag-item mx-0.5"><a href="/tags/Biomedical-Statistics/">#Biomedical Statistics</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/GWAS/">#GWAS</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/linkage-disequilibrium%EF%BC%88LD%EF%BC%89/">#linkage disequilibrium（LD）</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/Tools/">#Tools</a>&nbsp;</li></ul><div class="article-nav my-8 flex justify-between items-center px-2 sm:px-6 md:px-8"><div class="article-prev border-border-color shadow-redefine-flat shadow-shadow-color-2 rounded-medium px-4 py-2 hover:shadow-redefine-flat-hover hover:shadow-shadow-color-2"><a class="prev" rel="prev" href="/2025/06/11/CAS%20Evaluate%20teacher&#39;s%20work/"><span class="left arrow-icon flex justify-center items-center"><i class="fa-solid fa-chevron-left"></i> </span><span class="title flex justify-center items-center"><span class="post-nav-title-item">CAS Evaluate teacher&#39;s work</span> <span class="post-nav-item">Prev posts</span></span></a></div><div class="article-next border-border-color shadow-redefine-flat shadow-shadow-color-2 rounded-medium px-4 py-2 hover:shadow-redefine-flat-hover hover:shadow-shadow-color-2"><a class="next" rel="next" href="/2025/06/09/Interesting%20software%20for%20Mac/"><span class="title flex justify-center items-center"><span class="post-nav-title-item">Interesting software for MAC</span> <span class="post-nav-item">Next posts</span> </span><span class="right arrow-icon flex justify-center items-center"><i class="fa-solid fa-chevron-right"></i></span></a></div></div><div class="comment-container px-2 sm:px-6 md:px-8 pb-8"><div class="comments-container mt-10 w-full"><div id="comment-anchor" class="w-full h-2.5"></div><div class="comment-area-title w-full my-1.5 md:my-2.5 text-xl md:text-3xl font-bold">Comments</div><div id="giscus-container"></div><script data-swup-reload-script defer>async function loadGiscus(){const t={src:"https://giscus.app/client.js","data-repo":"Yuchen-starfall/Yuchen-starfall.github.io","data-repo-id":"R_kgDONY6r-Q","data-category":"Announcements# Github discussion category","data-category-id":"DIC_kwDONY6r-c4Ck8Zm","data-mapping":"pathname","data-strict":"0","data-reactions-enabled":"1","data-emit-metadata":"1","data-theme":"preferred_color_scheme","data-lang":"en","data-input-position":"bottom","data-loading":"lazy",crossorigin:"anonymous",async:!0},a=document.createElement("script");for(const e in t)a.setAttribute(e,t[e]);document.getElementById("giscus-container").appendChild(a)}{let t=setTimeout(()=>{loadGiscus(),clearTimeout(t)},1e3)}</script></div></div></div><div class="toc-content-container"><div class="post-toc-wrap"><div class="post-toc"><div class="toc-title">On this page</div><div class="page-title">GWAS -- PLINK</div><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#PLINK-format"><span class="nav-text">PLINK format</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#ped-map"><span class="nav-text">ped &#x2F; map</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#bed-fam-bim"><span class="nav-text">bed &#x2F; fam &#x2F; bim</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Imputation-dosage"><span class="nav-text">Imputation dosage</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Summary"><span class="nav-text">Summary</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#PLINK-tutorial"><span class="nav-text">PLINK tutorial</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Download-PLINK-1-9-2-0-and-then-unzip"><span class="nav-text">Download PLINK 1.9 &#x2F; 2.0 and then unzip</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Add-paths-to-environment-path"><span class="nav-text">Add paths to environment path</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Download-the-genotype-data"><span class="nav-text">Download the genotype data</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#QC"><span class="nav-text">QC</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Missing-rate-call-rate"><span class="nav-text">Missing rate (call rate)</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Allele-Frequency"><span class="nav-text">Allele Frequency</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Hardy-Weinberg-equilibrium-exact-test"><span class="nav-text">Hardy-Weinberg equilibrium exact test</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Applying-filters"><span class="nav-text">Applying filters</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#LD-Pruning"><span class="nav-text">LD Pruning</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Inbreeding-F-coefficient"><span class="nav-text">Inbreeding F coefficient</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Sample-SNP-filtering-extract-exclude-keep-remove"><span class="nav-text">Sample &amp; SNP filtering (extract&#x2F;exclude&#x2F;keep&#x2F;remove)</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#IBD-PI-HAT-kinship-coefficient"><span class="nav-text">IBD &#x2F; PI_HAT &#x2F; kinship coefficient</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#LD-calculation"><span class="nav-text">LD calculation</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Data-management-make-bed-recode"><span class="nav-text">Data management (make-bed&#x2F;recode)</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Apply-all-the-filters-to-obtain-a-clean-dataset"><span class="nav-text">Apply all the filters to obtain a clean dataset</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Principle-component-analysis-PCA"><span class="nav-text">Principle component analysis (PCA)</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Pipeline"><span class="nav-text">Pipeline</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Prepareation"><span class="nav-text">Prepareation</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#PAC-step"><span class="nav-text">PAC step</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Plotting-the-PCs"><span class="nav-text">Plotting the PCs</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#PCA-UMAP-optional"><span class="nav-text">PCA-UMAP (optional)</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Association-analysis"><span class="nav-text">Association analysis</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Overview"><span class="nav-text">Overview</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Model-election"><span class="nav-text">Model election</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Association-testing-basics"><span class="nav-text">Association testing basics</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Files-preparation"><span class="nav-text">Files preparation</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Association-tests-using-PLINK"><span class="nav-text">Association tests using PLINK</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Genomic-control"><span class="nav-text">Genomic control</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Significant-loci"><span class="nav-text">Significant loci</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Visualization"><span class="nav-text">Visualization</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Regional-plot"><span class="nav-text">Regional plot</span></a></li></ol></li></ol></div></div></div></div></div></div><div class="main-content-footer"><footer class="footer mt-5 py-5 h-auto text-base text-third-text-color relative border-t-2 border-t-border-color"><div class="info-container py-3 text-center"><div class="text-center">&copy; <span>2024</span> - 2025&nbsp;&nbsp;<i class="fa-solid fa-heart fa-beat" style="--fa-animation-duration:0.5s;color:#f54545"></i>&nbsp;&nbsp;<a href="/">Yuchen</a><p class="post-count space-x-0.5"><span>53 posts in total</span></p></div><script data-swup-reload-script src="https://cn.vercount.one/js"></script><div class="relative text-center lg:absolute lg:right-[20px] lg:top-1/2 lg:-translate-y-1/2 lg:text-right"><span id="busuanzi_container_site_uv" class="lg:!block"><span class="text-sm">VISITOR COUNT</span> <span id="busuanzi_value_site_uv"></span> </span><span id="busuanzi_container_site_pv" class="lg:!block"><span class="text-sm">TOTAL PAGE VIEWS</span> <span id="busuanzi_value_site_pv"></span></span></div><div class="relative text-center lg:absolute lg:left-[20px] lg:top-1/2 lg:-translate-y-1/2 lg:text-left"><span class="lg:block text-sm">POWERED BY <?xml version="1.0" encoding="utf-8"?><!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"><svg class="relative top-[2px] inline-block align-baseline" version="1.1" id="圖層_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="1rem" height="1rem" viewBox="0 0 512 512" enable-background="new 0 0 512 512" xml:space="preserve"><path fill="#0E83CD" d="M256.4,25.8l-200,115.5L56,371.5l199.6,114.7l200-115.5l0.4-230.2L256.4,25.8z M349,354.6l-18.4,10.7l-18.6-11V275H200v79.6l-18.4,10.7l-18.6-11v-197l18.5-10.6l18.5,10.8V237h112v-79.6l18.5-10.6l18.5,10.8V354.6z"/></svg><a target="_blank" class="text-base" href="https://hexo.io">Hexo</a></span> <span class="text-sm lg:block">THEME&nbsp;<a class="text-base" target="_blank" href="https://github.com/EvanNotFound/hexo-theme-redefine">Redefine v2.8.2</a></span></div><div>Blog up for <span class="odometer" id="runtime_days"></span> days <span class="odometer" id="runtime_hours"></span> hrs <span class="odometer" id="runtime_minutes"></span> Min <span class="odometer" id="runtime_seconds"></span> Sec</div><script data-swup-reload-script>try{function odometer_init(){document.querySelectorAll(".odometer").forEach(e=>{new Odometer({el:e,format:"( ddd).dd",duration:200})})}odometer_init()}catch(e){}</script></div></footer></div></div><div class="post-tools"><div class="post-tools-container"><ul class="article-tools-list"><li class="right-bottom-tools page-aside-toggle"><i class="fa-regular fa-outdent"></i></li><li class="go-comment"><i class="fa-regular fa-comments"></i></li></ul></div></div><div class="right-side-tools-container"><div class="side-tools-container"><ul class="hidden-tools-list"><li class="right-bottom-tools tool-font-adjust-plus flex justify-center items-center"><i class="fa-regular fa-magnifying-glass-plus"></i></li><li class="right-bottom-tools tool-font-adjust-minus flex justify-center items-center"><i class="fa-regular fa-magnifying-glass-minus"></i></li><li class="right-bottom-tools tool-dark-light-toggle flex justify-center items-center"><i class="fa-regular fa-moon"></i></li><li class="right-bottom-tools tool-scroll-to-top flex justify-center items-center"><i class="fa-regular fa-arrow-up"></i></li><li class="right-bottom-tools tool-scroll-to-bottom flex justify-center items-center"><i class="fa-regular fa-arrow-down"></i></li></ul><ul class="visible-tools-list"><li class="right-bottom-tools toggle-tools-list flex justify-center items-center"><i class="fa-regular fa-cog fa-spin"></i></li></ul></div></div><div class="image-viewer-container"><img src=""></div></main><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/Swup.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupSlideTheme.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupScriptsPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupProgressPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupScrollPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupPreloadPlugin.min.js"></script><script>const swup=new Swup({plugins:[new SwupScriptsPlugin({optin:!0}),new SwupProgressPlugin,new SwupScrollPlugin({offset:80}),new SwupSlideTheme({mainElement:".main-content-body"}),new SwupPreloadPlugin],containers:["#swup"]})</script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/imageViewer.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/utils.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/main.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/navbarShrink.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/scrollTopBottom.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/lightDarkSwitch.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/categoryList.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/codeBlock.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/lazyload.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/runtime.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/odometer.min.js"></script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/assets/odometer-theme-minimal.css"><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/Typed.min.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/typed.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/minimasonry.min.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/masonry.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/tocToggle.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/toc.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/tabs.js" data-swup-reload-script></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/moment-with-locales.min.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/essays.js" data-swup-reload-script></script></body></html>