<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><meta name="keywords" content="Hexo Theme Redefine"><meta name="author" content="Yuchen"><link rel="preconnect" href="https://fonts.googleapis.com"><link rel="preconnect" href="https://fonts.gstatic.com" crossorigin><link rel="preconnect" href="https://cdn.jsdelivr.net" crossorigin><link rel="canonical" href="http://example.com/2025/03/06/super pangenome of vitis/"><meta name="robots" content="index,follow"><meta name="googlebot" content="index,follow"><meta name="revisit-after" content="1 days"><meta name="description" content="Hexo Theme Redefine, Redefine Your Hexo Journey."><meta property="og:type" content="article"><meta property="og:title" content="Super pangenome of vitas"><meta property="og:url" content="http://example.com/2025/03/06/Super%20pangenome%20of%20Vitis/index.html"><meta property="og:site_name" content="Hexo"><meta property="og:description" content="Hexo Theme Redefine, Redefine Your Hexo Journey."><meta property="og:locale" content="en_US"><meta property="og:image" content="http://example.com/images/redefine-og.webp"><meta property="article:published_time" content="2025-03-06T01:22:00.000Z"><meta property="article:modified_time" content="2025-03-06T12:22:47.359Z"><meta property="article:author" content="John Doe"><meta property="article:tag" content="Article"><meta property="article:tag" content="Pangenome"><meta name="twitter:card" content="summary"><meta name="twitter:image" content="http://example.com/images/redefine-og.webp"><link rel="icon" type="image/png" href="/images/profile.jpg" sizes="192x192"><link rel="apple-touch-icon" sizes="180x180" href="/images/profile.jpg"><meta name="theme-color" content="#A31F34"><link rel="shortcut icon" href="/images/profile.jpg"><title>Super pangenome of vitas | Steps Among the Stars</title><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/Chillax/chillax.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/css/build/tailwind.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/GeistMono/geist-mono.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/Geist/geist.css"><script id="hexo-configurations">window.config={hostname:"example.com",root:"/",language:"en"},window.theme={articles:{style:{font_size:"16px",line_height:1.5,image_border_radius:"14px",image_alignment:"center",image_caption:!1,link_icon:!0,delete_mask:!1,title_alignment:"left",headings_top_spacing:{h1:"3.2rem",h2:"2.4rem",h3:"1.9rem",h4:"1.6rem",h5:"1.4rem",h6:"1.3rem"}},word_count:{enable:!0,count:!0,min2read:!0},author_label:{enable:!1,auto:!1,list:[]},code_block:{copy:!0,style:"mac",highlight_theme:{light:"github",dark:"vs2015"},font:{enable:!1,family:null,url:null}},toc:{enable:!0,max_depth:6,number:!1,expand:!0,init_open:!0},copyright:{enable:!0,default:"CC BY-NC-ND"},lazyload:!0,pangu_js:!1,recommendation:{enable:!1,title:"推荐阅读",limit:3,mobile_limit:2,placeholder:"/images/wallhaven-wqery6-light.webp",skip_dirs:[]}},colors:{primary:"#A31F34",secondary:null,default_mode:"light"},global:{fonts:{chinese:{enable:!1,family:null,url:null},english:{enable:!1,family:null,url:null},title:{enable:!1,family:null,url:null}},content_max_width:"1000px",sidebar_width:"210px",hover:{shadow:!0,scale:!1},scroll_progress:{bar:!1,percentage:!1},website_counter:{url:"https://cn.vercount.one/js",enable:!0,site_pv:!0,site_uv:!0,post_pv:!0},single_page:!0,preloader:{enable:"ture",custom_message:"Welcome to my little corner of chaos and brilliance!"},open_graph:{enable:!0,image:"/images/redefine-og.webp",description:"Hexo Theme Redefine, Redefine Your Hexo Journey."},google_analytics:{enable:!1,id:null}},home_banner:{enable:!0,style:"fixed",image:{light:"/images/light.jpeg",dark:"/images/dark.png"},title:"Resonant Journals",subtitle:{text:["Welcome to my little corner of chaos and brilliance!"],hitokoto:{enable:!1,show_author:!1,api:"https://v1.hitokoto.cn"},typing_speed:20,backing_speed:80,starting_delay:500,backing_delay:1500,loop:!1,smart_backspace:!0},text_color:{light:"#fff",dark:"#d1d1b6"},text_style:{title_size:"2.8rem",subtitle_size:"1.5rem",line_height:1.2},custom_font:{enable:!1,family:null,url:null},social_links:{enable:"ture",style:"default",links:{github:null,instagram:null,zhihu:null,twitter:null,email:"yuchen.starfall@outlook.com"},qrs:{weixin:null}}},plugins:{feed:{enable:!1},aplayer:{enable:!1,type:"fixed",audios:[{name:null,artist:null,url:null,cover:null,lrc:null}]},mermaid:{enable:!1,version:"11.4.1"}},version:"2.8.2",navbar:{auto_hide:"ture",color:{left:"#f78736",right:"#367df7",transparency:35},width:{home:"1200px",pages:"1000px"},links:{Home:{path:"/",icon:"fa-regular fa-house"},Archives:{path:"/archives",icon:"fa-regular fa-archive"},Categories:{icon:"fa-solid fa-folder",path:"/categories/"},Tags:{icon:"fa-solid fa-tags",path:"/tags/"}},search:{enable:!1,preload:!0}},page_templates:{friends_column:2,tags_style:"blur"},home:{sidebar:{enable:!0,position:"left",first_item:"menu",announcement:"Ephemeral Trails",show_on_mobile:!0,links:{Archives:{path:"/archives",icon:"fa-regular fa-archive"},Tags:{path:"/tags",icon:"fa-regular fa-tags"},Categories:{path:"/categories",icon:"fa-regular fa-folder"}}},article_date_format:"auto",excerpt_length:200,categories:{enable:!0,limit:3},tags:{enable:!0,limit:3}},footerStart:"2024/12/5 00:00:00"},window.lang_ago={second:"%s seconds ago",minute:"%s minutes ago",hour:"%s hours ago",day:"%s days ago",week:"%s weeks ago",month:"%s months ago",year:"%s years ago"},window.data={masonry:!0}</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/fontawesome.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/brands.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/solid.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/regular.min.css"><meta name="generator" content="Hexo 7.3.0"></head><body><div class="progress-bar-container"><span class="pjax-progress-bar"></span></div><main class="page-container" id="swup"><div class="main-content-container flex flex-col justify-between min-h-dvh"><div class="main-content-header"><header class="navbar-container px-6 md:px-12"><div class="navbar-content transition-navbar"><div class="left"><a class="logo-title" href="/">Steps Among the Stars</a></div><div class="right"><div class="desktop"><ul class="navbar-list"><li class="navbar-item"><a href="/"><i class="fa-regular fa-house fa-fw"></i> HOME</a></li><li class="navbar-item"><a href="/archives"><i class="fa-regular fa-archive fa-fw"></i> ARCHIVES</a></li><li class="navbar-item"><a href="/categories/"><i class="fa-solid fa-folder fa-fw"></i> CATEGORIES</a></li><li class="navbar-item"><a href="/tags/"><i class="fa-solid fa-tags fa-fw"></i> TAGS</a></li></ul></div><div class="mobile"><div class="icon-item navbar-bar"><div class="navbar-bar-middle"></div></div></div></div></div><div class="navbar-drawer h-dvh w-full absolute top-0 left-0 bg-background-color flex flex-col justify-between"><ul class="drawer-navbar-list flex flex-col px-4 justify-center items-start"><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/"><span>HOME </span><i class="fa-regular fa-house fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/archives"><span>ARCHIVES </span><i class="fa-regular fa-archive fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/categories/"><span>CATEGORIES </span><i class="fa-solid fa-folder fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/tags/"><span>TAGS </span><i class="fa-solid fa-tags fa-sm fa-fw"></i></a></li></ul><div class="statistics flex justify-around my-2.5"><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/tags"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">48</div><div class="label text-third-text-color text-sm">Tags</div></a><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/categories"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">6</div><div class="label text-third-text-color text-sm">Categories</div></a><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/archives"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">51</div><div class="label text-third-text-color text-sm">Posts</div></a></div></div><div class="window-mask"></div></header></div><div class="main-content-body transition-fade-up"><div class="main-content"><div class="post-page-container flex relative justify-between box-border w-full h-full"><div class="article-content-container"><div class="article-title relative w-full"><img src="/img/img-pangenome/1.png" alt="Super pangenome of vitas" class="w-full h-60 sm:h-72 md:h-80 object-cover sm:rounded-t-large dark:brightness-75"><div class="w-full flex items-center absolute bottom-0 justify-start"><h1 class="article-title-cover text-center mx-6 my-6 text-second-text-color bg-background-color-transparent px-4 py-3 text-3xl sm:text-4xl md:text-5xl font-semibold backdrop-blur-lg rounded-xl border border-border-color">Super pangenome of vitas</h1></div></div><div class="article-header flex flex-row gap-2 items-center px-2 sm:px-6 md:px-8"><div class="avatar w-[46px] h-[46px] flex-shrink-0 rounded-medium border border-border-color p-[1px]"><img src="/images/profile.jpg"></div><div class="info flex flex-col justify-between"><div class="author flex items-center"><span class="name text-default-text-color text-lg font-semibold">Yuchen</span></div><div class="meta-info"><div class="article-meta-info"><span class="article-date article-meta-item"><i class="fa-regular fa-pen-fancy"></i>&nbsp; <span class="desktop">2025-03-06 09:22</span> <span class="mobile">2025-03-06 09:22</span> <span class="hover-info">Created</span> </span><span class="article-date article-meta-item"><i class="fa-regular fa-wrench"></i>&nbsp; <span class="desktop">2025-03-06 20:22:47</span> <span class="mobile">2025-03-06 20:22:47</span> <span class="hover-info">Updated</span> </span><span class="article-categories article-meta-item"><i class="fa-regular fa-folders"></i>&nbsp;<ul><li><a href="/categories/Literature/">Literature</a>&nbsp;</li></ul></span><span class="article-tags article-meta-item"><i class="fa-regular fa-tags"></i>&nbsp;<ul><li><a href="/tags/Article/">Article</a>&nbsp;</li><li>| <a href="/tags/Pangenome/">Pangenome</a>&nbsp;</li></ul></span><span class="article-pv article-meta-item"><i class="fa-regular fa-eye"></i>&nbsp;<span id="busuanzi_value_page_pv"></span></span></div></div></div></div><div class="article-content markdown-body px-2 sm:px-6 md:px-8 pb-8"><h3 id="Assemble-evaluatet"><a href="#Assemble-evaluatet" class="headerlink" title="Assemble evaluatet"></a>Assemble evaluatet</h3><p>针对葡萄通过无性繁殖（如扦插）保持遗传一致性的特点（基因组中存在大量杂合位点），之前发布的T2T基因组PN40024组装将两个单倍型合并，导致杂合位点被错误合并，影响准确性。这种合并可能掩盖真实的遗传变异，限制了对葡萄遗传多样性和复杂性的理解。因此进行Haplotype-resolved，</p><h4 id="Assemble"><a href="#Assemble" class="headerlink" title="Assemble"></a>Assemble</h4><ul><li><p>PacBio HiFi reads (308x)，Oxford Nanopore Technologies (ONT) ultra-long (N50 &gt; 100 kb) reads (117x) and Hi-C reads (180x) for T2T genome assembly.（Hi-C数据用于将序列定位到染色体上）</p></li><li><p>证明组装的完整性与精确性：读段比对、Hi-C相互作用图谱、BUSCO、长末端重复序列（LTR）组装指数（LAI）和质量值（QV）</p><ul><li><p>HI-C</p><ul><li><p>通过三维基因组相互作用数据评估基因组组装的准确性和连续性，交联后的测序数据反应，两个远端DNA片段被连接并测序，反映它们在三维空间中的接近性。</p></li><li><p>Hi-C 互作矩阵用于评估基因组组装效果，因为它反映了 <strong>DNA 片段的空间接触模式</strong>。如果基因组组装正确，相邻片段的 Hi-C 互作应当更强，在矩阵中形成清晰的对角线结构。如果组装错误，可能会看到错位、倒置或异常互作信号。</p></li></ul></li><li><p><img lazyload src="/images/loading.svg" data-src="/img/img-pangenome/6.png"></p></li><li><p>BUSCO</p><ul><li><p>通过评估进化上保守的单拷贝直系同源基因在组装结果中的存在和完整性，来确定基因组、转录组或蛋白质组的质量。存在一个数据集，包含该谱系中普遍存在的单拷贝直系同源基因（通常数百到数千个）。评价的假设即这些基因在进化中高度保守，且在正常条件下应为单拷贝存在。</p></li><li><p>每个BUSCO基因根据匹配情况分为四类：</p><ol><li><strong>Complete（完整单拷贝）</strong>：<ul><li>基因长度与参考接近，且预测为单拷贝。</li><li>表明组装成功捕获了该基因。</li></ul></li><li><strong>Complete Duplicated（完整重复）</strong>：<ul><li>同一基因在组装中出现多次，可能因组装错误（如过度合并等位基因）或真实生物学重复。</li><li>重复率高可能提示组装质量问题。</li></ul></li><li><strong>Fragmented（碎片化）</strong>：<ul><li>仅部分基因被检测到（如缺失外显子或短片段）。</li><li>可能由组装断裂或预测工具灵敏度不足导致。</li></ul></li><li><strong>Missing（缺失）</strong>：<ul><li>完全未检测到该基因，可能因组装不完整或物种特异性丢失。</li></ul></li></ol></li><li><p>LAI</p><ul><li><p>它通过量化基因组中完整 LTR-RTs 的比例，反映组装的连续性和重复区域的准确性。基本假设为通过量化基因组中完整 LTR-RTs 的比例，反映组装的连续性和重复区域的准确性</p></li><li><p>TR-RTs 是真核生物基因组中常见的重复元件，由两个长末端重复序列（LTR）和一个内部蛋白编码区组成。完整 LTR-RT 的特征：两个 LTR 序列高度同源（对称）、内部区域完整、总长度符合预期（通常数 kb 到数十 kb）。</p></li><li><p><strong>分类 LTR-RTs</strong>：</p><ol><li><strong>完整（Intact）</strong>：两个 LTR 对称且内部区域未被截断。</li><li><strong>碎片化（Fragmented）</strong>：LTR 或内部区域不完整。</li><li><strong>单 LTR（Solo-LTR）</strong>：仅剩单个 LTR（可能由重组事件导致）。</li></ol></li><li><p>评分：基于基因组的滑动窗口计算的（通常是 <strong>1 Mb 窗口大小</strong>）</p><ul><li>LAI &lt; 10：低质量组装</li><li>LAI 10-20：中等质量</li><li>LAI &gt; 20：高质量基因组（通常接近参考基因组）</li></ul></li></ul></li><li><p><img lazyload src="/images/loading.svg" data-src="/img/img-pangenome/7.png"></p></li><li><p>QV</p><ul><li><p>质量值（<strong>QV，Quality Value</strong>）是基因组组装质量评估的重要指标，它衡量了组装序列的 <strong>准确性（accuracy）</strong>，通常用于评估<strong>碱基级别的错误率</strong>。$QV &#x3D; -10 \times \log_{10}(P_e)$</p></li><li><table><thead><tr><th><strong>QV 值</strong></th><th><strong>碱基错误率 (<strong>P_e</strong>)</strong></th><th><strong>组装准确率</strong></th></tr></thead><tbody><tr><td>QV30</td><td>1 &#x2F; 1,000 (0.1%)</td><td>99.9%</td></tr><tr><td>QV40</td><td>1 &#x2F; 10,000 (0.01%)</td><td>99.99%</td></tr><tr><td>QV50</td><td>1 &#x2F; 100,000 (0.001%)</td><td>99.999%</td></tr></tbody></table></li></ul></li><li><p>k-mer</p><ul><li><strong>k-mer</strong> 是指一段 DNA 序列中长度为 <strong>k</strong> 的连续子序列。以下图频率分布可视化为例，k为21，计算测序的原始 reads中的21个碱基片段长度，即reads每21个为一组。</li><li>​ Illumina 读长通常是 <strong>100-150 bp</strong>，选择 <strong>k&#x3D;21</strong> 能确保大部分 reads 能产生多个 k-mers，提供良好的覆盖度。如果使用长读测序（如 PacBio, Nanopore），可以使用更大的 k 值，如 k&#x3D;31 或 k&#x3D;51，以更好地利用长读的特点。</li><li>如果 k 选择 <strong>太短</strong>（如 k&#x3D;5），很多 k-mers 会在基因组中频繁出现，导致很难区分不同区域（因为短 k-mer 很容易在不同位置重复）。如果 k 选择 <strong>太长</strong>（如 k&#x3D;100），可能会有太多唯一的 k-mers，导致低频 k-mers 增多，增加随机噪音。</li></ul></li><li><p><img lazyload src="/images/loading.svg" data-src="/img/img-pangenome/5.png"></p></li></ul></li></ul></li></ul><hr><table><thead><tr><th><strong>指标</strong></th><th><strong>作用</strong></th><th><strong>适用场景</strong></th></tr></thead><tbody><tr><td><strong>QV（质量值）</strong></td><td>评估 <strong>碱基级</strong> 精确度（错误率）</td><td>关注高精度测序（如 PacBio HiFi），衡量序列错误率</td></tr><tr><td><strong>BUSCO</strong></td><td>评估 <strong>基因组完整性</strong>（基因保守性）</td><td>适用于真核&#x2F;细菌基因组，衡量保守基因的缺失情况</td></tr><tr><td><strong>Contig N50 &#x2F; Scaffold N50</strong></td><td>评估 <strong>组装连续性</strong>（片段长度）</td><td>衡量 contig&#x2F;scaffold 组装的完整性</td></tr><tr><td><strong>Hi-C 互作矩阵</strong></td><td>评估 <strong>染色体结构正确性</strong></td><td>适用于染色体级别组装，检查错拼、倒置、染色体锚定</td></tr><tr><td><strong>LAI（Ligation-Assisted Index）</strong></td><td>评估 <strong>基因组的长重复序列完整性</strong></td><td>适用于植物、动物基因组，衡量异染色质和重复区域的正确组装</td></tr><tr><td><strong>k-mer 一致性</strong></td><td>评估 <strong>组装的 k-mer 覆盖率</strong>，检查缺失&#x2F;错组装</td><td>适用于短读长（Illumina）数据，衡量基因组组装与原始数据的一致性</td></tr></tbody></table><p><img lazyload src="/images/loading.svg" data-src="/img/img-pangenome/2.png"></p><h4 id="Centromere"><a href="#Centromere" class="headerlink" title="Centromere"></a>Centromere</h4><ul><li>chip-seq证明组装基因组着丝粒长度大小符合</li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-pangenome/3.png"></p><ul><li>进行着丝粒的序列特点分析 – 着丝粒位置存在大量的LTR插入</li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-pangenome/4.png"></p><h3 id="Genetic-diversity"><a href="#Genetic-diversity" class="headerlink" title="Genetic diversity"></a>Genetic diversity</h3><ul><li>Maximum likelihood phylogenetic tree.</li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-pangenome/8.png"></p><ul><li>Principal component analysis<ul><li>揭示遗传结构和个体关系的核心工具，能够直观展示群体分层、连续基因流和混合事件</li></ul></li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-pangenome/9.png"></p><ul><li>ADMIXTURE ancestry analysis<ul><li>推断个体的祖先成分和群体的遗传结构</li><li>核心假设：离散祖先群体，每个群体内部满足HWE（哈代温伯格平衡），假设SNP位点间无连锁不平衡（LD），并且群体数（K）进行预设</li></ul></li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-pangenome/10.png"></p><h3 id="Hybridization-history"><a href="#Hybridization-history" class="headerlink" title="Hybridization history"></a>Hybridization history</h3><p>全基因组溯祖系统发育树 – 三个大陆（欧洲、北美、东亚）和圆叶葡萄（Muscadine）的单倍型之间的分离</p><p>与上面的二倍体物种树，大多数情况下是一致的，但也存在一些例外 – 一个<strong>Vitis pseudoreticulata</strong>与<strong>V. vinifera</strong>的杂交品种V125在二倍体树中被归类到欧洲分支，而在单倍体树中，它的两个单倍型分别被归类到欧洲和东亚分支</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-pangenome/11.png"></p><h3 id="Pangenome-analysis"><a href="#Pangenome-analysis" class="headerlink" title="Pangenome analysis"></a>Pangenome analysis</h3><p>将所有基因分类为64,517个基因家族，随着更多基因组的加入，基因家族的数量迅速增加，并在n &#x3D; 125个基因组时趋于平稳，表明72个品种的泛基因组是闭合的</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-pangenome/12.png"></p><p>pan- NLRome analysis</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-pangenome/13.png"></p><h3 id="SVs"><a href="#SVs" class="headerlink" title="SVs"></a>SVs</h3><p>大规模基因组变异，如结构变异（SVs），是植物进化和驯化的主要驱动力</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-pangenome/14.png"></p><h4 id="SV–eQTLs"><a href="#SV–eQTLs" class="headerlink" title="SV–eQTLs"></a>SV–eQTLs</h4><p>关联全基因组结构变异与基因表达量分析</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-pangenome/15.png"></p><h3 id="源码"><a href="#源码" class="headerlink" title="源码"></a>源码</h3><p>[]: <a class="link" target="_blank" rel="noopener" href="https://github.com/NotebookOFXiaoMing/SuperPangenomeofGrapevines">https://github.com/NotebookOFXiaoMing/SuperPangenomeofGrapevines<i class="fa-solid fa-arrow-up-right ml-[0.2em] font-light align-text-top text-[0.7em] link-icon"></i></a>	“源码”</p><p>注意，使用了<strong>Snakemake</strong>工作流管理系统，Snakemake是一个用于创建和管理生物信息学分析流程的工具，它基于Python语言，并利用类似于Makefile的语法来定义分析步骤和依赖关系。SMK文件是Snakemake流程的配置文件，其中包含了分析流程的规则、输入文件、输出文件以及执行命令。</p><div class="code-container" data-rel="Plaintext"><figure class="iseeu highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br></pre></td><td class="code"><pre><span class="line"># 定义参考基因组路径</span><br><span class="line">REFERENCE = &quot;reference.fa&quot;</span><br><span class="line"></span><br><span class="line"># 规则1：比对FASTQ文件</span><br><span class="line">rule align_reads:</span><br><span class="line">    input:</span><br><span class="line">        &quot;data/&#123;sample&#125;.fastq&quot;</span><br><span class="line">    output:</span><br><span class="line">        &quot;results/&#123;sample&#125;.bam&quot;</span><br><span class="line">    shell:</span><br><span class="line">        &quot;bwa mem &#123;REFERENCE&#125; &#123;input&#125; | samtools view -Sb - &gt; &#123;output&#125;&quot;</span><br><span class="line"></span><br><span class="line"># 规则2：对BAM文件进行排序</span><br><span class="line">rule sort_bam:</span><br><span class="line">    input:</span><br><span class="line">        &quot;results/&#123;sample&#125;.bam&quot;</span><br><span class="line">    output:</span><br><span class="line">        &quot;results/&#123;sample&#125;.sorted.bam&quot;</span><br><span class="line">    shell:</span><br><span class="line">        &quot;samtools sort &#123;input&#125; -o &#123;output&#125;&quot;</span><br><span class="line"></span><br><span class="line"># 规则3：生成索引文件</span><br><span class="line">rule index_bam:</span><br><span class="line">    input:</span><br><span class="line">        &quot;results/&#123;sample&#125;.sorted.bam&quot;</span><br><span class="line">    output:</span><br><span class="line">        &quot;results/&#123;sample&#125;.sorted.bam.bai&quot;</span><br><span class="line">    shell:</span><br><span class="line">        &quot;samtools index &#123;input&#125;&quot;</span><br></pre></td></tr></table></figure></div><ul><li><code>&#123;sample&#125;</code>是一种**通配符（wildcard）**语法。它用于动态匹配文件名或路径中的变量部分，使得规则可以灵活地应用于多个样本或文件，而不需要为每个样本单独编写规则。</li><li><code>&#123;REFERENCE&#125;</code>通常是一个<strong>变量（variable）</strong>，而不是通配符（wildcard）。它用于存储和引用某个值（例如文件路径、参数等），以便在规则中重复使用。变量在Snakemake中通过赋值定义，并在规则中通过<code>&#123;&#125;</code>语法引用。</li></ul></div><div class="post-copyright-info w-full my-8 px-2 sm:px-6 md:px-8"><div class="article-copyright-info-container"><ul><li><strong>Title:</strong> Super pangenome of vitas</li><li><strong>Author:</strong> Yuchen</li><li><strong>Created at :</strong> 2025-03-06 09:22:00</li><li><strong>Updated at :</strong> 2025-03-06 20:22:47</li><li><strong>Link:</strong> https://yuchen-starfall.github.io/2025/03/06/Super pangenome of Vitis/</li><li><strong>License: </strong>This work is licensed under <a class="license" target="_blank" rel="noopener" href="https://creativecommons.org/licenses/by-nc-sa/4.0">CC BY-NC-SA 4.0</a>.</li></ul></div></div><ul class="post-tags-box text-lg mt-1.5 flex-wrap justify-center flex md:hidden"><li class="tag-item mx-0.5"><a href="/tags/Article/">#Article</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/Pangenome/">#Pangenome</a>&nbsp;</li></ul><div class="article-nav my-8 flex justify-between items-center px-2 sm:px-6 md:px-8"><div class="article-prev border-border-color shadow-redefine-flat shadow-shadow-color-2 rounded-medium px-4 py-2 hover:shadow-redefine-flat-hover hover:shadow-shadow-color-2"><a class="prev" rel="prev" href="/2025/03/07/Deep%20learn%20--%20Multilayer%20Perceptrons%20/"><span class="left arrow-icon flex justify-center items-center"><i class="fa-solid fa-chevron-left"></i> </span><span class="title flex justify-center items-center"><span class="post-nav-title-item">Deep-learn -- Multilayer Perceptrons</span> <span class="post-nav-item">Prev posts</span></span></a></div><div class="article-next border-border-color shadow-redefine-flat shadow-shadow-color-2 rounded-medium px-4 py-2 hover:shadow-redefine-flat-hover hover:shadow-shadow-color-2"><a class="next" rel="next" href="/2025/03/04/Phylogeny/"><span class="title flex justify-center items-center"><span class="post-nav-title-item">Phylogenetic tree</span> <span class="post-nav-item">Next posts</span> </span><span class="right arrow-icon flex justify-center items-center"><i class="fa-solid fa-chevron-right"></i></span></a></div></div><div class="comment-container px-2 sm:px-6 md:px-8 pb-8"><div class="comments-container mt-10 w-full"><div id="comment-anchor" class="w-full h-2.5"></div><div class="comment-area-title w-full my-1.5 md:my-2.5 text-xl md:text-3xl font-bold">Comments</div><div id="giscus-container"></div><script data-swup-reload-script defer>async function loadGiscus(){const t={src:"https://giscus.app/client.js","data-repo":"Yuchen-starfall/Yuchen-starfall.github.io","data-repo-id":"R_kgDONY6r-Q","data-category":"Announcements# Github discussion category","data-category-id":"DIC_kwDONY6r-c4Ck8Zm","data-mapping":"pathname","data-strict":"0","data-reactions-enabled":"1","data-emit-metadata":"1","data-theme":"preferred_color_scheme","data-lang":"en","data-input-position":"bottom","data-loading":"lazy",crossorigin:"anonymous",async:!0},a=document.createElement("script");for(const e in t)a.setAttribute(e,t[e]);document.getElementById("giscus-container").appendChild(a)}{let t=setTimeout(()=>{loadGiscus(),clearTimeout(t)},1e3)}</script></div></div></div><div class="toc-content-container"><div class="post-toc-wrap"><div class="post-toc"><div class="toc-title">On this page</div><div class="page-title">Super pangenome of vitas</div><ol class="nav"><li class="nav-item nav-level-3"><a class="nav-link" href="#Assemble-evaluatet"><span class="nav-text">Assemble evaluatet</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Assemble"><span class="nav-text">Assemble</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Centromere"><span class="nav-text">Centromere</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Genetic-diversity"><span class="nav-text">Genetic diversity</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Hybridization-history"><span class="nav-text">Hybridization history</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Pangenome-analysis"><span class="nav-text">Pangenome analysis</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#SVs"><span class="nav-text">SVs</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#SV%E2%80%93eQTLs"><span class="nav-text">SV–eQTLs</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%BA%90%E7%A0%81"><span class="nav-text">源码</span></a></li></ol></div></div></div></div></div></div><div class="main-content-footer"><footer class="footer mt-5 py-5 h-auto text-base text-third-text-color relative border-t-2 border-t-border-color"><div class="info-container py-3 text-center"><div class="text-center">&copy; <span>2024</span> - 2025&nbsp;&nbsp;<i class="fa-solid fa-heart fa-beat" style="--fa-animation-duration:0.5s;color:#f54545"></i>&nbsp;&nbsp;<a href="/">Yuchen</a><p class="post-count space-x-0.5"><span>51 posts in total</span></p></div><script data-swup-reload-script src="https://cn.vercount.one/js"></script><div class="relative text-center lg:absolute lg:right-[20px] lg:top-1/2 lg:-translate-y-1/2 lg:text-right"><span id="busuanzi_container_site_uv" class="lg:!block"><span class="text-sm">VISITOR COUNT</span> <span id="busuanzi_value_site_uv"></span> </span><span id="busuanzi_container_site_pv" class="lg:!block"><span class="text-sm">TOTAL PAGE VIEWS</span> <span id="busuanzi_value_site_pv"></span></span></div><div class="relative text-center lg:absolute lg:left-[20px] lg:top-1/2 lg:-translate-y-1/2 lg:text-left"><span class="lg:block text-sm">POWERED BY <?xml version="1.0" encoding="utf-8"?><!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"><svg class="relative top-[2px] inline-block align-baseline" version="1.1" id="圖層_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="1rem" height="1rem" viewBox="0 0 512 512" enable-background="new 0 0 512 512" xml:space="preserve"><path fill="#0E83CD" d="M256.4,25.8l-200,115.5L56,371.5l199.6,114.7l200-115.5l0.4-230.2L256.4,25.8z M349,354.6l-18.4,10.7l-18.6-11V275H200v79.6l-18.4,10.7l-18.6-11v-197l18.5-10.6l18.5,10.8V237h112v-79.6l18.5-10.6l18.5,10.8V354.6z"/></svg><a target="_blank" class="text-base" href="https://hexo.io">Hexo</a></span> <span class="text-sm lg:block">THEME&nbsp;<a class="text-base" target="_blank" href="https://github.com/EvanNotFound/hexo-theme-redefine">Redefine v2.8.2</a></span></div><div>Blog up for <span class="odometer" id="runtime_days"></span> days <span class="odometer" id="runtime_hours"></span> hrs <span class="odometer" id="runtime_minutes"></span> Min <span class="odometer" id="runtime_seconds"></span> Sec</div><script data-swup-reload-script>try{function odometer_init(){document.querySelectorAll(".odometer").forEach(e=>{new Odometer({el:e,format:"( ddd).dd",duration:200})})}odometer_init()}catch(e){}</script></div></footer></div></div><div class="post-tools"><div class="post-tools-container"><ul class="article-tools-list"><li class="right-bottom-tools page-aside-toggle"><i class="fa-regular fa-outdent"></i></li><li class="go-comment"><i class="fa-regular fa-comments"></i></li></ul></div></div><div class="right-side-tools-container"><div class="side-tools-container"><ul class="hidden-tools-list"><li class="right-bottom-tools tool-font-adjust-plus flex justify-center items-center"><i class="fa-regular fa-magnifying-glass-plus"></i></li><li class="right-bottom-tools tool-font-adjust-minus flex justify-center items-center"><i class="fa-regular fa-magnifying-glass-minus"></i></li><li class="right-bottom-tools tool-dark-light-toggle flex justify-center items-center"><i class="fa-regular fa-moon"></i></li><li class="right-bottom-tools tool-scroll-to-top flex justify-center items-center"><i class="fa-regular fa-arrow-up"></i></li><li class="right-bottom-tools tool-scroll-to-bottom flex justify-center items-center"><i class="fa-regular fa-arrow-down"></i></li></ul><ul class="visible-tools-list"><li class="right-bottom-tools toggle-tools-list flex justify-center items-center"><i class="fa-regular fa-cog fa-spin"></i></li></ul></div></div><div class="image-viewer-container"><img src=""></div></main><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/Swup.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupSlideTheme.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupScriptsPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupProgressPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupScrollPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupPreloadPlugin.min.js"></script><script>const swup=new Swup({plugins:[new SwupScriptsPlugin({optin:!0}),new SwupProgressPlugin,new SwupScrollPlugin({offset:80}),new SwupSlideTheme({mainElement:".main-content-body"}),new SwupPreloadPlugin],containers:["#swup"]})</script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/imageViewer.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/utils.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/main.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/navbarShrink.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/scrollTopBottom.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/lightDarkSwitch.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/categoryList.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/codeBlock.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/lazyload.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/runtime.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/odometer.min.js"></script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/assets/odometer-theme-minimal.css"><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/Typed.min.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/typed.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/minimasonry.min.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/masonry.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/tocToggle.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/toc.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/tabs.js" data-swup-reload-script></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/moment-with-locales.min.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/essays.js" data-swup-reload-script></script></body></html>