<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><meta name="keywords" content="Hexo Theme Redefine"><meta name="author" content="Yuchen"><link rel="preconnect" href="https://fonts.googleapis.com"><link rel="preconnect" href="https://fonts.gstatic.com" crossorigin><link rel="preconnect" href="https://cdn.jsdelivr.net" crossorigin><link rel="canonical" href="http://example.com/2025/04/08/analysis of population genetic structure/"><meta name="robots" content="index,follow"><meta name="googlebot" content="index,follow"><meta name="revisit-after" content="1 days"><meta name="description" content="Hexo Theme Redefine, Redefine Your Hexo Journey."><meta property="og:type" content="article"><meta property="og:title" content="Analysis of Population Genetic Structure"><meta property="og:url" content="http://example.com/2025/04/08/Analysis%20of%20Population%20Genetic%20Structure/index.html"><meta property="og:site_name" content="Hexo"><meta property="og:description" content="Hexo Theme Redefine, Redefine Your Hexo Journey."><meta property="og:locale" content="en_US"><meta property="og:image" content="http://example.com/images/redefine-og.webp"><meta property="article:published_time" content="2025-04-08T12:16:00.000Z"><meta property="article:modified_time" content="2025-05-05T12:37:47.688Z"><meta property="article:author" content="John Doe"><meta property="article:tag" content="course"><meta property="article:tag" content="Hardy-Weinberg"><meta property="article:tag" content="Population Genetic Structure"><meta name="twitter:card" content="summary"><meta name="twitter:image" content="http://example.com/images/redefine-og.webp"><link rel="icon" type="image/png" href="/images/profile.jpg" sizes="192x192"><link rel="apple-touch-icon" sizes="180x180" href="/images/profile.jpg"><meta name="theme-color" content="#A31F34"><link rel="shortcut icon" href="/images/profile.jpg"><title>Analysis of Population Genetic Structure | Steps Among the Stars</title><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/Chillax/chillax.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/css/build/tailwind.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/GeistMono/geist-mono.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fonts/Geist/geist.css"><script id="hexo-configurations">window.config={hostname:"example.com",root:"/",language:"en"},window.theme={articles:{style:{font_size:"16px",line_height:1.5,image_border_radius:"14px",image_alignment:"center",image_caption:!1,link_icon:!0,delete_mask:!1,title_alignment:"left",headings_top_spacing:{h1:"3.2rem",h2:"2.4rem",h3:"1.9rem",h4:"1.6rem",h5:"1.4rem",h6:"1.3rem"}},word_count:{enable:!0,count:!0,min2read:!0},author_label:{enable:!1,auto:!1,list:[]},code_block:{copy:!0,style:"mac",highlight_theme:{light:"github",dark:"vs2015"},font:{enable:!1,family:null,url:null}},toc:{enable:!0,max_depth:6,number:!1,expand:!0,init_open:!0},copyright:{enable:!0,default:"CC BY-NC-ND"},lazyload:!0,pangu_js:!1,recommendation:{enable:!1,title:"推荐阅读",limit:3,mobile_limit:2,placeholder:"/images/wallhaven-wqery6-light.webp",skip_dirs:[]}},colors:{primary:"#A31F34",secondary:null,default_mode:"light"},global:{fonts:{chinese:{enable:!1,family:null,url:null},english:{enable:!1,family:null,url:null},title:{enable:!1,family:null,url:null}},content_max_width:"1000px",sidebar_width:"210px",hover:{shadow:!0,scale:!1},scroll_progress:{bar:!1,percentage:!1},website_counter:{url:"https://cn.vercount.one/js",enable:!0,site_pv:!0,site_uv:!0,post_pv:!0},single_page:!0,preloader:{enable:"ture",custom_message:"Welcome to my little corner of chaos and brilliance!"},open_graph:{enable:!0,image:"/images/redefine-og.webp",description:"Hexo Theme Redefine, Redefine Your Hexo Journey."},google_analytics:{enable:!1,id:null}},home_banner:{enable:!0,style:"fixed",image:{light:"/images/light.jpeg",dark:"/images/dark.png"},title:"Resonant Journals",subtitle:{text:["Welcome to my little corner of chaos and brilliance!"],hitokoto:{enable:!1,show_author:!1,api:"https://v1.hitokoto.cn"},typing_speed:20,backing_speed:80,starting_delay:500,backing_delay:1500,loop:!1,smart_backspace:!0},text_color:{light:"#fff",dark:"#d1d1b6"},text_style:{title_size:"2.8rem",subtitle_size:"1.5rem",line_height:1.2},custom_font:{enable:!1,family:null,url:null},social_links:{enable:"ture",style:"default",links:{github:null,instagram:null,zhihu:null,twitter:null,email:"yuchen.starfall@outlook.com"},qrs:{weixin:null}}},plugins:{feed:{enable:!1},aplayer:{enable:!1,type:"fixed",audios:[{name:null,artist:null,url:null,cover:null,lrc:null}]},mermaid:{enable:!1,version:"11.4.1"}},version:"2.8.2",navbar:{auto_hide:"ture",color:{left:"#f78736",right:"#367df7",transparency:35},width:{home:"1200px",pages:"1000px"},links:{Home:{path:"/",icon:"fa-regular fa-house"},Archives:{path:"/archives",icon:"fa-regular fa-archive"},Categories:{icon:"fa-solid fa-folder",path:"/categories/"},Tags:{icon:"fa-solid fa-tags",path:"/tags/"}},search:{enable:!1,preload:!0}},page_templates:{friends_column:2,tags_style:"blur"},home:{sidebar:{enable:!0,position:"left",first_item:"menu",announcement:"Ephemeral Trails",show_on_mobile:!0,links:{Archives:{path:"/archives",icon:"fa-regular fa-archive"},Tags:{path:"/tags",icon:"fa-regular fa-tags"},Categories:{path:"/categories",icon:"fa-regular fa-folder"}}},article_date_format:"auto",excerpt_length:200,categories:{enable:!0,limit:3},tags:{enable:!0,limit:3}},footerStart:"2024/12/5 00:00:00"},window.lang_ago={second:"%s seconds ago",minute:"%s minutes ago",hour:"%s hours ago",day:"%s days ago",week:"%s weeks ago",month:"%s months ago",year:"%s years ago"},window.data={masonry:!0}</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/fontawesome.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/brands.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/solid.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/fontawesome/regular.min.css"><meta name="generator" content="Hexo 7.3.0"></head><body><div class="progress-bar-container"><span class="pjax-progress-bar"></span></div><main class="page-container" id="swup"><div class="main-content-container flex flex-col justify-between min-h-dvh"><div class="main-content-header"><header class="navbar-container px-6 md:px-12"><div class="navbar-content transition-navbar"><div class="left"><a class="logo-title" href="/">Steps Among the Stars</a></div><div class="right"><div class="desktop"><ul class="navbar-list"><li class="navbar-item"><a href="/"><i class="fa-regular fa-house fa-fw"></i> HOME</a></li><li class="navbar-item"><a href="/archives"><i class="fa-regular fa-archive fa-fw"></i> ARCHIVES</a></li><li class="navbar-item"><a href="/categories/"><i class="fa-solid fa-folder fa-fw"></i> CATEGORIES</a></li><li class="navbar-item"><a href="/tags/"><i class="fa-solid fa-tags fa-fw"></i> TAGS</a></li></ul></div><div class="mobile"><div class="icon-item navbar-bar"><div class="navbar-bar-middle"></div></div></div></div></div><div class="navbar-drawer h-dvh w-full absolute top-0 left-0 bg-background-color flex flex-col justify-between"><ul class="drawer-navbar-list flex flex-col px-4 justify-center items-start"><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/"><span>HOME </span><i class="fa-regular fa-house fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/archives"><span>ARCHIVES </span><i class="fa-regular fa-archive fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/categories/"><span>CATEGORIES </span><i class="fa-solid fa-folder fa-sm fa-fw"></i></a></li><li class="drawer-navbar-item text-base my-1.5 flex flex-col w-full"><a class="py-1.5 px-2 flex flex-row items-center justify-between gap-1 hover:!text-primary active:!text-primary text-2xl font-semibold group border-b border-border-color hover:border-primary w-full" href="/tags/"><span>TAGS </span><i class="fa-solid fa-tags fa-sm fa-fw"></i></a></li></ul><div class="statistics flex justify-around my-2.5"><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/tags"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">48</div><div class="label text-third-text-color text-sm">Tags</div></a><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/categories"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">7</div><div class="label text-third-text-color text-sm">Categories</div></a><a class="item tag-count-item flex flex-col justify-center items-center w-20" href="/archives"><div class="number text-2xl sm:text-xl text-second-text-color font-semibold">52</div><div class="label text-third-text-color text-sm">Posts</div></a></div></div><div class="window-mask"></div></header></div><div class="main-content-body transition-fade-up"><div class="main-content"><div class="post-page-container flex relative justify-between box-border w-full h-full"><div class="article-content-container"><div class="article-title relative w-full"><div class="w-full flex items-center pt-6 justify-start"><h1 class="article-title-regular text-second-text-color tracking-tight text-4xl md:text-6xl font-semibold px-2 sm:px-6 md:px-8 py-3">Analysis of Population Genetic Structure</h1></div></div><div class="article-header flex flex-row gap-2 items-center px-2 sm:px-6 md:px-8"><div class="avatar w-[46px] h-[46px] flex-shrink-0 rounded-medium border border-border-color p-[1px]"><img src="/images/profile.jpg"></div><div class="info flex flex-col justify-between"><div class="author flex items-center"><span class="name text-default-text-color text-lg font-semibold">Yuchen</span></div><div class="meta-info"><div class="article-meta-info"><span class="article-date article-meta-item"><i class="fa-regular fa-pen-fancy"></i>&nbsp; <span class="desktop">2025-04-08 20:16</span> <span class="mobile">2025-04-08 20:16</span> <span class="hover-info">Created</span> </span><span class="article-date article-meta-item"><i class="fa-regular fa-wrench"></i>&nbsp; <span class="desktop">2025-05-05 20:37:47</span> <span class="mobile">2025-05-05 20:37:47</span> <span class="hover-info">Updated</span> </span><span class="article-categories article-meta-item"><i class="fa-regular fa-folders"></i>&nbsp;<ul><li><a href="/categories/Heredity-and-evolution/">Heredity and evolution</a>&nbsp;</li></ul></span><span class="article-tags article-meta-item"><i class="fa-regular fa-tags"></i>&nbsp;<ul><li><a href="/tags/course/">course</a>&nbsp;</li><li>| <a href="/tags/Hardy-Weinberg/">Hardy-Weinberg</a>&nbsp;</li><li>| <a href="/tags/Population-Genetic-Structure/">Population Genetic Structure</a>&nbsp;</li></ul></span><span class="article-pv article-meta-item"><i class="fa-regular fa-eye"></i>&nbsp;<span id="busuanzi_value_page_pv"></span></span></div></div></div></div><div class="article-content markdown-body px-2 sm:px-6 md:px-8 pb-8"><p>Genetic structure: 遗传多态性在<strong>时间</strong>（不同世代）上和<strong>空间</strong>（不同地理区域，基因组区）上的不同分布模式</p><h2 id="Identify-genetic-differences-among-populations"><a href="#Identify-genetic-differences-among-populations" class="headerlink" title="Identify genetic differences among populations"></a>Identify genetic differences among populations</h2><p>认为，相同群体的HWE 偏移相似，因此可以聚类F值，揭示群体分层现象。</p><p>其理论依据为：<br><strong>亚群体间等位基因频率方差（σ2）是HWE偏移的核心驱动力，而FST直接关联于此方差。</strong><br>此方法需结合遗传分化分析与统计验证（如AMOVA），以确保结果的稳健性。</p><h3 id="HWE-deviations-–-Genotype-Frequency-in-Sub-Populations"><a href="#HWE-deviations-–-Genotype-Frequency-in-Sub-Populations" class="headerlink" title="HWE deviations – Genotype Frequency in Sub-Populations"></a>HWE deviations – Genotype Frequency in Sub-Populations</h3><p><strong>实际</strong>观察到的基因型频率与HWE<strong>预期</strong>值不符时，称为HWE的偏离。 F 是固定指数，反映了基因型频率与Hardy-Weinberg预期的偏离程度。</p><ul><li>$X_{11}$ 是纯合基因型 A&#x2F;A 的频率，其期望频率为$ x_1^2$ 。偏离后：$X_{11} &#x3D; (1 - F) x_1^2 + F x_1 $。（	<strong>$(1 - F) x_1^2$</strong> 代表没有近交时的频率，仍然符合HWE预期。<strong>$F x_1$</strong> 表示由于近交而使得纯合基因型 A&#x2F;A 的频率增高的部分）</li><li>$X_{12}$ 是杂合基因型 A&#x2F;a 的频率，其期望频率为 $2x_1 x_2$ 。偏离后：$ X_{12} &#x3D; 2(1 - F) x_1 x_2$。（由于近交会增加纯合基因型（A&#x2F;A 或 a&#x2F;a）的比例，杂合基因型 A&#x2F;a 会减少。）</li><li>$X_{22}$ 是纯合基因型 a&#x2F;a 的频率，其期望频率为$ x_2^2$ 。偏离后：$X_{22} &#x3D; (1 - F) x_2^2 + F x_2 $</li></ul><p>最终得到$F &#x3D; \frac{2x_1 x_2 - X_{12}}{2 x_1 x_2}$.（通过已知的基因型频率$ X_{12} $和等位基因频率$ x_1$ 和$ x_2$ 来计算偏离的程度，从而得出固定指数 F ）</p><p>如果 $F &#x3D; 0$，则群体完全符合HWE，基因型频率与期望一致；如果 $F &gt; 0$，则群体存在近交，基因型频率会偏离预期，认为遗传群体出现结构化；如果 $F &lt; 0$，认为杂合优势巨大，杂合子存活几率更大，或者，有刻意避免近交。</p><div class="note-large default"><div class="notel-title rounded-t-lg p-3 font-bold text-lg flex flex-row gap-2 items-center"><i class="notel-icon fa-solid fa-info"></i><p>Effect</p></div><div class="notel-content"><p>The Wahlund effect refers to <strong>reduction of heterozygosit</strong>y in a population caused by subpopulation structure.</p><p>存在一个平衡，population structure减少异致性（即Aa增多），genetic drift, geographic barriers 增加异致性</p></div></div><p><img lazyload src="/images/loading.svg" data-src="/img/img-HWE/2.png"></p><h3 id="Genotype-Frequency-in-whole-population-（HEW-deviation）"><a href="#Genotype-Frequency-in-whole-population-（HEW-deviation）" class="headerlink" title="Genotype Frequency in whole population （HEW deviation）"></a>Genotype Frequency in whole population （HEW deviation）</h3><p>$$X_{11} &#x3D; \sum_{k&#x3D;1}^S w_k x_k^2 &#x3D; \overline{x}^2 + \sigma^2 $$,</p><p>$$X_{12} &#x3D; \sum_{k&#x3D;1}^S w_k 2x_k(1 - x_k) &#x3D; 2\overline{x}(1 - \overline{x}) - 2\sigma^2 $$,</p><p>$$X_{22} &#x3D; \sum_{k&#x3D;1}^S w_k (1 - x_k)^2 &#x3D; (1 - \overline{x})^2 + \sigma^2 $$</p><p>其中：$$\overline{x} &#x3D; \sum_{k&#x3D;1}^S w_k x_k \ $$,$$\sigma^2 &#x3D; \sum_{k&#x3D;1}^S w_k (x_k - \overline{x})^2$$</p><p>w代表是weighr，权重，将总体的基因型频率将其与上方的HWE deviations进行比较，得到</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-HWE/1.png"></p><ul><li><strong>理论预期</strong>：若所有亚群体完全同质（σ2&#x3D;0），整体基因型频率应满足HWE。</li><li><strong>实际偏离</strong>：当亚群体间存在遗传分化（σ2&gt;0），整体基因型频率将偏离HWE。</li></ul><p><strong>结论</strong>：方差 σ2直接量化了 <strong>群体分层程度</strong>（亚群体间遗传分化）。</p><h2 id="Population-Differentiation-Index"><a href="#Population-Differentiation-Index" class="headerlink" title="Population Differentiation Index"></a>Population Differentiation Index</h2><p>已经知道了群体存在亚结构分化，我有很多群体，如何比较不同群体之间分化的程度的差异？直接比较两个群体中所有SNP变异的mean？这是肯定不行的，两个mean可以比较的前提是两个数据满足<strong>正态+方差齐性</strong>，SNP作为一个偏态统计量，比较分位点</p><h3 id="F-statistics"><a href="#F-statistics" class="headerlink" title="F-statistics"></a>F-statistics</h3><p><img lazyload src="/images/loading.svg" data-src="/img/img-HWE/3.png"></p><p>Different F-statistics for different scales</p><ul><li>Individual (I)</li><li>Subpopulation (S)</li><li>Total population (T)</li></ul><p>$$F_{IS}&#x3D;1-\frac{H_I}{H_S}$$</p><p>$$F_{ST}&#x3D;1-\frac{H_S}{H_T}$$</p><p>$$F_{IT}&#x3D;1-\frac{H_I}{H_T}$$</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-HWE/3.png"></p><ul><li><strong>FST</strong>：亚群体间的分化 – <strong>两种群差异</strong></li><li><strong>FIS</strong>：亚群体内的近交 – <strong>种内差异</strong></li><li><strong>FIT</strong>：总“损失”的杂合度 – <strong>种内+两种间差异</strong></li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-HWE/4.png"></p><h3 id="Unbiased-estimates-of-FST"><a href="#Unbiased-estimates-of-FST" class="headerlink" title="Unbiased estimates of FST"></a>Unbiased estimates of FST</h3><p>FST是衡量群体间遗传分化的核心指标，传统计算基于杂合度（Ht和Hs）的比值。</p><p>无偏估计的目标是修正传统方法的系统性偏差，使估计值更接近真实FST。核心策略包括：</p><ul><li><strong>考虑样本量权重</strong>：通过加权平均（如Weir-Cockerham方法）减少小样本的误差。</li><li><strong>分解方差来源</strong>：用方差分析（ANOVA）分离群体间和群体内的遗传变异。</li><li><strong>校正等位基因频率波动</strong>：利用统计学方法（如矩估计、最大似然）调整抽样误差。</li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-HWE/5.png"></p><h3 id="Degree-of-F-statistics"><a href="#Degree-of-F-statistics" class="headerlink" title="Degree of F statistics"></a>Degree of F statistics</h3><ul><li>FST ranges from 0-1</li><li>0 &#x3D; no genetic differentiation</li><li>0.00-0.05 &#x3D; little genetic diff.</li><li>0.05-0.15 &#x3D; moderate genetic diff.</li><li>0.15-0.25 &#x3D; great genetic diff.</li><li>0.25-1.00 &#x3D; very great genetic diff.</li><li>1 &#x3D; complete genetic diff.</li></ul><div class="note-large default"><div class="notel-title rounded-t-lg p-3 font-bold text-lg flex flex-row gap-2 items-center"><i class="notel-icon fa-solid fa-info"></i><p>感性认知</p></div><div class="notel-content"><p>中欧差异0.12，中日0.06</p></div></div><h2 id="种群稳态（Equilibrium）"><a href="#种群稳态（Equilibrium）" class="headerlink" title="种群稳态（Equilibrium）"></a>种群稳态（Equilibrium）</h2><p><strong>种群稳态</strong>指在基因流（迁移）和遗传漂变（随机波动）的共同作用下，群体间的遗传分化（FST）达到动态平衡，不再随时间变化。此时，基因流带来的混合效应与遗传漂变造成的分化效应相互抵消。</p><p>$$F_{ST}&#x3D;\frac{1}{1+4Nm}$$</p><ul><li>N：亚群体的<strong>有效种群大小</strong>（个体数）。</li><li>m：<strong>迁移率</strong>（每代个体迁入的比例）。</li><li>4Nm：反映基因流对分化的抵消能力。</li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-HWE/6.png"></p><p>感性的认知到：就算每一代又一个出去通婚，造成了FST急剧的下降</p><h2 id="Population-genetic-structure-analysis-method"><a href="#Population-genetic-structure-analysis-method" class="headerlink" title="Population genetic structure analysis method"></a>Population genetic structure analysis method</h2><h3 id="Hierarchical-F-statistics"><a href="#Hierarchical-F-statistics" class="headerlink" title="Hierarchical F statistics"></a>Hierarchical F statistics</h3><p>见上</p><h3 id="Gene-tree-based"><a href="#Gene-tree-based" class="headerlink" title="Gene tree based"></a>Gene tree based</h3><p>Phylogenetic tree of 280 Oryza accessions, based on whole-genome SNPs.</p><p><img lazyload src="/images/loading.svg" data-src="/img/img-HWE/7.png"></p><p><img lazyload src="/images/loading.svg" data-src="/img/img-HWE/9.png"></p><h3 id="Factor-analysis"><a href="#Factor-analysis" class="headerlink" title="Factor analysis"></a>Factor analysis</h3><p><img lazyload src="/images/loading.svg" data-src="/img/img-HWE/8.png"></p><h3 id="Principal-Component-Analysis"><a href="#Principal-Component-Analysis" class="headerlink" title="Principal Component Analysis"></a>Principal Component Analysis</h3><ol><li><p>Data standardization (Zero-mean)</p></li><li><p>Calculate the covariance matrix</p><ol><li>协方差矩阵衡量的是两个特征之间的线性相关性</li><li>$\text{Cov} &#x3D; \begin{bmatrix} \text{var}(x1) &amp; \text{cov}(x1, x2) \ \text{cov}(x2, x1) &amp; \text{var}(x2) \end{bmatrix}$</li><li>记录了各特征之间的相关性</li></ol></li><li><p>Calculate eigenvalues and eigenvectors – <strong>对协方差矩阵做特征分解</strong></p><ol><li><p>解这个方程：$C \mathbf{v} &#x3D; \lambda \mathbf{v}$</p><p>结果是：</p><ul><li>一组特征值$ \lambda_1 \ge \lambda_2 \ge \cdots \ge \lambda_d$</li><li>对应的特征向量$ \mathbf{v}_1, \mathbf{v}_2, \cdots, \mathbf{v}_d$</li></ul><p>这些特征向量就是<strong>新的坐标轴方向</strong>（主成分方向），每个方向的重要程度由特征值大小决定（即解释的数据方差）。</p></li><li><p>特征值(eigenvalues)告诉你每个主成分的“重要程度”（即捕获的方差信息）</p></li><li><p>特征向量(eigenvectors)告诉你主成分的方向</p></li><li><p>特征向量和特征值总是成对出现，因此每个特征向量都有一个特征值， 它们的数量等于数据的维数。</p></li></ol></li><li><p>Choose the largest main ingredient</p></li><li><p>Projection data, complete the dimension reduction</p></li></ol><h3 id="STRUCTURE-analysis"><a href="#STRUCTURE-analysis" class="headerlink" title="STRUCTURE analysis"></a>STRUCTURE analysis</h3><ul><li>基于Bayesian模型推断个体的种群归属及混合比例。</li><li><strong>基本思想</strong>：通过假设K个种群，最大化个体基因型的解释能力。</li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-HWE/10.png"></p><h3 id="Frappe"><a href="#Frappe" class="headerlink" title="Frappe"></a>Frappe</h3><ul><li>与STRUCTURE类似，也用于推断种群结构。</li><li><strong>基本思想</strong>：基于<strong>最大似然</strong>而不是Bayesian方法。</li></ul><p><img lazyload src="/images/loading.svg" data-src="/img/img-HWE/11.png"></p><h3 id="Admixture"><a href="#Admixture" class="headerlink" title="Admixture"></a>Admixture</h3><ul><li>推断个体的祖先组成。</li><li><strong>基本思想</strong>：模型与STRUCTURE类似，但使用最大似然估计，速度更快。</li></ul><h3 id="SABER"><a href="#SABER" class="headerlink" title="SABER"></a>SABER</h3><ul><li>估算局部祖先（local ancestry）。</li><li><strong>基本思想</strong>：使用隐马尔可夫模型（HMM）追踪个体染色体上的祖先来源。</li></ul><h3 id="HAPMIX"><a href="#HAPMIX" class="headerlink" title="HAPMIX"></a>HAPMIX</h3><ul><li>推断局部祖先并检测选择信号。</li><li><strong>基本思想</strong>：也是基于HMM模型，特别适用于双源混合群体（如非洲裔&#x2F;欧洲裔混合）。</li></ul></div><div class="post-copyright-info w-full my-8 px-2 sm:px-6 md:px-8"><div class="article-copyright-info-container"><ul><li><strong>Title:</strong> Analysis of Population Genetic Structure</li><li><strong>Author:</strong> Yuchen</li><li><strong>Created at :</strong> 2025-04-08 20:16:00</li><li><strong>Updated at :</strong> 2025-05-05 20:37:47</li><li><strong>Link:</strong> https://yuchen-starfall.github.io/2025/04/08/Analysis of Population Genetic Structure/</li><li><strong>License: </strong>This work is licensed under <a class="license" target="_blank" rel="noopener" href="https://creativecommons.org/licenses/by-nc-sa/4.0">CC BY-NC-SA 4.0</a>.</li></ul></div></div><ul class="post-tags-box text-lg mt-1.5 flex-wrap justify-center flex md:hidden"><li class="tag-item mx-0.5"><a href="/tags/course/">#course</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/Hardy-Weinberg/">#Hardy-Weinberg</a>&nbsp;</li><li class="tag-item mx-0.5"><a href="/tags/Population-Genetic-Structure/">#Population Genetic Structure</a>&nbsp;</li></ul><div class="article-nav my-8 flex justify-between items-center px-2 sm:px-6 md:px-8"><div class="article-prev border-border-color shadow-redefine-flat shadow-shadow-color-2 rounded-medium px-4 py-2 hover:shadow-redefine-flat-hover hover:shadow-shadow-color-2"><a class="prev" rel="prev" href="/2025/04/26/SV%20calling/"><span class="left arrow-icon flex justify-center items-center"><i class="fa-solid fa-chevron-left"></i> </span><span class="title flex justify-center items-center"><span class="post-nav-title-item">SV calling</span> <span class="post-nav-item">Prev posts</span></span></a></div><div class="article-next border-border-color shadow-redefine-flat shadow-shadow-color-2 rounded-medium px-4 py-2 hover:shadow-redefine-flat-hover hover:shadow-shadow-color-2"><a class="next" rel="next" href="/2025/04/01/GWAS%20statistics%20methods/"><span class="title flex justify-center items-center"><span class="post-nav-title-item">Statistical methods for population association studies</span> <span class="post-nav-item">Next posts</span> </span><span class="right arrow-icon flex justify-center items-center"><i class="fa-solid fa-chevron-right"></i></span></a></div></div><div class="comment-container px-2 sm:px-6 md:px-8 pb-8"><div class="comments-container mt-10 w-full"><div id="comment-anchor" class="w-full h-2.5"></div><div class="comment-area-title w-full my-1.5 md:my-2.5 text-xl md:text-3xl font-bold">Comments</div><div id="giscus-container"></div><script data-swup-reload-script defer>async function loadGiscus(){const t={src:"https://giscus.app/client.js","data-repo":"Yuchen-starfall/Yuchen-starfall.github.io","data-repo-id":"R_kgDONY6r-Q","data-category":"Announcements# Github discussion category","data-category-id":"DIC_kwDONY6r-c4Ck8Zm","data-mapping":"pathname","data-strict":"0","data-reactions-enabled":"1","data-emit-metadata":"1","data-theme":"preferred_color_scheme","data-lang":"en","data-input-position":"bottom","data-loading":"lazy",crossorigin:"anonymous",async:!0},a=document.createElement("script");for(const e in t)a.setAttribute(e,t[e]);document.getElementById("giscus-container").appendChild(a)}{let t=setTimeout(()=>{loadGiscus(),clearTimeout(t)},1e3)}</script></div></div></div><div class="toc-content-container"><div class="post-toc-wrap"><div class="post-toc"><div class="toc-title">On this page</div><div class="page-title">Analysis of Population Genetic Structure</div><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#Identify-genetic-differences-among-populations"><span class="nav-text">Identify genetic differences among populations</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#HWE-deviations-%E2%80%93-Genotype-Frequency-in-Sub-Populations"><span class="nav-text">HWE deviations – Genotype Frequency in Sub-Populations</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Genotype-Frequency-in-whole-population-%EF%BC%88HEW-deviation%EF%BC%89"><span class="nav-text">Genotype Frequency in whole population （HEW deviation）</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Population-Differentiation-Index"><span class="nav-text">Population Differentiation Index</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#F-statistics"><span class="nav-text">F-statistics</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Unbiased-estimates-of-FST"><span class="nav-text">Unbiased estimates of FST</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Degree-of-F-statistics"><span class="nav-text">Degree of F statistics</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%A7%8D%E7%BE%A4%E7%A8%B3%E6%80%81%EF%BC%88Equilibrium%EF%BC%89"><span class="nav-text">种群稳态（Equilibrium）</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Population-genetic-structure-analysis-method"><span class="nav-text">Population genetic structure analysis method</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Hierarchical-F-statistics"><span class="nav-text">Hierarchical F statistics</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Gene-tree-based"><span class="nav-text">Gene tree based</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Factor-analysis"><span class="nav-text">Factor analysis</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Principal-Component-Analysis"><span class="nav-text">Principal Component Analysis</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#STRUCTURE-analysis"><span class="nav-text">STRUCTURE analysis</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Frappe"><span class="nav-text">Frappe</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Admixture"><span class="nav-text">Admixture</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#SABER"><span class="nav-text">SABER</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#HAPMIX"><span class="nav-text">HAPMIX</span></a></li></ol></li></ol></div></div></div></div></div></div><div class="main-content-footer"><footer class="footer mt-5 py-5 h-auto text-base text-third-text-color relative border-t-2 border-t-border-color"><div class="info-container py-3 text-center"><div class="text-center">&copy; <span>2024</span> - 2025&nbsp;&nbsp;<i class="fa-solid fa-heart fa-beat" style="--fa-animation-duration:0.5s;color:#f54545"></i>&nbsp;&nbsp;<a href="/">Yuchen</a><p class="post-count space-x-0.5"><span>52 posts in total</span></p></div><script data-swup-reload-script src="https://cn.vercount.one/js"></script><div class="relative text-center lg:absolute lg:right-[20px] lg:top-1/2 lg:-translate-y-1/2 lg:text-right"><span id="busuanzi_container_site_uv" class="lg:!block"><span class="text-sm">VISITOR COUNT</span> <span id="busuanzi_value_site_uv"></span> </span><span id="busuanzi_container_site_pv" class="lg:!block"><span class="text-sm">TOTAL PAGE VIEWS</span> <span id="busuanzi_value_site_pv"></span></span></div><div class="relative text-center lg:absolute lg:left-[20px] lg:top-1/2 lg:-translate-y-1/2 lg:text-left"><span class="lg:block text-sm">POWERED BY <?xml version="1.0" encoding="utf-8"?><!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"><svg class="relative top-[2px] inline-block align-baseline" version="1.1" id="圖層_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="1rem" height="1rem" viewBox="0 0 512 512" enable-background="new 0 0 512 512" xml:space="preserve"><path fill="#0E83CD" d="M256.4,25.8l-200,115.5L56,371.5l199.6,114.7l200-115.5l0.4-230.2L256.4,25.8z M349,354.6l-18.4,10.7l-18.6-11V275H200v79.6l-18.4,10.7l-18.6-11v-197l18.5-10.6l18.5,10.8V237h112v-79.6l18.5-10.6l18.5,10.8V354.6z"/></svg><a target="_blank" class="text-base" href="https://hexo.io">Hexo</a></span> <span class="text-sm lg:block">THEME&nbsp;<a class="text-base" target="_blank" href="https://github.com/EvanNotFound/hexo-theme-redefine">Redefine v2.8.2</a></span></div><div>Blog up for <span class="odometer" id="runtime_days"></span> days <span class="odometer" id="runtime_hours"></span> hrs <span class="odometer" id="runtime_minutes"></span> Min <span class="odometer" id="runtime_seconds"></span> Sec</div><script data-swup-reload-script>try{function odometer_init(){document.querySelectorAll(".odometer").forEach(e=>{new Odometer({el:e,format:"( ddd).dd",duration:200})})}odometer_init()}catch(e){}</script></div></footer></div></div><div class="post-tools"><div class="post-tools-container"><ul class="article-tools-list"><li class="right-bottom-tools page-aside-toggle"><i class="fa-regular fa-outdent"></i></li><li class="go-comment"><i class="fa-regular fa-comments"></i></li></ul></div></div><div class="right-side-tools-container"><div class="side-tools-container"><ul class="hidden-tools-list"><li class="right-bottom-tools tool-font-adjust-plus flex justify-center items-center"><i class="fa-regular fa-magnifying-glass-plus"></i></li><li class="right-bottom-tools tool-font-adjust-minus flex justify-center items-center"><i class="fa-regular fa-magnifying-glass-minus"></i></li><li class="right-bottom-tools tool-dark-light-toggle flex justify-center items-center"><i class="fa-regular fa-moon"></i></li><li class="right-bottom-tools tool-scroll-to-top flex justify-center items-center"><i class="fa-regular fa-arrow-up"></i></li><li class="right-bottom-tools tool-scroll-to-bottom flex justify-center items-center"><i class="fa-regular fa-arrow-down"></i></li></ul><ul class="visible-tools-list"><li class="right-bottom-tools toggle-tools-list flex justify-center items-center"><i class="fa-regular fa-cog fa-spin"></i></li></ul></div></div><div class="image-viewer-container"><img src=""></div></main><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/Swup.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupSlideTheme.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupScriptsPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupProgressPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupScrollPlugin.min.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/SwupPreloadPlugin.min.js"></script><script>const swup=new Swup({plugins:[new SwupScriptsPlugin({optin:!0}),new SwupProgressPlugin,new SwupScrollPlugin({offset:80}),new SwupSlideTheme({mainElement:".main-content-body"}),new SwupPreloadPlugin],containers:["#swup"]})</script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/imageViewer.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/utils.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/main.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/navbarShrink.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/scrollTopBottom.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/lightDarkSwitch.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/categoryList.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/codeBlock.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/lazyload.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/runtime.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/odometer.min.js"></script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/assets/odometer-theme-minimal.css"><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/Typed.min.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/typed.js"></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/minimasonry.min.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/masonry.js"></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/tools/tocToggle.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/toc.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/plugins/tabs.js" data-swup-reload-script></script><script src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/libs/moment-with-locales.min.js" data-swup-reload-script></script><script type="module" src="https://cdn.jsdelivr.net/npm/hexo-theme-redefine@2.8.2/source/js/build/layouts/essays.js" data-swup-reload-script></script></body></html>